# load packages
import pandas as pd

# load samples
samples = pd.read_table("data/samples.tsv", sep="\t", dtype={"run": str, "replicate": str}).set_index(["run","replicate"], drop=False)

# can't have print statements if buidling DAG with dot
# print(samples)

# split into paired-end and single-end
samplesPe = samples[samples["layout"] == "PE"]
#print(samplesPe)

samplesSe = samples[samples["layout"] == "SE"]
#print(samplesSe)

# load list of chromosome names
chroms = pd.read_table("data/chromosomes.tsv", sep="\t", dtype={"assembly": str, "chromosome": str}).set_index(["assembly", "chromosome"], drop=False)

#ruleorder: download_pe > download_se

#ruleorder: fastp_pe > fastp_se

#ruleorder: bwa_mem_pe > bwa_mem_se

#ruleorder: cat_replicates_se > fastp_se > download_pe > download_se 

wildcard_constraints:
	sampleSe="[A-Za-z0-9]+",
	runSe="[A-Za-z0-9]+"

# function to determine if --use-conda is enabled
def get_conda_status(wildcards):
	return workflow.use_conda

# function to define thread count per rule
# 1 thread per gigabyte of input data + 1, up to a max of 128
def get_thread_count(wildcards, input):
	a = (input.size//1000000000 + 1)
	return min(a, 128)

# function to define memory limit
# 2 GB per thread, but increase limit linearly if job fails
def get_mem_mb(wildcards, attempt):
    return 2000 * attempt

#rule all:
#	input:
		#expand("{sampleSe}.fastq.gz", sampleSe=samplesSe.index),
		#expand("{samplePe}_{end}.fastq.gz", samplePe=samplesPe.index, end = [1,2]),
		#expand("trimmed_reads/{sampleSe}_trim.fastq.gz", sampleSe=samplesSe.index),
		#expand("trimmed_reads/{samplePe}_{end}_trim.fastq.gz", samplePe=samplesPe.index, end = [1,2]),
#		"multiqc_report.html",
		#expand("mapped_reads/{samplePe}_pe.bam", samplePe=samplesPe.index),
		#expand("mapped_reads/{sampleSe}_se.bam", sampleSe=samplesSe.index)
		#expand("sorted_reads/{samplePe}_pe.bam", samplePe=samplesPe.index),
		#expand("sorted_reads/{sampleSe}_se.bam", sampleSe=samplesSe.index)
		#expand("sorted_marked_reads/{sampleSe}_se.bam", sampleSe=samplesSe.index),
		#expand("sorted_marked_reads/{samplePe}_pe.bam", samplePe=samplesPe.index)
		#expand("sorted_marked_reads/{sampleSe}_se.bam.bai", sampleSe=samplesSe.index),
		#expand("sorted_marked_reads/{samplePe}_pe.bam.bai", samplePe=samplesPe.index)
		#expand("calls/{sampleSe}_se.g.vcf.gz", sampleSe=samplesSe.index),
		#expand("calls/{samplePe}_pe.g.vcf.gz", samplePe=samplesPe.index),
		#expand("combinedCalls_{chrom}.g.vcf.gz", chrom=chroms.index),
#		expand("jointGenotypes_{chrom}.vcf.gz", chrom=chroms.index)

rule all:
	input:
		expand("filtered_variantAndInvariant_{assembly}_{chromosome}.frq.count", zip, assembly=chroms.index.get_level_values("assembly"), chromosome=chroms.index.get_level_values("chromosome")),
		"multiqc_report.html",
		expand("out_{assembly}_{chromosome}.frq", zip, assembly=chroms.index.get_level_values("assembly"), chromosome=chroms.index.get_level_values("chromosome")),
		expand("out_{assembly}_{chromosome}.ldepth.mean", zip, assembly=chroms.index.get_level_values("assembly"), chromosome=chroms.index.get_level_values("chromosome")),
		expand("out_{assembly}_{chromosome}.lqual", zip, assembly=chroms.index.get_level_values("assembly"), chromosome=chroms.index.get_level_values("chromosome")),
		expand("out_{assembly}_{chromosome}.lmiss", zip, assembly=chroms.index.get_level_values("assembly"), chromosome=chroms.index.get_level_values("chromosome")),
		expand("{assembly}_bcd.txt", assembly=set(chroms.index.get_level_values("assembly"))),
		expand("{assembly}_jac.txt", assembly=set(chroms.index.get_level_values("assembly"))),
		expand("{assembly}_pairwise-pi.txt", assembly=set(chroms.index.get_level_values("assembly"))),
		"readStats.txt"

include: "rules/download_pe.smk"
include: "rules/download_se.smk"
include: "rules/seqkit_stats.smk"
include: "rules/fastp_se.smk"
include: "rules/fastp_pe.smk"
include: "rules/cat_replicates_se.smk"
include: "rules/cat_replicates_pe.smk"
include: "rules/get_cds_seq.smk"
include: "rules/build_coding_kmer_database.smk"
#include: "rules/kmc_tools_filter_pe.smk"
#include: "rules/kmc_tools_filter_se.smk"
include: "rules/kmc_pe.smk"
include: "rules/kmc_se.smk"
include: "rules/list_kmers.smk"
include: "rules/remove_kmer_dups.smk"
include: "rules/random_kmer_subset.smk"
include: "rules/join_kmer_counts_se.smk"
include: "rules/join_kmer_counts_pe.smk"
include: "rules/cbind_kmer_counts.smk"
include: "rules/bray_curtis_dissimilarity.smk"
include: "rules/jaccard_dissimilarity.smk"
include: "rules/multiqc.smk"
include: "rules/bwa_index.smk"
include: "rules/bwa_mem_se.smk"
include: "rules/bwa_mem_pe.smk"
include: "rules/samtools_sort.smk"
include: "rules/picard_mark_duplicates.smk"
include: "rules/samtools_index.smk"
include: "rules/samtools_faidx.smk"
include: "rules/picard_create_sequence_dictionary.smk"
#include: "rules/subset_genome.smk"
include: "rules/degenotate.smk"
include: "rules/gatk_haplotype_caller.smk"
include: "rules/gatk_combine_gvcfs.smk"
include: "rules/gatk_genotype_gvcfs.smk"
include: "rules/separate_variant_invariant_sites.smk"
include: "rules/vcftools_check_genotype_calls.smk"
include: "rules/gatk_variant_filtration_variant_sites.smk"
include: "rules/gatk_variant_filtration_invariant_sites.smk"
include: "rules/vcftools_filter_variant_sites.smk"
include: "rules/vcftools_filter_invariant_sites.smk"
include: "rules/bcftools_concat.smk"
include: "rules/vcftools_counts.smk"
include: "rules/pairwise_pi.smk"
