# Purpose: organize sra data from multiple species for input into snakemake workflow

# clear wd and load packages
```{r}
rm(list = ls())

library(data.table)

# if (!require("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("SRAdb")

# speciesList = unique(read.csv("../workflow/data/speciesList.csv")$Species)

# create empty objects for growing list of reads
allReads = NULL
```

# Compile SRA run info for each species into master list, separate paired end and single-end reads
## Amaranthus hypochondriacus
```{r}
# load data for species
amaranthus_hypochondriacus = fread("../workflow/data/SRArunInfo/SraRunInfo_amaranthus_hypochondriacus_PRJEB30531.csv.gz", data.table = F)

# subset to only appropriate reads
amaranthus_hypochondriacus = amaranthus_hypochondriacus[(amaranthus_hypochondriacus$ScientificName == "Amaranthus hypochondriacus"),]

# Check if there are any technical replicates
any(duplicated(amaranthus_hypochondriacus$Experiment))
any(duplicated(amaranthus_hypochondriacus$BioSample))

# check if any reads are single-end
any(amaranthus_hypochondriacus$spots_with_mates == 0)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = amaranthus_hypochondriacus[,"Run"], replicate = amaranthus_hypochondriacus[,"BioSample"], layout = "PE", genome = "amaranthus_hypochondriacus", ploidy = 2))
```

## Ananas comosus
```{r}
# load data for species
ananas_comosus = fread("../workflow/data/SRArunInfo/SraRunInfo_ananas_comosus_PRJNA389669.csv.gz", data.table = F)

# remove outgroups
ananas_comosus = ananas_comosus[grepl("Ananas",ananas_comosus$ScientificName),]

# Check if there are any technical replicates
any(duplicated(ananas_comosus$Experiment))
any(duplicated(ananas_comosus$SampleName))

# check if any reads are single-end
any(ananas_comosus$spots_with_mates == 0)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = ananas_comosus[,"Run"], replicate = ananas_comosus[,"BioSample"], layout = "PE", genome = "ananas_comosus", ploidy = 2))
```

## Arabidopsis halleri
```{r}
# load data for species
arabidopsis_halleri = fread("../workflow/data/SRArunInfo/SraRunInfo_arabidopsis_halleri_PRJDB3777.csv.gz", data.table = F)

# just use hiseq2000 reads so that you don't have to merge technical reps
arabidopsis_halleri = arabidopsis_halleri[(arabidopsis_halleri$Model == "Illumina HiSeq 2000"),]

# check if any reads are single-end
any(arabidopsis_halleri$spots_with_mates == 0)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = arabidopsis_halleri[,"Run"], replicate = arabidopsis_halleri[,"BioSample"], layout = "PE", genome = "arabidopsis_halleri", ploidy = 2))
```

## Arabidopsis lyrata
```{r}
arabidopsis_lyrata_1 = fread("../workflow/data/SRArunInfo/SraRunInfo_arabidopsis_lyrata_PRJEB23202.csv.gz", data.table = F)
arabidopsis_lyrata_2 = fread("../workflow/data/SRArunInfo/SraRunInfo_arabidopsis_lyrata_PRJEB30473.csv.gz", data.table = F)
arabidopsis_lyrata_3 = fread("../workflow/data/SRArunInfo/SraRunInfo_arabidopsis_lyrata_PRJNA284572.csv.gz", data.table = F)

# subset only A. lyrata
arabidopsis_lyrata_1 = arabidopsis_lyrata_1[grepl("Arabidopsis lyrata", arabidopsis_lyrata_1$ScientificName),]
arabidopsis_lyrata_2 = arabidopsis_lyrata_2[grepl("Arabidopsis lyrata", arabidopsis_lyrata_2$ScientificName),]
arabidopsis_lyrata_3 = arabidopsis_lyrata_3[grepl("Arabidopsis lyrata", arabidopsis_lyrata_3$ScientificName),]

# concatenate datasets
arabidopsis_lyrata = rbind(arabidopsis_lyrata_1, arabidopsis_lyrata_2, arabidopsis_lyrata_3)

# check if any reads are single-end
any(arabidopsis_lyrata$spots_with_mates == 0)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = arabidopsis_lyrata[,"Run"], replicate = arabidopsis_lyrata[,"BioSample"], layout = "PE", genome = "arabidopsis_lyrata", ploidy = 2))
```


## Arabidopsis thaliana
```{r}
# load data for species
arabidopsis_thaliana = fread("../workflow/data/SRArunInfo/SraRunInfo_arabidopsis_thaliana_PRJNA273563.csv.gz", data.table = F)

# assign reads as single end or paired end
arabidopsis_thaliana$layout = NA
arabidopsis_thaliana$layout[(arabidopsis_thaliana$spots_with_mates > 0)] = "PE"
arabidopsis_thaliana$layout[(arabidopsis_thaliana$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = arabidopsis_thaliana[,"Run"], replicate = arabidopsis_thaliana[,"BioSample"], layout = arabidopsis_thaliana[,"layout"], genome = "arabidopsis_thaliana", ploidy = 2))
```

## Beta vulgaris
```{r}
# load data for species
beta_vulgaris = fread("../workflow/data/SRArunInfo/SraRunInfo_beta_vulgaris_PRJNA815240.csv.gz", data.table = F)

# remove close relatives
beta_vulgaris = beta_vulgaris[grepl("Beta vulgaris",beta_vulgaris$ScientificName),]

# check if any reads are single-end
# double-checked that spots without mates are actually PE
any(beta_vulgaris$spots_with_mates == 0)

beta_vulgaris$layout = NA
beta_vulgaris$layout = "PE"
# beta_vulgaris$layout[(beta_vulgaris$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = beta_vulgaris[,"Run"], replicate = beta_vulgaris[,"BioSample"], layout = beta_vulgaris[,"layout"], genome = "beta_vulgaris", ploidy = 2))

```

## Boechera stricta
```{r}
boechera_stricta_1 = fread("../workflow/data/SRArunInfo/SraRunInfo_boechera_stricta_1_2022-12-01.csv.gz", data.table = F)
boechera_stricta_2 = fread("../workflow/data/SRArunInfo/SraRunInfo_boechera_stricta_2_2022-12-01.csv.gz", data.table = F)

# subset second dataset by values in this publication
# https://doi.org/10.1093/gbe/evac043
# SRP134283-SRP134373, SRP134393-SRP134433, SRP134436-SRP134479, SRP134481-SRP134572, and SRP134581-SRP134671
sraprojects = paste("SRP", c(134283:134373, 134393:134433, 134436:134479, 134481:134572, 134581:134671), sep = "")

boechera_stricta_2 = boechera_stricta_2[(boechera_stricta_2$SRAStudy %in% sraprojects),]

# bind both datasets
boechera_stricta = rbind(boechera_stricta_1, boechera_stricta_2)

# add layout
boechera_stricta$layout = NA
boechera_stricta$layout[(boechera_stricta$spots_with_mates > 0)] = "PE"
boechera_stricta$layout[(boechera_stricta$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = boechera_stricta[,"Run"], replicate = boechera_stricta[,"BioSample"], layout = boechera_stricta[,"layout"], genome = "boechera_stricta", ploidy = 2))

# remove technical replicates from dataset 2
# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(LibraryName in unique(boechera_stricta_2$LibraryName)){
#   techreps = boechera_stricta_2[(boechera_stricta_2$LibraryName == LibraryName),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }
# 
# boechera_stricta_2 = boechera_stricta_2[(boechera_stricta_2$Run %in% runs),]

# add reads to growing list
# allReads = rbind(allReads, data.frame(samples = boechera_stricta_1[(boechera_stricta_1$spots_with_mates > 0),"Run"], genome = "boechera_stricta"))
# 
# allReads = rbind(allReads, data.frame(samples = boechera_stricta_2[(boechera_stricta_2$spots_with_mates > 0),"Run"], genome = "boechera_stricta"))

```

## Brachypodium distachyon
```{r}
# load data for species
brachypodium_distachyon = fread("../workflow/data/SRArunInfo/SraRunInfo_brachypodium_distachyon_PRJNA605320.csv.gz", data.table = F)

# isolate only WGS datasets
brachypodium_distachyon = brachypodium_distachyon[(brachypodium_distachyon$LibraryStrategy == "WGS"),]

# check for technical replicates
any(duplicated(brachypodium_distachyon$Experiment))
any(duplicated(brachypodium_distachyon$SampleName))
any(duplicated(brachypodium_distachyon$BioSample))
any(duplicated(brachypodium_distachyon$LibraryName))

# add layout
all(brachypodium_distachyon$spots_with_mates > 0)
brachypodium_distachyon$layout = NA
brachypodium_distachyon$layout[(brachypodium_distachyon$spots_with_mates > 0)] = "PE"
brachypodium_distachyon$layout[(brachypodium_distachyon$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = brachypodium_distachyon[,"Run"], replicate = brachypodium_distachyon[,"BioSample"], layout = brachypodium_distachyon[,"layout"], genome = "brachypodium_distachyon", ploidy = 2))

```

## Brassica rapa and Brassica oleraceae
```{r}
brassica = fread("../workflow/data/SRArunInfo/SraRunInfo_PRJNA312457.csv.gz", data.table = F)

# split data into the two species
brassica_oleracea = brassica[(brassica$ScientificName == "Brassica oleracea"),]
brassica_rapa = brassica[(brassica$ScientificName == "Brassica rapa"),]

# check if any reads are single-end
any(brassica_rapa$spots_with_mates == 0)
any(brassica_oleracea$spots_with_mates == 0)

# add layout
brassica_rapa$layout = NA
brassica_rapa$layout[(brassica_rapa$spots_with_mates > 0)] = "PE"
brassica_rapa$layout[(brassica_rapa$spots_with_mates == 0)] = "SE"

brassica_oleracea$layout = NA
brassica_oleracea$layout[(brassica_oleracea$spots_with_mates > 0)] = "PE"
brassica_oleracea$layout[(brassica_oleracea$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = brassica_rapa[,"Run"], replicate = brassica_rapa[,"BioSample"], layout = brassica_rapa[,"layout"], genome = "brassica_rapa", ploidy = 2))

allReads = rbind(allReads, data.frame(run = brassica_oleracea[,"Run"], replicate = brassica_oleracea[,"BioSample"], layout = brassica_oleracea[,"layout"], genome = "brassica_oleracea", ploidy = 2))
```

## Capsella grandiflora
```{r}
capsella_grandiflora = fread("../workflow/data/SRArunInfo/SraRunInfo_capsella_grandiflora_PRJEB39830.csv.gz", data.table = F)

# subset only WGS reads
capsella_grandiflora = capsella_grandiflora[(capsella_grandiflora$LibraryStrategy == "WGS"),]

# check if any reads are single-end
any(capsella_grandiflora$spots_with_mates == 0)

# add layout
capsella_grandiflora$layout = NA
capsella_grandiflora$layout[(capsella_grandiflora$spots_with_mates > 0)] = "PE"
capsella_grandiflora$layout[(capsella_grandiflora$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = capsella_grandiflora[,"Run"], replicate = capsella_grandiflora[,"BioSample"], layout = capsella_grandiflora[,"layout"], genome = "capsella_grandiflora", ploidy = 2))
```

## Capsella rubella
```{r}
# load data
capsella_rubella = fread("../workflow/data/SRArunInfo/SraRunInfo_capsella_rubella_PRJEB6689.csv.gz", data.table = F)

# subset only this relevant species
capsella_rubella = capsella_rubella[(capsella_rubella$ScientificName == "Capsella rubella"),]

# check for technical replicates
any(duplicated(capsella_rubella$Experiment))
any(duplicated(capsella_rubella$SampleName))
any(duplicated(capsella_rubella$BioSample))

# check if any reads are single-end
any(capsella_rubella$spots_with_mates == 0)

# add layout
# checked that spots without mates are PE
capsella_rubella$layout = NA
capsella_rubella$layout = "PE"
#capsella_rubella$layout[(capsella_rubella$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = capsella_rubella[,"Run"], replicate = capsella_rubella[,"BioSample"], layout = capsella_rubella[,"layout"], genome = "capsella_rubella", ploidy = 2))
```

## Chenopodium quinoa
```{r}
# load data
chenopodium_quinoa = fread("../workflow/data/SRArunInfo/SraRunInfo_chenopodium_quinoa_PRJNA673789.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(chenopodium_quinoa$Experiment))
any(duplicated(chenopodium_quinoa$SampleName))
any(duplicated(chenopodium_quinoa$BioSample))

# check if any reads are single-end
any(chenopodium_quinoa$spots_with_mates == 0)

# add layout
chenopodium_quinoa$layout = NA
chenopodium_quinoa$layout[(chenopodium_quinoa$spots_with_mates > 0)] = "PE"
chenopodium_quinoa$layout[(chenopodium_quinoa$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = chenopodium_quinoa[,"Run"], replicate = chenopodium_quinoa[,"BioSample"], layout = chenopodium_quinoa[,"layout"], genome = "chenopodium_quinoa", ploidy = 4))
```

## Cicer arietinum
```{r}
# load data
cicer_arietinum = fread("../workflow/data/SRArunInfo/SraRunInfo_cicer_arietinum_PRJNA657888.csv.gz", data.table = F)

# subset only this relevant species
cicer_arietinum = cicer_arietinum[(cicer_arietinum$ScientificName == "Cicer arietinum"),]

# check for technical replicates
any(duplicated(cicer_arietinum$Experiment))
any(duplicated(cicer_arietinum$SampleName))
any(duplicated(cicer_arietinum$BioSample))

# check if any reads are single-end
any(cicer_arietinum$spots_with_mates == 0)

# add layout
cicer_arietinum$layout = NA
cicer_arietinum$layout[(cicer_arietinum$spots_with_mates > 0)] = "PE"
cicer_arietinum$layout[(cicer_arietinum$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = cicer_arietinum[,"Run"], replicate = cicer_arietinum[,"BioSample"], layout = cicer_arietinum[,"layout"], genome = "cicer_arietinum", ploidy = 2))
```

## Citrullus lanatus
```{r}
# load data
citrullus_lanatus = fread("../workflow/data/SRArunInfo/SraRunInfo_citrullus_lanatus_PRJNA527790.csv.gz", data.table = F)

# subset only this relevant species
citrullus_lanatus = citrullus_lanatus[(citrullus_lanatus$ScientificName == "Citrullus lanatus"),]

# check for technical replicates
any(duplicated(citrullus_lanatus$Experiment))
any(duplicated(citrullus_lanatus$SampleName))
any(duplicated(citrullus_lanatus$BioSample))

# check if any reads are single-end
any(citrullus_lanatus$spots_with_mates == 0)

# add layout
citrullus_lanatus$layout = NA
citrullus_lanatus$layout[(citrullus_lanatus$spots_with_mates > 0)] = "PE"
citrullus_lanatus$layout[(citrullus_lanatus$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = citrullus_lanatus[,"Run"], replicate = citrullus_lanatus[,"BioSample"], layout = citrullus_lanatus[,"layout"], genome = "citrullus_lanatus", ploidy = 2))
```

## Cucumis sativus
```{r}
# load data
cucumis_sativus = fread("../workflow/data/SRArunInfo/SraRunInfo_cucumis_sativus_PRJNA171718.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(cucumis_sativus$Experiment))
any(duplicated(cucumis_sativus$SampleName))
any(duplicated(cucumis_sativus$BioSample))

# check if any reads are single-end
any(cucumis_sativus$spots_with_mates == 0)

# add layout
# checked that spots without mates are PE
cucumis_sativus$layout = NA
cucumis_sativus$layout = "PE"
#cucumis_sativus$layout[(cucumis_sativus$spots_with_mates > 0)] = "PE"
#cucumis_sativus$layout[(cucumis_sativus$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = cucumis_sativus[,"Run"], replicate = cucumis_sativus[,"BioSample"], layout = cucumis_sativus[,"layout"], genome = "cucumis_sativus", ploidy = 2))
```

## Glycine max
```{r}
# load data
glycine_max_1 = fread("../workflow/data/SRArunInfo/SraRunInfo_glycine_max_PRJNA681974.csv.gz", data.table = F)
glycine_max_2 = fread("../workflow/data/SRArunInfo/SraRunInfo_glycine_max_PRJNA257011.csv.gz", data.table = F)
glycine_max_3 = fread("../workflow/data/SRArunInfo/SraRunInfo_glycine_max_PRJNA552939.csv.gz", data.table = F)

# remove glycine soja accessions
glycine_max_2 = glycine_max_2[(glycine_max_2$ScientificName == "Glycine max"),]

# check for technical replicates
any(duplicated(glycine_max_1$Experiment))
any(duplicated(glycine_max_1$SampleName))
any(duplicated(glycine_max_1$BioSample))
any(duplicated(glycine_max_1$LibraryName))

any(duplicated(glycine_max_2$Experiment))
any(duplicated(glycine_max_2$SampleName))
any(duplicated(glycine_max_2$BioSample))
any(duplicated(glycine_max_2$LibraryName))

any(duplicated(glycine_max_3$Experiment))
any(duplicated(glycine_max_3$SampleName))
any(duplicated(glycine_max_3$BioSample))
any(duplicated(glycine_max_3$LibraryName))

# bind studies together
glycine_max = rbind(glycine_max_1, glycine_max_2, glycine_max_3)

# Check for duplicates
any(duplicated(glycine_max$Experiment))
any(duplicated(glycine_max$SampleName))
any(duplicated(glycine_max$BioSample))
any(duplicated(glycine_max$LibraryName))

# add layout
glycine_max$layout = NA
glycine_max$layout[(glycine_max$spots_with_mates > 0)] = "PE"
glycine_max$layout[(glycine_max$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = glycine_max[,"Run"], replicate = glycine_max[,"BioSample"], layout = glycine_max[,"layout"], genome = "glycine_max", ploidy = 2))
```

## Glycine soja
```{r}
# load data
glycine_soja = fread("../workflow/data/SRArunInfo/SraRunInfo_glycine_soja_PRJNA597660.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(glycine_soja$Experiment))
any(duplicated(glycine_soja$SampleName))
any(duplicated(glycine_soja$BioSample))

# check if any reads are single-end
any(glycine_soja$spots_with_mates == 0)

# add layout
glycine_soja$layout = NA
glycine_soja$layout[(glycine_soja$spots_with_mates > 0)] = "PE"
glycine_soja$layout[(glycine_soja$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = glycine_soja[,"Run"], replicate = glycine_soja[,"BioSample"], layout = glycine_soja[,"layout"], genome = "glycine_soja", ploidy = 2))
```

## Gossypium barbadense
```{r}
gossypium_barbadense = fread("../workflow/data/SRArunInfo/SraRunInfo_gossypium_barbadense_PRJNA720818.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(gossypium_barbadense$Experiment))
any(duplicated(gossypium_barbadense$SampleName))
any(duplicated(gossypium_barbadense$BioSample))

# check if any reads are single-end
any(gossypium_barbadense$spots_with_mates == 0)

# add layout
gossypium_barbadense$layout = NA
gossypium_barbadense$layout[(gossypium_barbadense$spots_with_mates > 0)] = "PE"
gossypium_barbadense$layout[(gossypium_barbadense$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = gossypium_barbadense[,"Run"], replicate = gossypium_barbadense[,"BioSample"], layout = gossypium_barbadense[,"layout"], genome = "gossypium_barbadense", ploidy = 4))
```

## Gossypium hirsutum
```{r}
gossypium_hirsutum = fread("../workflow/data/SRArunInfo/SraRunInfo_gossypium_hirsutum_PRJNA680449.csv.gz", data.table = F)

# subset only WGS reads
gossypium_hirsutum = gossypium_hirsutum[(gossypium_hirsutum$LibraryStrategy == "WGS" & gossypium_hirsutum$ScientificName == "Gossypium hirsutum" & gossypium_hirsutum$Platform == "ILLUMINA" & gossypium_hirsutum$LibrarySelection == "RANDOM"),]

# check for technical replicates
any(duplicated(gossypium_hirsutum$Experiment))
any(duplicated(gossypium_hirsutum$SampleName))
any(duplicated(gossypium_hirsutum$BioSample))

# check if any reads are single-end
any(gossypium_hirsutum$spots_with_mates == 0)

# add layout
gossypium_hirsutum$layout = NA
gossypium_hirsutum$layout[(gossypium_hirsutum$spots_with_mates > 0)] = "PE"
gossypium_hirsutum$layout[(gossypium_hirsutum$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = gossypium_hirsutum[,"Run"], replicate = gossypium_hirsutum[,"BioSample"], layout = gossypium_hirsutum[,"layout"], genome = "gossypium_hirsutum", ploidy = 4))
```

## Hordeum vulgare
```{r}
hordeum_vulgare = fread("../workflow/data/SRArunInfo/SraRunInfo_hordeum_vulgare_PRJEB36577.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(hordeum_vulgare$Experiment))
any(duplicated(hordeum_vulgare$SampleName))
any(duplicated(hordeum_vulgare$BioSample))

# check if any reads are single-end
any(hordeum_vulgare$spots_with_mates == 0)

# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(lib in unique(hordeum_vulgare$LibraryName)){
#   techreps = hordeum_vulgare[(hordeum_vulgare$LibraryName == lib),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }

# add layout
hordeum_vulgare$layout = NA
hordeum_vulgare$layout[(hordeum_vulgare$spots_with_mates > 0)] = "PE"
hordeum_vulgare$layout[(hordeum_vulgare$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = hordeum_vulgare[,"Run"], replicate = hordeum_vulgare[,"BioSample"], layout = hordeum_vulgare[,"layout"], genome = "hordeum_vulgare", ploidy = 2))
```

## Lactuca sativa
```{r}
lactuca_sativa = fread("../workflow/data/SRArunInfo/SraRunInfo_lactuca_sativa_PRJNA693894.csv.gz", data.table = F)

# subset relevant reads
lactuca_sativa = lactuca_sativa[(lactuca_sativa$LibraryStrategy == "WGS" & lactuca_sativa$ScientificName == "Lactuca sativa"),]

# check for technical replicates
any(duplicated(lactuca_sativa$Experiment))
any(duplicated(lactuca_sativa$SampleName))
any(duplicated(lactuca_sativa$BioSample))

# add layout
# checked that reads without mates are actually SE
lactuca_sativa$layout = NA
lactuca_sativa$layout[(lactuca_sativa$spots_with_mates > 0)] = "PE"
lactuca_sativa$layout[(lactuca_sativa$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = lactuca_sativa[,"Run"], replicate = lactuca_sativa[,"BioSample"], layout = lactuca_sativa[,"layout"], genome = "lactuca_sativa", ploidy = 2))
```

## Lotus japonicus
### Paper says 136 accessions, but I found 137 in the SRA
```{r}
lotus_japonicus = fread("../workflow/data/SRArunInfo/SraRunInfo_lotus_japonicus_PRJEB27969.csv.gz", data.table = F)

# subset relevant reads
lotus_japonicus = lotus_japonicus[(lotus_japonicus$bases != 0),]

# check for technical replicates
any(duplicated(lotus_japonicus$Experiment))
any(duplicated(lotus_japonicus$SampleName))
any(duplicated(lotus_japonicus$BioSample))

# add layout
lotus_japonicus$layout = NA
lotus_japonicus$layout[(lotus_japonicus$spots_with_mates > 0)] = "PE"
lotus_japonicus$layout[(lotus_japonicus$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = lotus_japonicus[,"Run"], replicate = lotus_japonicus[,"BioSample"], layout = lotus_japonicus[,"layout"], genome = "lotus_japonicus", ploidy = 2))
```

## Manihot esculenta
```{r}
manihot_esculenta = fread("../workflow/data/SRArunInfo/SraRunInfo_manihot_esculenta_PRJNA234389.csv.gz", data.table = F)

# subset relevant reads
manihot_esculenta = manihot_esculenta[(manihot_esculenta$LibraryStrategy == "WGS" & manihot_esculenta$ScientificName == "Manihot esculenta" & manihot_esculenta$Platform == "ILLUMINA"),]

# check for technical replicates
any(duplicated(manihot_esculenta$Experiment))
any(duplicated(manihot_esculenta$SampleName))
any(duplicated(manihot_esculenta$BioSample))

# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(biosample in unique(manihot_esculenta$BioSample)){
#   techreps = manihot_esculenta[(manihot_esculenta$BioSample == biosample),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }
# 
# manihot_esculenta = manihot_esculenta[(manihot_esculenta$Run %in% runs),]

# add layout
# reads without mates are actually PE
manihot_esculenta$layout = NA
manihot_esculenta$layout[(manihot_esculenta$spots_with_mates > 0)] = "PE"
manihot_esculenta$layout[(manihot_esculenta$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = manihot_esculenta[,"Run"], replicate = manihot_esculenta[,"BioSample"], layout = manihot_esculenta[,"layout"], genome = "manihot_esculenta", ploidy = 2))
```

## Medicago truncatula
```{r}
medicago_truncatula = fread("../workflow/data/SRArunInfo/SraRunInfo_medicago_truncatula_SRA026748.csv.gz", data.table = F)

# subset relevant reads
any(duplicated(medicago_truncatula$BioSample))
length(unique(medicago_truncatula$SampleName))

# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(biosample in unique(medicago_truncatula$SampleName)){
#   techreps = medicago_truncatula[(medicago_truncatula$SampleName == biosample),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }
# 
# medicago_truncatula = medicago_truncatula[(medicago_truncatula$Run %in% runs),]

# add layout
medicago_truncatula$layout = NA
medicago_truncatula$layout[(medicago_truncatula$spots_with_mates > 0)] = "PE"
medicago_truncatula$layout[(medicago_truncatula$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = medicago_truncatula[,"Run"], replicate = medicago_truncatula[,"BioSample"], layout = medicago_truncatula[,"layout"], genome = "medicago_truncatula", ploidy = 2))
```

## Mimulus guttatus
```{r}
mimulus_guttatus = fread("../workflow/data/SRArunInfo/SraRunInfo_mimulus_guttatus_PRJNA344904.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(mimulus_guttatus$Experiment))
any(duplicated(mimulus_guttatus$SampleName))
any(duplicated(mimulus_guttatus$BioSample))

# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(biosample in unique(mimulus_guttatus$SampleName)){
#   techreps = mimulus_guttatus[(mimulus_guttatus$SampleName == biosample),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }
# 
# mimulus_guttatus = mimulus_guttatus[(mimulus_guttatus$Run %in% runs),]

# subset relevant runs
# create sample names vector based on info in paper
samples = gsub("^", "SAMN0", 5852485:5852522)
mimulus_guttatus= mimulus_guttatus[(mimulus_guttatus$BioSample %in% samples),]

# add layout
mimulus_guttatus$layout = NA
mimulus_guttatus$layout[(mimulus_guttatus$spots_with_mates > 0)] = "PE"
mimulus_guttatus$layout[(mimulus_guttatus$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = mimulus_guttatus[,"Run"], replicate = mimulus_guttatus[,"BioSample"], layout = mimulus_guttatus[,"layout"], genome = "mimulus_guttatus", ploidy = 2))
```
## Olea europaea
```{r}
olea_europaea = fread("../workflow/data/SRArunInfo/SraRunInfo_olea_europaea_PRJEB35540.csv.gz", data.table = F)

# species has duplicate biosamples but I don't think they're actually technical replicates
length(unique(olea_europaea$BioSample))

# add layout
olea_europaea$layout = NA
olea_europaea$layout[(olea_europaea$spots_with_mates > 0)] = "PE"
olea_europaea$layout[(olea_europaea$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = olea_europaea[,"Run"], replicate = olea_europaea[,"BioSample"], layout = olea_europaea[,"layout"], genome = "olea_europaea", ploidy = 2))
```

## Oryza sativa
```{r}
oryza_sativa = fread("../workflow/data/SRArunInfo/SraRunInfo_oryza_sativa_PRJEB6180.csv.gz", data.table = F)

# how many genotypes are there?
length(unique(oryza_sativa$BioSample))

# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(biosample in unique(oryza_sativa$BioSample)){
#   techreps = oryza_sativa[(oryza_sativa$BioSample == biosample),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }

# oryza_sativa = oryza_sativa[(oryza_sativa$Run %in% runs),]

# add layout
oryza_sativa$layout = NA
oryza_sativa$layout = "PE" # even the samples without mates are actually paired-end

# add reads to growing list
allReads = rbind(allReads, data.frame(run = oryza_sativa[,"Run"], replicate = oryza_sativa[,"BioSample"], layout = oryza_sativa[,"layout"], genome = "oryza_sativa", ploidy = 2))
```


## Panicum halli
```{r}
# To get run numbers for panicum halli
# Copy bioprojects from table S1 in this publication: https://doi.org/10.1038/s41467-018-07669-x
# to a list in doc/panicum_halli_search.txt
# then use regular expressions to make a query to be used with ENTREZ
# cat panicum_halli_search.txt | tr '\n' ' ' | sed 's/ / OR /g'
# esearch -db sra -query $QUERY | efetch --format runinfo > panicum_halli.csv
panicum_halli = fread("../workflow/data/SRArunInfo/SraRunInfo_panicum_halli.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(panicum_halli$Experiment))
any(duplicated(panicum_halli$SampleName))
any(duplicated(panicum_halli$BioSample))

# runs = NULL
# for(biosample in unique(panicum_halli$BioSample)){
#   techreps = panicum_halli[(panicum_halli$BioSample == biosample),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }
# 
# panicum_halli = panicum_halli[(panicum_halli$Run %in% runs),]

# remove pacbio reads
panicum_halli = panicum_halli[(panicum_halli$Platform != "PACBIO_SMRT"),]

# add layout
# reads without mates are actually PE
panicum_halli$layout = NA
panicum_halli$layout = "PE"
#panicum_halli$layout[(panicum_halli$spots_with_mates > 0)] = "PE"
#panicum_halli$layout[(panicum_halli$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = panicum_halli[,"Run"], replicate = panicum_halli[,"BioSample"], layout = panicum_halli[,"layout"], genome = "panicum_halli", ploidy = 2))
```

## Panicum virgatum
```{r}
panicum_virgatum = fread("../workflow/data/SRArunInfo/SraRunInfo_panicum_virgatum_PRJNA622568.csv.gz", data.table = F)

# add layout
panicum_virgatum$layout = NA
panicum_virgatum$layout[(panicum_virgatum$spots_with_mates > 0)] = "PE"
panicum_virgatum$layout[(panicum_virgatum$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = panicum_virgatum[,"Run"], replicate = panicum_virgatum[,"BioSample"], layout = panicum_virgatum[,"layout"], genome = "panicum_virgatum", ploidy = 4))
```

## Phaseolus vulgaris
```{r}
phaseolus_vulgaris = fread("../workflow/data/SRArunInfo/SraRunInfo_phaseolus_vulgaris_PRJNA515107.csv.gz", data.table = F)

# subset relevant reads, need random library selection
phaseolus_vulgaris = phaseolus_vulgaris[(phaseolus_vulgaris$LibrarySelection == "RANDOM"),]

# check for technical replicates
any(duplicated(phaseolus_vulgaris$Experiment))
any(duplicated(phaseolus_vulgaris$SampleName))
any(duplicated(phaseolus_vulgaris$BioSample))

# add layout
phaseolus_vulgaris$layout = NA
phaseolus_vulgaris$layout[(phaseolus_vulgaris$spots_with_mates > 0)] = "PE"
phaseolus_vulgaris$layout[(phaseolus_vulgaris$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = phaseolus_vulgaris[,"Run"], replicate = phaseolus_vulgaris[,"BioSample"], layout = phaseolus_vulgaris[,"layout"], genome = "phaseolus_vulgaris", ploidy = 2))
```

## Populus trichocarpa
```{r}
populus_trichocarpa = fread("../workflow/data/SRArunInfo/SraRunInfo_populus_trichocarpa_PRJNA276056.csv.gz", data.table = F)

# subset relevant reads
populus_trichocarpa = populus_trichocarpa[(populus_trichocarpa$ScientificName == "Populus trichocarpa"),]

# add layout
populus_trichocarpa$layout = NA
populus_trichocarpa$layout = "PE"
#populus_trichocarpa$layout[(populus_trichocarpa$spots_with_mates > 0)] = "PE"
#populus_trichocarpa$layout[(populus_trichocarpa$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = populus_trichocarpa[,"Run"], replicate = populus_trichocarpa[,"BioSample"], layout = populus_trichocarpa[,"layout"], genome = "populus_trichocarpa", ploidy = 2))
```

## Prunus persica
```{r}
prunus_persica = fread("../workflow/data/SRArunInfo/SraRunInfo_prunus_persica_PRJNA388029.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(prunus_persica$Experiment))
any(duplicated(prunus_persica$SampleName))
any(duplicated(prunus_persica$BioSample))

# Some of these accessions are not Pruus persica, load table S1 from publication to filter
prunus_persica_supp = fread("../workflow/data/SRArunInfo/doi10.1186s13059-019-1648-9_prunus_persica_TableS1_2022-10-20.csv.gz", data.table = F)

ids = prunus_persica_supp$ID[(prunus_persica_supp$"Botanical varieties" == "P. persica L.")]

prunus_persica = prunus_persica[(prunus_persica$SampleName %in% ids),]

# add layout
prunus_persica$layout = NA
prunus_persica$layout[(prunus_persica$spots_with_mates > 0)] = "PE"
prunus_persica$layout[(prunus_persica$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = prunus_persica[,"Run"], replicate = prunus_persica[,"BioSample"], layout = prunus_persica[,"layout"], genome = "prunus_persica", ploidy = 2))
```

## Setaria italica
```{r}
setaria_italica = fread("../workflow/data/SRArunInfo/SraRunInfo_setaria_italica_PRJEB1234.csv.gz", data.table = F)

# remove Setaria virdis lines, though maybe I could keep them later?
setaria_italica = setaria_italica[(setaria_italica$ScientificName == "Setaria italica"),]

# check for technical replicates
any(duplicated(setaria_italica$Experiment))
any(duplicated(setaria_italica$SampleName))
any(duplicated(setaria_italica$BioSample))

# add layout
setaria_italica$layout = NA
setaria_italica$layout[(setaria_italica$spots_with_mates > 0)] = "PE"
setaria_italica$layout[(setaria_italica$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = setaria_italica[,"Run"], replicate = setaria_italica[,"BioSample"], layout = setaria_italica[,"layout"], genome = "setaria_italica", ploidy = 2))
```

## Setaria viridis
```{r}
setaria_viridis = fread("../workflow/data/SRArunInfo/SraRunInfo_setaria_viridis_PRJNA633413.csv.gz", data.table = F)

setaria_viridis = setaria_viridis[(setaria_viridis$ScientificName == "Setaria viridis"),]

# add layout
setaria_viridis$layout = NA
setaria_viridis$layout[(setaria_viridis$spots_with_mates > 0)] = "PE"
setaria_viridis$layout[(setaria_viridis$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = setaria_viridis[,"Run"], replicate = setaria_viridis[,"BioSample"], layout = setaria_viridis[,"layout"], genome = "setaria_viridis", ploidy = 2))
```

## Solanum lycopersicum
```{r}
solanum_lycopersicum = fread("../workflow/data/SRArunInfo/SraRunInfo_solanum_lycopersicum_PRJNA259308.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(solanum_lycopersicum$Experiment))
any(duplicated(solanum_lycopersicum$SampleName))
any(duplicated(solanum_lycopersicum$BioSample))

# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(biosample in unique(solanum_lycopersicum$BioSample)){
#   techreps = solanum_lycopersicum[(solanum_lycopersicum$BioSample == biosample),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }
# 
# solanum_lycopersicum = solanum_lycopersicum[(solanum_lycopersicum$Run %in% runs),]

# add layout
solanum_lycopersicum$layout = NA
solanum_lycopersicum$layout[(solanum_lycopersicum$spots_with_mates > 0)] = "PE"
solanum_lycopersicum$layout[(solanum_lycopersicum$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = solanum_lycopersicum[,"Run"], replicate = solanum_lycopersicum[,"BioSample"], layout = solanum_lycopersicum[,"layout"], genome = "solanum_lycopersicum", ploidy = 2))
```
## Sorghum bicolor
```{r}
sorghum_bicolor = fread("../workflow/data/SRArunInfo/SraRunInfo_sorghum_bicolor_PRJEB50066.csv.gz", data.table = F)

# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(biosample in unique(sorghum_bicolor$BioSample)){
#   techreps = sorghum_bicolor[(sorghum_bicolor$BioSample == biosample),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }
# 
# sorghum_bicolor = sorghum_bicolor[(sorghum_bicolor$Run %in% runs),]

# add layout
# reads without mates are still PE
sorghum_bicolor$layout = NA
sorghum_bicolor$layout = "PE"
#sorghum_bicolor$layout[(sorghum_bicolor$spots_with_mates > 0)] = "PE"
#sorghum_bicolor$layout[(sorghum_bicolor$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = sorghum_bicolor[,"Run"], replicate = sorghum_bicolor[,"BioSample"], layout = sorghum_bicolor[,"layout"], genome = "sorghum_bicolor", ploidy = 2))
```

## Spinacia oleracea
```{r}
spinacia_oleracea = fread("../workflow/data/SRArunInfo/SraRunInfo_spinacia_oleracea_PRJNA598728.csv.gz", data.table = F)

spinacia_oleracea = spinacia_oleracea[(spinacia_oleracea$ScientificName == "Spinacia oleracea"),]

# check for technical replicates
any(duplicated(spinacia_oleracea$Experiment))
any(duplicated(spinacia_oleracea$SampleName))
any(duplicated(spinacia_oleracea$BioSample))

# add layout
# checked that reads without mates are PE
spinacia_oleracea$layout = NA
spinacia_oleracea$layout = "PE"
#spinacia_oleracea$layout[(spinacia_oleracea$spots_with_mates > 0)] = "PE"
#spinacia_oleracea$layout[(spinacia_oleracea$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = spinacia_oleracea[,"Run"], replicate = spinacia_oleracea[,"BioSample"], layout = spinacia_oleracea[,"layout"], genome = "spinacia_oleracea", ploidy = 2))
```

## Spirodela polyrhiza
```{r}
spirodela_polyrhiza = fread("../workflow/data/SRArunInfo/SraRunInfo_spirodela_polyrhiza_PRJNA552613.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(spirodela_polyrhiza$Experiment))
any(duplicated(spirodela_polyrhiza$SampleName))
any(duplicated(spirodela_polyrhiza$BioSample))

# add layout
# reads without mates are SE
spirodela_polyrhiza$layout = NA
spirodela_polyrhiza$layout[(spirodela_polyrhiza$spots_with_mates > 0)] = "PE"
spirodela_polyrhiza$layout[(spirodela_polyrhiza$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = spirodela_polyrhiza[,"Run"], replicate = spirodela_polyrhiza[,"BioSample"], layout = spirodela_polyrhiza[,"layout"], genome = "spirodela_polyrhiza", ploidy = 2))
```

## Triticum aestivum
```{r}
# load data
triticum_aestivum_1 = fread("../workflow/data/SRArunInfo/SraRunInfo_triticum_aestivum_PRJEB48738.csv.gz", data.table = F)
triticum_aestivum_2 = fread("../workflow/data/SRArunInfo/SraRunInfo_triticum_aestivum_PRJEB48988.csv.gz", data.table = F)

# concatenate
triticum_aestivum = rbind(triticum_aestivum_1, triticum_aestivum_2)

# check for technical replicates
any(duplicated(triticum_aestivum$Experiment))
any(duplicated(triticum_aestivum$SampleName))
any(duplicated(triticum_aestivum$BioSample))

# add layout
# checked that reads without mates are PE
triticum_aestivum$layout = NA
triticum_aestivum$layout = "PE"
#triticum_aestivum$layout[(triticum_aestivum$spots_with_mates > 0)] = "PE"
#triticum_aestivum$layout[(triticum_aestivum$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = triticum_aestivum[,"Run"], replicate = triticum_aestivum[,"BioSample"], layout = triticum_aestivum[,"layout"], genome = "triticum_aestivum", ploidy = 6))

```

## Vitis vinifera
```{r}
vitis_vinifera = fread("../workflow/data/SRArunInfo/SraRunInfo_vitis_vinifera_PRJNA373967.csv.gz", data.table = F)

# subset WGS reads
vitis_vinifera = vitis_vinifera[(vitis_vinifera$LibraryStrategy == "WGS" & vitis_vinifera$ScientificName == "Vitis vinifera"),]

# check for technical replicates
any(duplicated(vitis_vinifera$Experiment))
any(duplicated(vitis_vinifera$SampleName))
any(duplicated(vitis_vinifera$BioSample))

# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(biosample in unique(vitis_vinifera$BioSample)){
#   techreps = vitis_vinifera[(vitis_vinifera$BioSample == biosample),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }
# 
# vitis_vinifera = vitis_vinifera[(vitis_vinifera$Run %in% runs),]

# add layout
# reads without mates are SE
vitis_vinifera$layout = NA
vitis_vinifera$layout[(vitis_vinifera$spots_with_mates > 0)] = "PE"
vitis_vinifera$layout[(vitis_vinifera$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = vitis_vinifera[,"Run"], replicate = vitis_vinifera[,"BioSample"], layout = vitis_vinifera[,"layout"], genome = "vitis_vinifera", ploidy = 2))
```

## Zea mays
```{r}
# load data
zea_mays_1 = fread("../workflow/data/SRArunInfo/SraRunInfo_zea_mays_PRJNA260788.csv.gz", data.table = F)
zea_mays_2 = fread("../workflow/data/SRArunInfo/SraRunInfo_zea_mays_PRJNA389800.csv.gz", data.table = F)
zea_mays_3 = fread("../workflow/data/SRArunInfo/SraRunInfo_zea_mays_PRJNA399729.csv.gz", data.table = F)

# concatenate
zea_mays = rbind(zea_mays_1, zea_mays_2, zea_mays_3)

# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(biosample in unique(zea_mays$BioSample)){
#   techreps = zea_mays[(zea_mays$BioSample == biosample),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }
# 
# zea_mays = zea_mays[(zea_mays$Run %in% runs),]

# confirm no duplicates
any(duplicated(zea_mays$Experiment))
any(duplicated(zea_mays$SampleName))
any(duplicated(zea_mays$BioSample))

# add layout
# reads without mates are SE
zea_mays$layout = NA
zea_mays$layout[(zea_mays$spots_with_mates > 0)] = "PE"
zea_mays$layout[(zea_mays$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = zea_mays[,"Run"], replicate = zea_mays[,"BioSample"], layout = zea_mays[,"layout"], genome = "zea_mays", ploidy = 2))
```

# Summarize read counts
```{r}
# breakdown of reads per species
table(allReads$genome, allReads$layout)

# number of species included thus far
length(unique(allReads$genome))

# number of unique genotypes
length(unique(allReads$replicate))

# number of unique genotypes per species
aggregate(replicate ~ genome, data = allReads, FUN = unique)

# total number of read sets
nrow(allReads)
```


# Loop over replicates, if there are both SE and PE reads for a single sample, just use the PE reads
```{r}
# I could either throw out read 2 of the PE reads and treat all data as single end, or I could retain just paired-end data... Not sure which is better
# I'll throw out the single-end reads because the paired-end data will align better
allReadsFinal = NULL

for(replicate in unique(allReads$replicate)){
  dataSubset = allReads[(allReads$replicate == replicate),]
  if(all(c("PE", "SE") %in% dataSubset[,"layout"])){
    dataSubset = dataSubset[(dataSubset$layout == "PE"),]
  }
  allReadsFinal = rbind(allReadsFinal, dataSubset)
}

# breakdown of PE vs SE for each genome
table(allReadsFinal$genome, allReadsFinal$layout)

# number of unique genotypes included
length(unique(allReadsFinal$replicate))

# number of read sets
nrow(allReadsFinal)

# number of genomes
length(unique(allReadsFinal$genome))
```

# output read metadata for snakemake input
```{r}
# test dataset for merging reads into single datasets
mergeTestSet = allReads[(allReads$replicate %in% c("SAMEA2567517", "SAMEA2567535", "SAMEA2567633", "SAMEA2567634", "SAMN03005107", "SAMN03005132")),]

write.table(mergeTestSet, file = "samples.tsv", sep = "\t", quote = F, row.names = F)

# write run numbers and metadata to workflow input
# need to replace write path with "../workflow/data/samples_pe.tsv" when I'm ready
#write.table(seReads, file = "samples_se.tsv", sep = "\t", quote = F, row.names = F)
#write.table(allReads, file = "samples_pe.tsv", sep = "\t", quote = F, row.names = F)

```
















# Manually check top few bioprojects for each species to see if they're WGS data
```{r}

```

# Split SRA run data into single-end and paired end
## Arabidopsis thaliana
```{r}
athalSraRunInfo = fread("../workflow/data/SraRunInfo_arabidopsis_thaliana.csv.gz", data.table = F)

# separate paired end and single-end data
# if spot_with_mates == 0 in sra run info, then data is single-end
athalSe = athalSraRunInfo[(athalSraRunInfo$spots_with_mates == 0),"Run"]
athalPe = athalSraRunInfo[(athalSraRunInfo$spots_with_mates != 0), "Run"]

# add assembly name to file
# can subset data here to get toy dataset
athalSe = data.frame(samples = athalSe, genome = "arabidopsis_thaliana")
athalPe = data.frame(samples = athalPe, genome = "arabidopsis_thaliana")

# write single-end and paired-end samples to separate files
write.table(athalSe, file = "../workflow/data/samples_se.tsv", sep = "\t", quote = F, row.names = F)
write.table(athalPe, file = "../workflow/data/samples_pe.tsv", sep = "\t", quote = F, row.names = F)
```