# Purpose: organize sra data from multiple species for input into snakemake workflow

# clear wd and load packages
```{r}
rm(list = ls())

library(data.table)

# if (!require("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("SRAdb")

# speciesList = unique(read.csv("../workflow/data/speciesList.csv")$Species)

# create empty objects for growing list of reads
allReads = NULL
```

# Compile SRA run info for each species into master list, separate paired end and single-end reads

## Actinidia chinensis
```{r}
# load data for species
actinidia_chinensis = fread("../workflow/data/SRArunInfo/SraRunInfo_actinidia_chinensis_PRJNA302330.csv.gz", data.table = F)

# assign reads to replicates based on info in publication
# KFA and KFB are clones
actinidia_chinensis$replicate = NA
actinidia_chinensis$replicate[grepl("KFA|KFB", actinidia_chinensis$SampleName)] = "AB"
actinidia_chinensis$replicate[grepl("KFC", actinidia_chinensis$SampleName)] = "C"
actinidia_chinensis$replicate[grepl("KFD", actinidia_chinensis$SampleName)] = "D"
actinidia_chinensis$replicate[grepl("KFE", actinidia_chinensis$SampleName)] = "E"
actinidia_chinensis$replicate[grepl("KFF", actinidia_chinensis$SampleName)] = "F"

# add reads to growing list
# not sure about this ploidy
allReads = rbind(allReads, data.frame(run = actinidia_chinensis[,"Run"], replicate = actinidia_chinensis[,"replicate"], layout = "SE", genome = "actinidia_chinensis", ploidy = 2, taxonKeys = "3644693 7270743 3644698 3644711"))
```

## Aegilops tauschii
```{r}
# load data for species
aegilops_tauschii = fread("../workflow/data/SRArunInfo/SraRunInfo_aegilops_tauschii_PRJNA685125.csv.gz", data.table = F)

# subset to only appropriate reads
aegilops_tauschii = aegilops_tauschii[(aegilops_tauschii$ScientificName == "Aegilops tauschii"),]

# Check if there are any technical replicates
any(duplicated(aegilops_tauschii$Experiment))
any(duplicated(aegilops_tauschii$BioSample))

# check if any reads are single-end
any(aegilops_tauschii$spots_with_mates == 0)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = aegilops_tauschii[,"Run"], replicate = aegilops_tauschii[,"BioSample"], layout = "PE", genome = "aegilops_tauschii", ploidy = 2, taxonKeys = "5289779 5677746 5677750 7226877 5677755 5677762 5677759 5677753"))
```

## Amaranthus hypochondriacus
```{r}
# load data for species
amaranthus_hypochondriacus = fread("../workflow/data/SRArunInfo/SraRunInfo_amaranthus_hypochondriacus_PRJEB30531.csv.gz", data.table = F)

# subset to only appropriate reads
amaranthus_hypochondriacus = amaranthus_hypochondriacus[(amaranthus_hypochondriacus$ScientificName == "Amaranthus hypochondriacus"),]

# Check if there are any technical replicates
any(duplicated(amaranthus_hypochondriacus$Experiment))
any(duplicated(amaranthus_hypochondriacus$BioSample))

# check if any reads are single-end
any(amaranthus_hypochondriacus$spots_with_mates == 0)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = amaranthus_hypochondriacus[,"Run"], replicate = amaranthus_hypochondriacus[,"BioSample"], layout = "PE", genome = "amaranthus_hypochondriacus", ploidy = 2, taxonKeys = "5384364 7554918 8111324 7442708 10145116 10538632 10634896 10279930 5547885 7867601"))
```

## Ananas comosus
```{r}
# load data for species
ananas_comosus = fread("../workflow/data/SRArunInfo/SraRunInfo_ananas_comosus_PRJNA389669.csv.gz", data.table = F)

# remove outgroups
ananas_comosus = ananas_comosus[grepl("Ananas",ananas_comosus$ScientificName),]

# Check if there are any technical replicates
any(duplicated(ananas_comosus$Experiment))
any(duplicated(ananas_comosus$SampleName))

# check if any reads are single-end
any(ananas_comosus$spots_with_mates == 0)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = ananas_comosus[,"Run"], replicate = ananas_comosus[,"BioSample"], layout = "PE", genome = "ananas_comosus", ploidy = 2, taxonKeys = "5288770"))
```

## Arabidopsis halleri
```{r}
# load data for species
arabidopsis_halleri = fread("../workflow/data/SRArunInfo/SraRunInfo_arabidopsis_halleri_PRJDB3777.csv.gz", data.table = F)

# just use hiseq2000 reads so that you don't have to merge technical reps
arabidopsis_halleri = arabidopsis_halleri[(arabidopsis_halleri$Model == "Illumina HiSeq 2000"),]

# check if any reads are single-end
any(arabidopsis_halleri$spots_with_mates == 0)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = arabidopsis_halleri[,"Run"], replicate = arabidopsis_halleri[,"BioSample"], layout = "PE", genome = "arabidopsis_halleri", ploidy = 2, taxonKeys = "3052450 9403166 3052459 11097917 3689665 7225571 10906894 3052455 10999708 10940826"))
```

## Arabidopsis lyrata
```{r}
arabidopsis_lyrata_1 = fread("../workflow/data/SRArunInfo/SraRunInfo_arabidopsis_lyrata_PRJEB23202.csv.gz", data.table = F)
arabidopsis_lyrata_2 = fread("../workflow/data/SRArunInfo/SraRunInfo_arabidopsis_lyrata_PRJEB30473.csv.gz", data.table = F)
arabidopsis_lyrata_3 = fread("../workflow/data/SRArunInfo/SraRunInfo_arabidopsis_lyrata_PRJNA284572.csv.gz", data.table = F)

# subset only A. lyrata
arabidopsis_lyrata_1 = arabidopsis_lyrata_1[grepl("Arabidopsis lyrata", arabidopsis_lyrata_1$ScientificName),]
arabidopsis_lyrata_2 = arabidopsis_lyrata_2[grepl("Arabidopsis lyrata", arabidopsis_lyrata_2$ScientificName),]
arabidopsis_lyrata_3 = arabidopsis_lyrata_3[grepl("Arabidopsis lyrata", arabidopsis_lyrata_3$ScientificName),]

# concatenate datasets
arabidopsis_lyrata = rbind(arabidopsis_lyrata_1, arabidopsis_lyrata_2, arabidopsis_lyrata_3)

# check if any reads are single-end
any(arabidopsis_lyrata$spots_with_mates == 0)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = arabidopsis_lyrata[,"Run"], replicate = arabidopsis_lyrata[,"BioSample"], layout = "PE", genome = "arabidopsis_lyrata", ploidy = 2, taxonKeys = "3052468 3052469 3052478 7225570 9372560"))
```


## Arabidopsis suecica
```{r}
# load data for species
arabidopsis_suecica = fread("../workflow/data/SRArunInfo/SraRunInfo_arabidopsis_suecica_PRJNA309929.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = arabidopsis_suecica[,"Run"], replicate = arabidopsis_suecica[,"BioSample"], layout = "PE", genome = "arabidopsis_suecica", ploidy = 4, taxonKeys = "7915981 3052544"))

```

## Arabidopsis thaliana
```{r}
# load data for species
arabidopsis_thaliana = fread("../workflow/data/SRArunInfo/SraRunInfo_arabidopsis_thaliana_PRJNA273563.csv.gz", data.table = F)

# assign reads as single end or paired end
arabidopsis_thaliana$layout = NA
arabidopsis_thaliana$layout[(arabidopsis_thaliana$spots_with_mates > 0)] = "PE"
arabidopsis_thaliana$layout[(arabidopsis_thaliana$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = arabidopsis_thaliana[,"Run"], replicate = arabidopsis_thaliana[,"BioSample"], layout = arabidopsis_thaliana[,"layout"], genome = "arabidopsis_thaliana", ploidy = 2, taxonKeys = "3052436 7758053 3052446"))
```

## Arabis alpina
```{r}
# load data for species
arabis_alpina = fread("../workflow/data/SRArunInfo/SraRunInfo_arabis_alpina_PRJNA489364.csv.gz", data.table = F)

# check species
unique(arabis_alpina$ScientificName)

# check if any reads are single-end
any(arabis_alpina$spots_with_mates == 0)

# assign reads as single end or paired end
arabis_alpina$layout = NA
arabis_alpina$layout = "PE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = arabis_alpina[,"Run"], replicate = arabis_alpina[,"BioSample"], layout = arabis_alpina[,"layout"], genome = "arabis_alpina", ploidy = 2))
```


## Arabis nemorensis
```{r}
# load data for species
arabis_nemorensis = fread("../workflow/data/SRArunInfo/SraRunInfo_arabis_nemorensis_PRJEB39992.csv.gz", data.table = F)

# get just WGS reads
arabis_nemorensis = arabis_nemorensis[(arabis_nemorensis$LibraryStrategy == "WGS" & arabis_nemorensis$ScientificName == "Arabis nemorensis"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = arabis_nemorensis[,"Run"], replicate = arabis_nemorensis[,"BioSample"], layout = "SE", genome = "arabis_nemorensis", ploidy = 2))
```

## Arachis hypogaea and Arachis duranensis
```{r}
# load data for species
arachis_1 = fread("../workflow/data/SRArunInfo/SraRunInfo_arachis_PRJNA490832.csv.gz", data.table = F)
arachis_2 = fread("../workflow/data/SRArunInfo/SraRunInfo_arachis_PRJNA511155.csv.gz", data.table = F)
arachis_3 = fread("../workflow/data/SRArunInfo/SraRunInfo_arachis_PRJNA525866.csv.gz", data.table = F)

# split reads by species
arachis = rbind(arachis_1, arachis_2, arachis_3)

arachis_hypogaea = arachis[grepl("Arachis hypogaea", arachis$ScientificName),]

arachis_duranensis = arachis[(arachis$ScientificName == "Arachis duranensis"),]

arachis_ipaensis = arachis[(arachis$ScientificName == "Arachis ipaensis"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = arachis_hypogaea[,"Run"], replicate = arachis_hypogaea[,"BioSample"], layout = "PE", genome = "arachis_hypogaea", ploidy = 4))

allReads = rbind(allReads, data.frame(run = arachis_duranensis[,"Run"], replicate = arachis_duranensis[,"BioSample"], layout = "PE", genome = "arachis_duranensis", ploidy = 2))

allReads = rbind(allReads, data.frame(run = arachis_ipaensis[,"Run"], replicate = arachis_ipaensis[,"BioSample"], layout = "PE", genome = "arachis_ipaensis", ploidy = 2))
```

## Benincasa hispida
```{r}
# load data for species
benincasa_hispida = fread("../workflow/data/SRArunInfo/SraRunInfo_benincasa_hispida_PRJNA576075.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = benincasa_hispida[,"Run"], replicate = benincasa_hispida[,"BioSample"], layout = "PE", genome = "benincasa_hispida", ploidy = 2))
```

## Beta vulgaris
```{r}
# load data for species
beta_vulgaris = fread("../workflow/data/SRArunInfo/SraRunInfo_beta_vulgaris_PRJNA815240.csv.gz", data.table = F)

# remove close relatives
beta_vulgaris = beta_vulgaris[grepl("Beta vulgaris",beta_vulgaris$ScientificName),]

# check if any reads are single-end
# double-checked that spots without mates are actually PE
any(beta_vulgaris$spots_with_mates == 0)

beta_vulgaris$layout = NA
beta_vulgaris$layout = "PE"
# beta_vulgaris$layout[(beta_vulgaris$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = beta_vulgaris[,"Run"], replicate = beta_vulgaris[,"BioSample"], layout = beta_vulgaris[,"layout"], genome = "beta_vulgaris", ploidy = 2))

```

## Boechera stricta
```{r}
boechera_stricta_1 = fread("../workflow/data/SRArunInfo/SraRunInfo_boechera_stricta_1_2022-12-01.csv.gz", data.table = F)
boechera_stricta_2 = fread("../workflow/data/SRArunInfo/SraRunInfo_boechera_stricta_2_2022-12-01.csv.gz", data.table = F)

# subset second dataset by values in this publication
# https://doi.org/10.1093/gbe/evac043
# SRP134283-SRP134373, SRP134393-SRP134433, SRP134436-SRP134479, SRP134481-SRP134572, and SRP134581-SRP134671
sraprojects = paste("SRP", c(134283:134373, 134393:134433, 134436:134479, 134481:134572, 134581:134671), sep = "")

boechera_stricta_2 = boechera_stricta_2[(boechera_stricta_2$SRAStudy %in% sraprojects),]

# bind both datasets
boechera_stricta = rbind(boechera_stricta_1, boechera_stricta_2)

# add layout
boechera_stricta$layout = NA
boechera_stricta$layout[(boechera_stricta$spots_with_mates > 0)] = "PE"
boechera_stricta$layout[(boechera_stricta$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = boechera_stricta[,"Run"], replicate = boechera_stricta[,"BioSample"], layout = boechera_stricta[,"layout"], genome = "boechera_stricta", ploidy = 2))

# remove technical replicates from dataset 2
# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(LibraryName in unique(boechera_stricta_2$LibraryName)){
#   techreps = boechera_stricta_2[(boechera_stricta_2$LibraryName == LibraryName),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }
# 
# boechera_stricta_2 = boechera_stricta_2[(boechera_stricta_2$Run %in% runs),]

# add reads to growing list
# allReads = rbind(allReads, data.frame(samples = boechera_stricta_1[(boechera_stricta_1$spots_with_mates > 0),"Run"], genome = "boechera_stricta"))
# 
# allReads = rbind(allReads, data.frame(samples = boechera_stricta_2[(boechera_stricta_2$spots_with_mates > 0),"Run"], genome = "boechera_stricta"))

```

## Brachypodium distachyon
```{r}
# load data for species
brachypodium_distachyon = fread("../workflow/data/SRArunInfo/SraRunInfo_brachypodium_distachyon_PRJNA605320.csv.gz", data.table = F)

# isolate only WGS datasets
brachypodium_distachyon = brachypodium_distachyon[(brachypodium_distachyon$LibraryStrategy == "WGS"),]

# check for technical replicates
any(duplicated(brachypodium_distachyon$Experiment))
any(duplicated(brachypodium_distachyon$SampleName))
any(duplicated(brachypodium_distachyon$BioSample))
any(duplicated(brachypodium_distachyon$LibraryName))

# add layout
all(brachypodium_distachyon$spots_with_mates > 0)
brachypodium_distachyon$layout = NA
brachypodium_distachyon$layout[(brachypodium_distachyon$spots_with_mates > 0)] = "PE"
brachypodium_distachyon$layout[(brachypodium_distachyon$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = brachypodium_distachyon[,"Run"], replicate = brachypodium_distachyon[,"BioSample"], layout = brachypodium_distachyon[,"layout"], genome = "brachypodium_distachyon", ploidy = 2))

```

## Brassica juncea
```{r}
# load data for species
brassica_juncea = fread("../workflow/data/SRArunInfo/SraRunInfo_brassica_juncea_PRJNA615316.csv.gz", data.table = F)

# isolate only WGS datasets
brassica_juncea = brassica_juncea[(brassica_juncea$LibraryStrategy == "WGS" & brassica_juncea$Platform == "ILLUMINA" & brassica_juncea$ScientificName == "Brassica juncea"),]

# check for technical replicates
any(duplicated(brassica_juncea$Experiment))
any(duplicated(brassica_juncea$SampleName))
any(duplicated(brassica_juncea$BioSample))
any(duplicated(brassica_juncea$LibraryName))

# add layout
all(brassica_juncea$spots_with_mates > 0)
brassica_juncea$layout = "PE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = brassica_juncea[,"Run"], replicate = brassica_juncea[,"BioSample"], layout = brassica_juncea[,"layout"], genome = "brassica_juncea", ploidy = 4))

```

## Brassica napus
```{r}
# load data for species
brassica_napus = fread("../workflow/data/SRArunInfo/SraRunInfo_brassica_napus_PRJNA476657.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(brassica_napus$Experiment))
any(duplicated(brassica_napus$SampleName))
any(duplicated(brassica_napus$BioSample))
any(duplicated(brassica_napus$LibraryName))

# add layout
all(brassica_napus$spots_with_mates > 0)

# add reads to growing list
# use experiment ID instead of BioSample ID because BioSample ID is not unique for each sample
allReads = rbind(allReads, data.frame(run = brassica_napus[,"Run"], replicate = brassica_napus[,"Experiment"], layout = "PE", genome = "brassica_napus", ploidy = 4))

```

## Brassica nigra
```{r}
# load data for species
brassica_nigra = fread("../workflow/data/SRArunInfo/SraRunInfo_brassica_juncea_PRJNA615316.csv.gz", data.table = F)

# isolate only WGS datasets
brassica_nigra = brassica_nigra[(brassica_nigra$LibraryStrategy == "WGS" & brassica_nigra$Platform == "ILLUMINA" & brassica_nigra$ScientificName == "Brassica nigra"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = brassica_nigra[,"Run"], replicate = brassica_nigra[,"BioSample"], layout = "PE", genome = "brassica_nigra", ploidy = 2))
```

## Brassica rapa and Brassica oleraceae
```{r}
brassica = fread("../workflow/data/SRArunInfo/SraRunInfo_PRJNA312457.csv.gz", data.table = F)

# split data into the two species
brassica_oleracea = brassica[(brassica$ScientificName == "Brassica oleracea"),]
brassica_rapa = brassica[(brassica$ScientificName == "Brassica rapa"),]

# check if any reads are single-end
any(brassica_rapa$spots_with_mates == 0)
any(brassica_oleracea$spots_with_mates == 0)

# add layout
brassica_rapa$layout = NA
brassica_rapa$layout[(brassica_rapa$spots_with_mates > 0)] = "PE"
brassica_rapa$layout[(brassica_rapa$spots_with_mates == 0)] = "SE"

brassica_oleracea$layout = NA
brassica_oleracea$layout[(brassica_oleracea$spots_with_mates > 0)] = "PE"
brassica_oleracea$layout[(brassica_oleracea$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = brassica_rapa[,"Run"], replicate = brassica_rapa[,"BioSample"], layout = brassica_rapa[,"layout"], genome = "brassica_rapa", ploidy = 2))

allReads = rbind(allReads, data.frame(run = brassica_oleracea[,"Run"], replicate = brassica_oleracea[,"BioSample"], layout = brassica_oleracea[,"layout"], genome = "brassica_oleracea", ploidy = 2))
```

## Buddleja alternifolia
```{r}
# load data for species
buddleja_alternifolia = fread("../workflow/data/SRArunInfo/SraRunInfo_buddleja_alternifolia_PRJNA577174.csv.gz", data.table = F)

# only include whole genome resequencing samples, not data used for genome assembly
buddleja_alternifolia = buddleja_alternifolia[!(buddleja_alternifolia$LibraryName %in% c("wgs-pb", "hic")),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = buddleja_alternifolia[,"Run"], replicate = buddleja_alternifolia[,"BioSample"], layout = "PE", genome = "buddleja_alternifolia", ploidy = 2))
```

## Cajanus cajan
```{r}
# load data for species
cajanus_cajan = fread("../workflow/data/SRArunInfo/SraRunInfo_cajanus_cajan_PRJNA383013.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = cajanus_cajan[,"Run"], replicate = cajanus_cajan[,"BioSample"], layout = "PE", genome = "cajanus_cajan", ploidy = 2))
```

## Camelina sativa
```{r}
# load data for species
camelina_sativa = fread("../workflow/data/SRArunInfo/SraRunInfo_camelina_sativa_PRJNA477644.csv.gz", data.table = F)

# subset just C. sativa
unique(camelina_sativa$ScientificName)

camelina_sativa = camelina_sativa[(camelina_sativa$ScientificName == "Camelina sativa"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = camelina_sativa[,"Run"], replicate = camelina_sativa[,"BioSample"], layout = "PE", genome = "camelina_sativa", ploidy = 6))

```

## Camellia sinensis
```{r}
# load data for species
camellia_sinensis = fread("../workflow/data/SRArunInfo/SraRunInfo_camellia_sinensis_PRJNA646044.csv.gz", data.table = F)

# subset reads for this species only
camellia_sinensis = camellia_sinensis[grepl("Camellia sinensis",camellia_sinensis$ScientificName),]

# assign reads as single end or paired end
camellia_sinensis$layout = NA
camellia_sinensis$layout[(camellia_sinensis$spots_with_mates > 0)] = "PE"
camellia_sinensis$layout[(camellia_sinensis$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = camellia_sinensis[,"Run"], replicate = camellia_sinensis[,"BioSample"], layout = camellia_sinensis[,"layout"], genome = "camellia_sinensis", ploidy = 2))
```

## Cannabis sativa
```{r}
cannabis_sativa = fread("../workflow/data/SRArunInfo/SraRunInfo_cannabis_sativa_PRJNA866500.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = cannabis_sativa[,"Run"], replicate = cannabis_sativa[,"BioSample"], layout = "PE", genome = "cannabis_sativa", ploidy = 2))
```

## Capsella grandiflora
```{r}
capsella_grandiflora = fread("../workflow/data/SRArunInfo/SraRunInfo_capsella_grandiflora_PRJEB39830.csv.gz", data.table = F)

# subset only WGS reads
capsella_grandiflora = capsella_grandiflora[(capsella_grandiflora$LibraryStrategy == "WGS"),]

# check if any reads are single-end
any(capsella_grandiflora$spots_with_mates == 0)

# add layout
capsella_grandiflora$layout = NA
capsella_grandiflora$layout[(capsella_grandiflora$spots_with_mates > 0)] = "PE"
capsella_grandiflora$layout[(capsella_grandiflora$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = capsella_grandiflora[,"Run"], replicate = capsella_grandiflora[,"BioSample"], layout = capsella_grandiflora[,"layout"], genome = "capsella_grandiflora", ploidy = 2))
```

## Capsella rubella
```{r}
# load data
capsella_rubella = fread("../workflow/data/SRArunInfo/SraRunInfo_capsella_rubella_PRJEB6689.csv.gz", data.table = F)

# subset only this relevant species
capsella_rubella = capsella_rubella[(capsella_rubella$ScientificName == "Capsella rubella"),]

# check for technical replicates
any(duplicated(capsella_rubella$Experiment))
any(duplicated(capsella_rubella$SampleName))
any(duplicated(capsella_rubella$BioSample))

# check if any reads are single-end
any(capsella_rubella$spots_with_mates == 0)

# add layout
# checked that spots without mates are PE
capsella_rubella$layout = NA
capsella_rubella$layout = "PE"
#capsella_rubella$layout[(capsella_rubella$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = capsella_rubella[,"Run"], replicate = capsella_rubella[,"BioSample"], layout = capsella_rubella[,"layout"], genome = "capsella_rubella", ploidy = 2))
```

## Capsicum annuum
```{r}
capsicum_annuum = fread("../workflow/data/SRArunInfo/SraRunInfo_capsicum_annuum_PRJEB45375.csv.gz", data.table = F)

# subset just C. annuum
unique(capsicum_annuum$ScientificName)

capsicum_annuum = capsicum_annuum[grepl("Capsicum annuum", capsicum_annuum$ScientificName),]

length(unique(capsicum_annuum$BioSample))

# add reads to growing list
allReads = rbind(allReads, data.frame(run = capsicum_annuum[,"Run"], replicate = capsicum_annuum[,"BioSample"], layout = "SE", genome = "capsicum_annuum", ploidy = 2))

# WGS data
# load data
# capsicum_annuum = fread("../workflow/data/SRArunInfo/SraRunInfo_capsicum_annuum_PRJDB14137.csv.gz", data.table = F)

# Subset only WGS reads for this species
#capsicum_annuum = capsicum_annuum[(capsicum_annuum$LibraryStrategy == "WGS" & capsicum_annuum$Platform == "ILLUMINA" & capsicum_annuum$ScientificName == "Capsicum annuum"),]

# add reads to growing list
# allReads = rbind(allReads, data.frame(run = capsicum_annuum[,"Run"], replicate = capsicum_annuum[,"BioSample"], layout = "PE", genome = "capsicum_annuum", ploidy = 2))
```

## Castanea mollissima
```{r}
# load data
castanea_mollissima = fread("../workflow/data/SRArunInfo/SraRunInfo_castanea_mollissima_PRJNA540917.csv.gz", data.table = F)

# remove reads used for genome assembly
castanea_mollissima = castanea_mollissima[!(castanea_mollissima$Run %in% c("SRR10672059", "SRR10672058")),]

# subset to just one species
castanea_mollissima = castanea_mollissima[(castanea_mollissima$ScientificName == "Castanea mollissima"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = castanea_mollissima[,"Run"], replicate = castanea_mollissima[,"BioSample"], layout = "SE", genome = "castanea_mollissima", ploidy = 2))
```

## Chenopodium quinoa
```{r}
# load data
chenopodium_quinoa = fread("../workflow/data/SRArunInfo/SraRunInfo_chenopodium_quinoa_PRJNA673789.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(chenopodium_quinoa$Experiment))
any(duplicated(chenopodium_quinoa$SampleName))
any(duplicated(chenopodium_quinoa$BioSample))

# check if any reads are single-end
any(chenopodium_quinoa$spots_with_mates == 0)

# add layout
chenopodium_quinoa$layout = NA
chenopodium_quinoa$layout[(chenopodium_quinoa$spots_with_mates > 0)] = "PE"
chenopodium_quinoa$layout[(chenopodium_quinoa$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = chenopodium_quinoa[,"Run"], replicate = chenopodium_quinoa[,"BioSample"], layout = chenopodium_quinoa[,"layout"], genome = "chenopodium_quinoa", ploidy = 4))
```

## Cicer arietinum
```{r}
# load data
cicer_arietinum = fread("../workflow/data/SRArunInfo/SraRunInfo_cicer_arietinum_PRJNA657888.csv.gz", data.table = F)

# subset only this relevant species
cicer_arietinum = cicer_arietinum[(cicer_arietinum$ScientificName == "Cicer arietinum"),]

# check for technical replicates
any(duplicated(cicer_arietinum$Experiment))
any(duplicated(cicer_arietinum$SampleName))
any(duplicated(cicer_arietinum$BioSample))

# check if any reads are single-end
any(cicer_arietinum$spots_with_mates == 0)

# add layout
cicer_arietinum$layout = NA
cicer_arietinum$layout[(cicer_arietinum$spots_with_mates > 0)] = "PE"
cicer_arietinum$layout[(cicer_arietinum$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = cicer_arietinum[,"Run"], replicate = cicer_arietinum[,"BioSample"], layout = cicer_arietinum[,"layout"], genome = "cicer_arietinum", ploidy = 2))
```

## Citrullus lanatus
```{r}
# load data
citrullus_lanatus = fread("../workflow/data/SRArunInfo/SraRunInfo_citrullus_lanatus_PRJNA527790.csv.gz", data.table = F)

# subset only this relevant species
citrullus_lanatus = citrullus_lanatus[(citrullus_lanatus$ScientificName == "Citrullus lanatus"),]

# check for technical replicates
any(duplicated(citrullus_lanatus$Experiment))
any(duplicated(citrullus_lanatus$SampleName))
any(duplicated(citrullus_lanatus$BioSample))

# check if any reads are single-end
any(citrullus_lanatus$spots_with_mates == 0)

# add layout
citrullus_lanatus$layout = NA
citrullus_lanatus$layout[(citrullus_lanatus$spots_with_mates > 0)] = "PE"
citrullus_lanatus$layout[(citrullus_lanatus$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = citrullus_lanatus[,"Run"], replicate = citrullus_lanatus[,"BioSample"], layout = citrullus_lanatus[,"layout"], genome = "citrullus_lanatus", ploidy = 2))
```

## Coffea arabica
```{r}
# load data
coffea_arabica = fread("../workflow/data/SRArunInfo/SraRunInfo_coffea_arabica_PRJNA790687.csv.gz", data.table = F)

# add reads to growing list
# Use experiment label because biosample labels are not unique
allReads = rbind(allReads, data.frame(run = coffea_arabica[,"Run"], replicate = coffea_arabica[,"Experiment"], layout = "PE", genome = "coffea_arabica", ploidy = 4))
```

## Coffea canephora
```{r}
# load data
coffea_canephora = fread("../workflow/data/SRArunInfo/SraRunInfo_coffea_canephora_PRJNA505204.csv.gz", data.table = F)

# subset only this species
coffea_canephora = coffea_canephora[(coffea_canephora$ScientificName == "Coffea canephora"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = coffea_canephora[,"Run"], replicate = coffea_canephora[,"BioSample"], layout = "PE", genome = "coffea_canephora", ploidy = 2))

```

## Corylus avellana
```{r}
# load data
corylus_avellana = fread("../workflow/data/SRArunInfo/SraRunInfo_corylus_avellana_PRJEB32239.csv.gz", data.table = F)

# subset just C. avellana
unique(corylus_avellana$ScientificName)

corylus_avellana = corylus_avellana[(corylus_avellana$ScientificName == "Corylus avellana"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = corylus_avellana[,"Run"], replicate = corylus_avellana[,"BioSample"], layout = "PE", genome = "corylus_avellana", ploidy = 2))
```

## Cucumis melo
```{r}
# load data
cucumis_melo = fread("../workflow/data/SRArunInfo/SraRunInfo_cucumis_melo_PRJNA565104.csv.gz", data.table = F)

# remove reads from outgroup based on publication, 
# see Supplementary table 1
runs2drop = paste("MS-", c(893:898, 900, 901, 995), sep = "")

cucumis_melo = cucumis_melo[!(cucumis_melo$SampleName %in% runs2drop),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = cucumis_melo[,"Run"], replicate = cucumis_melo[,"BioSample"], layout = "PE", genome = "cucumis_melo", ploidy = 2))
```

## Cucumis sativus
```{r}
# load data
cucumis_sativus = fread("../workflow/data/SRArunInfo/SraRunInfo_cucumis_sativus_PRJNA171718.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(cucumis_sativus$Experiment))
any(duplicated(cucumis_sativus$SampleName))
any(duplicated(cucumis_sativus$BioSample))

# check if any reads are single-end
any(cucumis_sativus$spots_with_mates == 0)

# add layout
# checked that spots without mates are PE
cucumis_sativus$layout = NA
cucumis_sativus$layout = "PE"
#cucumis_sativus$layout[(cucumis_sativus$spots_with_mates > 0)] = "PE"
#cucumis_sativus$layout[(cucumis_sativus$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = cucumis_sativus[,"Run"], replicate = cucumis_sativus[,"BioSample"], layout = cucumis_sativus[,"layout"], genome = "cucumis_sativus", ploidy = 2))
```

## Cucurbita argyrosperma
```{r}
# load data
cucurbita_argyrosperma = fread("../workflow/data/SRArunInfo/SraRunInfo_cucurbita_argyrosperma_PRJNA485527.csv.gz", data.table = F)

# subset just C. argyrosperma
unique(cucurbita_argyrosperma$ScientificName)
cucurbita_argyrosperma = cucurbita_argyrosperma[grepl("Cucurbita argyrosperma", cucurbita_argyrosperma$ScientificName),]

# get just RAD-seq data
cucurbita_argyrosperma = cucurbita_argyrosperma[(cucurbita_argyrosperma$LibraryStrategy == "RAD-Seq"),]

# add reads to growing list
# not super sure about this ploidy
# seems like most cucurbits are diploids
allReads = rbind(allReads, data.frame(run = cucurbita_argyrosperma[,"Run"], replicate = cucurbita_argyrosperma[,"BioSample"], layout = "SE", genome = "cucurbita_argyrosperma", ploidy = 2))
```

## Cucurbita pepo
```{r}
# load data
cucurbita_pepo = fread("../workflow/data/SRArunInfo/SraRunInfo_cucurbita_pepo_PRJNA523120.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = cucurbita_pepo[,"Run"], replicate = cucurbita_pepo[,"BioSample"], layout = "SE", genome = "cucurbita_pepo", ploidy = 2))
```

## Digitaria exilis
```{r}
# load data
digitaria_exilis = fread("../workflow/data/SRArunInfo/SraRunInfo_digitaria_exilis_PRJEB36539.csv.gz", data.table = F)

# subset to only this species
digitaria_exilis = digitaria_exilis[(digitaria_exilis$ScientificName == "Digitaria exilis" & digitaria_exilis$LibraryStrategy == "WGS"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = digitaria_exilis[,"Run"], replicate = digitaria_exilis[,"BioSample"], layout = "PE", genome = "digitaria_exilis", ploidy = 4))
```

## Dioscorea alata
```{r}
# load data
dioscorea_alata = fread("../workflow/data/SRArunInfo/SraRunInfo_dioscorea_alata_PRJNA576311.csv.gz", data.table = F)

# load supplementary table with inferred ploidy levels
dioscorea_alata_ploidy = fread("../workflow/data/SRArunInfo/tableS1_dioscorea_alata_PRJNA576311.csv.gz", data.table = F)

dioscorea_alata_ploidy = dioscorea_alata_ploidy[,c("Accession", "Inferred Ploidy")]
names(dioscorea_alata_ploidy)[1] = "SampleName"

dioscorea_alata = merge(dioscorea_alata, dioscorea_alata_ploidy, by = "SampleName", all.x = T)

# What's the most common type of ploidy
table(dioscorea_alata$`Inferred Ploidy`)

# if ploidy is NA, just assume it's 2 because that's by far the most common
dioscorea_alata$`Inferred Ploidy`[is.na(dioscorea_alata$`Inferred Ploidy`)] = 2

# add reads to growing list
allReads = rbind(allReads, data.frame(run = dioscorea_alata[,"Run"], replicate = dioscorea_alata[,"BioSample"], layout = "SE", genome = "dioscorea_alata", ploidy = dioscorea_alata$`Inferred Ploidy`))
```

## Dioscorea rotundata
```{r}
# load data
dioscorea_rotundata = fread("../workflow/data/SRArunInfo/SraRunInfo_dioscorea_rotundata_PRJNA515691.csv.gz", data.table = F)

# subset just one species
dioscorea_rotundata = dioscorea_rotundata[(dioscorea_rotundata$ScientificName == "Dioscorea rotundata"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = dioscorea_rotundata[,"Run"], replicate = dioscorea_rotundata[,"BioSample"], layout = "PE", genome = "dioscorea_rotundata", ploidy = 4))
```

## Durio zibethinus
```{r}
# load data
durio_zibethinus = fread("../workflow/data/SRArunInfo/SraRunInfo_durio_zibethinus_PRJNA867527.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = durio_zibethinus[,"Run"], replicate = durio_zibethinus[,"BioSample"], layout = "PE", genome = "durio_zibethinus", ploidy = 2))
```

## Elaeis guineensis
```{r}
# load data
elaeis_guineensis = fread("../workflow/data/SRArunInfo/SraRunInfo_elaeis_guineensis_PRJDB11628.csv.gz", data.table = F)

# remove long reads
elaeis_guineensis = elaeis_guineensis[(elaeis_guineensis$Platform == "ILLUMINA"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = elaeis_guineensis[,"Run"], replicate = elaeis_guineensis[,"BioSample"], layout = "PE", genome = "elaeis_guineensis", ploidy = 2))
```

## Eleusine coracana
```{r}
# load data
eleusine_coracana = fread("../workflow/data/SRArunInfo/SraRunInfo_eleusine_coracana_PRJNA610152.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = eleusine_coracana[,"Run"], replicate = eleusine_coracana[,"BioSample"], layout = "PE", genome = "eleusine_coracana", ploidy = 4))

```

## Ficus carica
```{r}
# load data
ficus_carica = fread("../workflow/data/SRArunInfo/SraRunInfo_ficus_carica_PRJDB11005.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = ficus_carica[,"Run"], replicate = ficus_carica[,"BioSample"], layout = "PE", genome = "ficus_carica", ploidy = 2))

```

## Glycine max
```{r}
# load data
glycine_max_1 = fread("../workflow/data/SRArunInfo/SraRunInfo_glycine_max_PRJNA681974.csv.gz", data.table = F)
glycine_max_2 = fread("../workflow/data/SRArunInfo/SraRunInfo_glycine_max_PRJNA257011.csv.gz", data.table = F)
glycine_max_3 = fread("../workflow/data/SRArunInfo/SraRunInfo_glycine_max_PRJNA552939.csv.gz", data.table = F)

# remove glycine soja accessions
glycine_max_2 = glycine_max_2[(glycine_max_2$ScientificName == "Glycine max"),]

# check for technical replicates
any(duplicated(glycine_max_1$Experiment))
any(duplicated(glycine_max_1$SampleName))
any(duplicated(glycine_max_1$BioSample))
any(duplicated(glycine_max_1$LibraryName))

any(duplicated(glycine_max_2$Experiment))
any(duplicated(glycine_max_2$SampleName))
any(duplicated(glycine_max_2$BioSample))
any(duplicated(glycine_max_2$LibraryName))

any(duplicated(glycine_max_3$Experiment))
any(duplicated(glycine_max_3$SampleName))
any(duplicated(glycine_max_3$BioSample))
any(duplicated(glycine_max_3$LibraryName))

# bind studies together
glycine_max = rbind(glycine_max_1, glycine_max_2, glycine_max_3)

# Check for duplicates
any(duplicated(glycine_max$Experiment))
any(duplicated(glycine_max$SampleName))
any(duplicated(glycine_max$BioSample))
any(duplicated(glycine_max$LibraryName))

# add layout
glycine_max$layout = NA
glycine_max$layout[(glycine_max$spots_with_mates > 0)] = "PE"
glycine_max$layout[(glycine_max$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = glycine_max[,"Run"], replicate = glycine_max[,"BioSample"], layout = glycine_max[,"layout"], genome = "glycine_max", ploidy = 2))
```

## Glycine soja
```{r}
# load data
glycine_soja = fread("../workflow/data/SRArunInfo/SraRunInfo_glycine_soja_PRJNA597660.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(glycine_soja$Experiment))
any(duplicated(glycine_soja$SampleName))
any(duplicated(glycine_soja$BioSample))

# check if any reads are single-end
any(glycine_soja$spots_with_mates == 0)

# add layout
glycine_soja$layout = NA
glycine_soja$layout[(glycine_soja$spots_with_mates > 0)] = "PE"
glycine_soja$layout[(glycine_soja$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = glycine_soja[,"Run"], replicate = glycine_soja[,"BioSample"], layout = glycine_soja[,"layout"], genome = "glycine_soja", ploidy = 2))
```

## Gossypium arboreum
```{r}
# load data
gossypium_arboreum = fread("../workflow/data/SRArunInfo/SraRunInfo_gossypium_arboreum_PRJNA349094.csv.gz", data.table = F)

# subset this species
gossypium_arboreum = gossypium_arboreum[(gossypium_arboreum$ScientificName == "Gossypium arboreum"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = gossypium_arboreum[,"Run"], replicate = gossypium_arboreum[,"BioSample"], layout = "PE", genome = "gossypium_arboreum", ploidy = 2))
```

## Gossypium barbadense
```{r}
gossypium_barbadense = fread("../workflow/data/SRArunInfo/SraRunInfo_gossypium_barbadense_PRJNA720818.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(gossypium_barbadense$Experiment))
any(duplicated(gossypium_barbadense$SampleName))
any(duplicated(gossypium_barbadense$BioSample))

# check if any reads are single-end
any(gossypium_barbadense$spots_with_mates == 0)

# add layout
gossypium_barbadense$layout = NA
gossypium_barbadense$layout[(gossypium_barbadense$spots_with_mates > 0)] = "PE"
gossypium_barbadense$layout[(gossypium_barbadense$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = gossypium_barbadense[,"Run"], replicate = gossypium_barbadense[,"BioSample"], layout = gossypium_barbadense[,"layout"], genome = "gossypium_barbadense", ploidy = 4))
```

## Gossypium hirsutum
```{r}
gossypium_hirsutum = fread("../workflow/data/SRArunInfo/SraRunInfo_gossypium_hirsutum_PRJNA680449.csv.gz", data.table = F)

# subset only WGS reads
gossypium_hirsutum = gossypium_hirsutum[(gossypium_hirsutum$LibraryStrategy == "WGS" & gossypium_hirsutum$ScientificName == "Gossypium hirsutum" & gossypium_hirsutum$Platform == "ILLUMINA" & gossypium_hirsutum$LibrarySelection == "RANDOM"),]

# check for technical replicates
any(duplicated(gossypium_hirsutum$Experiment))
any(duplicated(gossypium_hirsutum$SampleName))
any(duplicated(gossypium_hirsutum$BioSample))

# check if any reads are single-end
any(gossypium_hirsutum$spots_with_mates == 0)

# add layout
gossypium_hirsutum$layout = NA
gossypium_hirsutum$layout[(gossypium_hirsutum$spots_with_mates > 0)] = "PE"
gossypium_hirsutum$layout[(gossypium_hirsutum$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = gossypium_hirsutum[,"Run"], replicate = gossypium_hirsutum[,"BioSample"], layout = gossypium_hirsutum[,"layout"], genome = "gossypium_hirsutum", ploidy = 4))
```

## Helianthus annuus
```{r}
helianthus_annuus = fread("../workflow/data/SRArunInfo/SraRunInfo_helianthus_annuus_PRJNA292796.csv.gz", data.table = F)

# subset just H. annuus
unique(helianthus_annuus$ScientificName)

helianthus_annuus = helianthus_annuus[(helianthus_annuus$ScientificName == "Helianthus annuus"),]

# add layout
helianthus_annuus$layout = NA
helianthus_annuus$layout[(helianthus_annuus$spots_with_mates > 0)] = "PE"
helianthus_annuus$layout[(helianthus_annuus$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = helianthus_annuus[,"Run"], replicate = helianthus_annuus[,"BioSample"], layout = helianthus_annuus[,"layout"], genome = "helianthus_annuus", ploidy = 2))
```

## Heliosperma pusillum
```{r}
heliosperma_pusillum = fread("../workflow/data/SRArunInfo/SraRunInfo_heliosperma_pusillum_PRJNA308183.csv.gz", data.table = F)

# subset just heliosperma pusillum
unique(heliosperma_pusillum$ScientificName)

heliosperma_pusillum = heliosperma_pusillum[(heliosperma_pusillum$ScientificName == "Heliosperma pusillum"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = heliosperma_pusillum[,"Run"], replicate = heliosperma_pusillum[,"BioSample"], layout = "SE", genome = "heliosperma_pusillum", ploidy = 2))
```

## Hordeum vulgare
```{r}
hordeum_vulgare = fread("../workflow/data/SRArunInfo/SraRunInfo_hordeum_vulgare_PRJEB36577.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(hordeum_vulgare$Experiment))
any(duplicated(hordeum_vulgare$SampleName))
any(duplicated(hordeum_vulgare$BioSample))

# check if any reads are single-end
any(hordeum_vulgare$spots_with_mates == 0)

# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(lib in unique(hordeum_vulgare$LibraryName)){
#   techreps = hordeum_vulgare[(hordeum_vulgare$LibraryName == lib),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }

# add layout
hordeum_vulgare$layout = NA
hordeum_vulgare$layout[(hordeum_vulgare$spots_with_mates > 0)] = "PE"
hordeum_vulgare$layout[(hordeum_vulgare$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = hordeum_vulgare[,"Run"], replicate = hordeum_vulgare[,"BioSample"], layout = hordeum_vulgare[,"layout"], genome = "hordeum_vulgare", ploidy = 2))
```

## Juglans regia
```{r}
juglans_regia = fread("../workflow/data/SRArunInfo/SraRunInfo_juglans_regia_PRJNA721107.csv.gz", data.table = F)

# subset relevant reads
juglans_regia = juglans_regia[(juglans_regia$ScientificName == "Juglans regia"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = juglans_regia[,"Run"], replicate = juglans_regia[,"BioSample"], layout = "PE", genome = "juglans_regia", ploidy = 2))
```

## Lactuca sativa
```{r}
lactuca_sativa = fread("../workflow/data/SRArunInfo/SraRunInfo_lactuca_sativa_PRJNA693894.csv.gz", data.table = F)

# subset relevant reads
lactuca_sativa = lactuca_sativa[(lactuca_sativa$LibraryStrategy == "WGS" & lactuca_sativa$ScientificName == "Lactuca sativa"),]

# check for technical replicates
any(duplicated(lactuca_sativa$Experiment))
any(duplicated(lactuca_sativa$SampleName))
any(duplicated(lactuca_sativa$BioSample))

# add layout
# checked that reads without mates are actually SE
lactuca_sativa$layout = NA
lactuca_sativa$layout[(lactuca_sativa$spots_with_mates > 0)] = "PE"
lactuca_sativa$layout[(lactuca_sativa$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = lactuca_sativa[,"Run"], replicate = lactuca_sativa[,"BioSample"], layout = lactuca_sativa[,"layout"], genome = "lactuca_sativa", ploidy = 2))
```

## Lens culinaris
```{r}
lens_culinaris = fread("../workflow/data/SRArunInfo/SraRunInfo_lens_culinaris_PRJNA639210.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = lens_culinaris[,"Run"], replicate = lens_culinaris[,"BioSample"], layout = "SE", genome = "lens_culinaris", ploidy = 2))
```

## Lens ervoides
```{r}
lens_ervoides = fread("../workflow/data/SRArunInfo/SraRunInfo_lens_ervoides_PRJEB38912.csv.gz", data.table = F)

# subset just one species
unique(lens_ervoides$ScientificName)

lens_ervoides = lens_ervoides[(lens_ervoides$ScientificName == "Lens ervoides"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = lens_ervoides[,"Run"], replicate = lens_ervoides[,"BioSample"], layout = "SE", genome = "lens_ervoides", ploidy = 2))

```


## Linum usitatissimum
```{r}
linum_usitatissimum = fread("../workflow/data/SRArunInfo/SraRunInfo_linum_usitatissimum_PRJNA590636.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(linum_usitatissimum$Experiment))
any(duplicated(linum_usitatissimum$SampleName))
any(duplicated(linum_usitatissimum$BioSample))

# add layout
# checked that reads without mates are actually SE
linum_usitatissimum$layout = NA
linum_usitatissimum$layout = "PE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = linum_usitatissimum[,"Run"], replicate = linum_usitatissimum[,"BioSample"], layout = linum_usitatissimum[,"layout"], genome = "linum_usitatissimum", ploidy = 2))

```

## Lotus japonicus
### Paper says 136 accessions, but I found 137 in the SRA
```{r}
lotus_japonicus = fread("../workflow/data/SRArunInfo/SraRunInfo_lotus_japonicus_PRJEB27969.csv.gz", data.table = F)

# subset relevant reads
lotus_japonicus = lotus_japonicus[(lotus_japonicus$bases != 0),]

# check for technical replicates
any(duplicated(lotus_japonicus$Experiment))
any(duplicated(lotus_japonicus$SampleName))
any(duplicated(lotus_japonicus$BioSample))

# add layout
lotus_japonicus$layout = NA
lotus_japonicus$layout[(lotus_japonicus$spots_with_mates > 0)] = "PE"
lotus_japonicus$layout[(lotus_japonicus$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = lotus_japonicus[,"Run"], replicate = lotus_japonicus[,"BioSample"], layout = lotus_japonicus[,"layout"], genome = "lotus_japonicus", ploidy = 2))
```

## Lupinus angustifolius
```{r}
lupinus_angustifolius = fread("../workflow/data/SRArunInfo/SraRunInfo_lupinus_angustifolius_PRJNA797109.csv.gz", data.table = F)

lupinus_angustifolius = lupinus_angustifolius[(lupinus_angustifolius$Platform == "ILLUMINA"),]

length(unique(lupinus_angustifolius$BioSample))
length(unique(lupinus_angustifolius$Experiment))

# add reads to growing list
# Use experiment labels, because 51 biosamples doesn't match the 55 mentioned in the publication
allReads = rbind(allReads, data.frame(run = lupinus_angustifolius[,"Run"], replicate = lupinus_angustifolius[,"Experiment"], layout = "PE", genome = "lupinus_angustifolius", ploidy = 2))

```

## Macadamia integrifolia
```{r}
# load data
macadamia_integrifolia = fread("../workflow/data/SRArunInfo/SraRunInfo_macadamia_integrifolia_PRJNA909356.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = macadamia_integrifolia[,"Run"], replicate = macadamia_integrifolia[,"BioSample"], layout = "SE", genome = "macadamia_integrifolia", ploidy = 2))

```

## Malus domestica and Malus sylvestris
```{r}
# load data
malus = fread("../workflow/data/SRArunInfo/SraRunInfo_malus_PRJNA322175.csv.gz", data.table = F)

# split data by species
malus_sylvestris = malus[(malus$ScientificName == "Malus sylvestris"),]

malus_domestica = malus[(malus$ScientificName == "Malus domestica"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = malus_sylvestris[,"Run"], replicate = malus_sylvestris[,"BioSample"], layout = "PE", genome = "malus_sylvestris", ploidy = 2))

allReads = rbind(allReads, data.frame(run = malus_domestica[,"Run"], replicate = malus_domestica[,"BioSample"], layout = "PE", genome = "malus_domestica", ploidy = 2))
```

## Mangifera indica
```{r}
mangifera_indica = fread("../workflow/data/SRArunInfo/SraRunInfo_mangifera_indica_PRJNA487154.csv.gz", data.table = F)

mangifera_indica = mangifera_indica[(mangifera_indica$Platform == "ILLUMINA" & mangifera_indica$ScientificName == "Mangifera indica" & mangifera_indica$LibraryStrategy == "WGS"),]

allReads = rbind(allReads, data.frame(run = mangifera_indica[,"Run"], replicate = mangifera_indica[,"BioSample"], layout = "PE", genome = "mangifera_indica", ploidy = 2))
```

## Manihot esculenta
```{r}
manihot_esculenta = fread("../workflow/data/SRArunInfo/SraRunInfo_manihot_esculenta_PRJNA234389.csv.gz", data.table = F)

# subset relevant reads
manihot_esculenta = manihot_esculenta[(manihot_esculenta$LibraryStrategy == "WGS" & manihot_esculenta$ScientificName == "Manihot esculenta" & manihot_esculenta$Platform == "ILLUMINA"),]

# check for technical replicates
any(duplicated(manihot_esculenta$Experiment))
any(duplicated(manihot_esculenta$SampleName))
any(duplicated(manihot_esculenta$BioSample))

# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(biosample in unique(manihot_esculenta$BioSample)){
#   techreps = manihot_esculenta[(manihot_esculenta$BioSample == biosample),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }
# 
# manihot_esculenta = manihot_esculenta[(manihot_esculenta$Run %in% runs),]

# add layout
# reads without mates are actually PE
manihot_esculenta$layout = NA
manihot_esculenta$layout[(manihot_esculenta$spots_with_mates > 0)] = "PE"
manihot_esculenta$layout[(manihot_esculenta$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = manihot_esculenta[,"Run"], replicate = manihot_esculenta[,"BioSample"], layout = manihot_esculenta[,"layout"], genome = "manihot_esculenta", ploidy = 2))
```



## Medicago truncatula
```{r}
medicago_truncatula = fread("../workflow/data/SRArunInfo/SraRunInfo_medicago_truncatula_SRA026748.csv.gz", data.table = F)

# subset relevant reads
any(duplicated(medicago_truncatula$BioSample))
length(unique(medicago_truncatula$SampleName))

# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(biosample in unique(medicago_truncatula$SampleName)){
#   techreps = medicago_truncatula[(medicago_truncatula$SampleName == biosample),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }
# 
# medicago_truncatula = medicago_truncatula[(medicago_truncatula$Run %in% runs),]

# add layout
medicago_truncatula$layout = NA
medicago_truncatula$layout[(medicago_truncatula$spots_with_mates > 0)] = "PE"
medicago_truncatula$layout[(medicago_truncatula$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = medicago_truncatula[,"Run"], replicate = medicago_truncatula[,"BioSample"], layout = medicago_truncatula[,"layout"], genome = "medicago_truncatula", ploidy = 2))
```

## Mimulus guttatus
```{r}
mimulus_guttatus = fread("../workflow/data/SRArunInfo/SraRunInfo_mimulus_guttatus_PRJNA344904.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(mimulus_guttatus$Experiment))
any(duplicated(mimulus_guttatus$SampleName))
any(duplicated(mimulus_guttatus$BioSample))

# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(biosample in unique(mimulus_guttatus$SampleName)){
#   techreps = mimulus_guttatus[(mimulus_guttatus$SampleName == biosample),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }
# 
# mimulus_guttatus = mimulus_guttatus[(mimulus_guttatus$Run %in% runs),]

# subset relevant runs
# create sample names vector based on info in paper
samples = gsub("^", "SAMN0", 5852485:5852522)
mimulus_guttatus= mimulus_guttatus[(mimulus_guttatus$BioSample %in% samples),]

# add layout
mimulus_guttatus$layout = NA
mimulus_guttatus$layout[(mimulus_guttatus$spots_with_mates > 0)] = "PE"
mimulus_guttatus$layout[(mimulus_guttatus$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = mimulus_guttatus[,"Run"], replicate = mimulus_guttatus[,"BioSample"], layout = mimulus_guttatus[,"layout"], genome = "mimulus_guttatus", ploidy = 2))
```

## Miscanthus sinensis
```{r}
# load data
miscanthus_sinensis = fread("../workflow/data/SRArunInfo/SraRunInfo_miscanthus_sinensis_PRJNA207721.csv.gz", data.table = F)

# subset just M. sinensis
unique(miscanthus_sinensis$ScientificName)

miscanthus_sinensis = miscanthus_sinensis[grepl("Miscanthus sinensis", miscanthus_sinensis$ScientificName),]

length(miscanthus_sinensis$BioSample)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = miscanthus_sinensis[,"Run"], replicate = miscanthus_sinensis[,"BioSample"], layout = "SE", genome = "miscanthus_sinensis", ploidy = 2))
```

## Momordica charantia
```{r}
# load data
momordica_charantia = fread("../workflow/data/SRArunInfo/SraRunInfo_momordica_charantia_PRJEB24033.csv.gz", data.table = F)

# subset to just one species
momordica_charantia = momordica_charantia[grepl("Momordica charantia", momordica_charantia$ScientificName),]

momordica_charantia = momordica_charantia[(momordica_charantia$LibraryStrategy == "WGS"),]

momordica_charantia = momordica_charantia[!is.na(momordica_charantia$LoadDate),]

# number of individuals
length(unique(momordica_charantia$BioSample))

# add reads to growing list
allReads = rbind(allReads, data.frame(run = momordica_charantia[,"Run"], replicate = momordica_charantia[,"BioSample"], layout = "PE", genome = "momordica_charantia", ploidy = 2))

```


## Musa
```{r}

musa = fread("../workflow/data/SRArunInfo/SraRunInfo_musa_PRJEB58004.csv.gz", data.table = F)

musa_acuminata = musa[grepl("Musa acuminata", musa$ScientificName),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = musa_acuminata[,"Run"], replicate = musa_acuminata[,"BioSample"], layout = "PE", genome = "musa_acuminata", ploidy = 3))
```

## Nelumbo nucifera
```{r}
# load data
nelumbo_nucifera = fread("../workflow/data/SRArunInfo/SraRunInfo_nelumbo_nucifera_PRJNA357653.csv.gz", data.table = F)

# subset one species
nelumbo_nucifera = nelumbo_nucifera[(nelumbo_nucifera$ScientificName == "Nelumbo nucifera"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = nelumbo_nucifera[,"Run"], replicate = nelumbo_nucifera[,"BioSample"], layout = "PE", genome = "nelumbo_nucifera", ploidy = 2))
```

## Nicotiana tabacum
```{r}
nicotiana_tabacum = fread("../workflow/data/SRArunInfo/SraRunInfo_nicotiana_tabacum_PRJNA904563.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = nicotiana_tabacum[,"Run"], replicate = nicotiana_tabacum[,"BioSample"], layout = "PE", genome = "nicotiana_tabacum", ploidy = 4))
```

## Olea europaea
```{r}
olea_europaea = fread("../workflow/data/SRArunInfo/SraRunInfo_olea_europaea_PRJEB35540.csv.gz", data.table = F)

# species has duplicate biosamples but I don't think they're actually technical replicates
length(unique(olea_europaea$BioSample))

# add layout
olea_europaea$layout = NA
olea_europaea$layout[(olea_europaea$spots_with_mates > 0)] = "PE"
olea_europaea$layout[(olea_europaea$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = olea_europaea[,"Run"], replicate = olea_europaea[,"BioSample"], layout = olea_europaea[,"layout"], genome = "olea_europaea", ploidy = 2))
```

## Oryza barthii
```{r eval=FALSE, include=FALSE}
oryza_barthii = fread("../workflow/data/SRArunInfo/SraRunInfo_oryza_barthii_PRJDB4702.csv.gz", data.table = F)

# add reads to growing list
# ***not sure about the ploidy here***
allReads = rbind(allReads, data.frame(run = oryza_barthii[,"Run"], replicate = oryza_barthii[,"BioSample"], layout = "PE", genome = "oryza_barthii", ploidy = 2))
```

## Oryza brachyantha
```{r}
oryza_brachyantha = fread("../workflow/data/SRArunInfo/SraRunInfo_oryza_brachyantha_PRJDB4059.csv.gz", data.table = F)

allReads = rbind(allReads, data.frame(run = oryza_brachyantha[,"Run"], replicate = oryza_brachyantha[,"BioSample"], layout = "PE", genome = "oryza_brachyantha", ploidy = 2))
```

## Oryza glaberrima and Oryza barthii
```{r}
oryza = fread("../workflow/data/SRArunInfo/SraRunInfo_PRJEB21312.csv.gz", data.table = F)

# breakdown between species
table(oryza$ScientificName)

# separte species
oryza_barthii = oryza[(oryza$ScientificName == "Oryza barthii"),]
oryza_glaberrima = oryza[(oryza$ScientificName == "Oryza glaberrima"),]

allReads = rbind(allReads, data.frame(run = oryza_barthii[,"Run"], replicate = oryza_barthii[,"BioSample"], layout = "PE", genome = "oryza_barthii", ploidy = 2))

allReads = rbind(allReads, data.frame(run = oryza_glaberrima[,"Run"], replicate = oryza_glaberrima[,"BioSample"], layout = "PE", genome = "oryza_glaberrima", ploidy = 2))
```

## Oryza glumipatula
```{r}
oryza_glumipatula = fread("../workflow/data/SRArunInfo/SraRunInfo_oryza_glumipatula_PRJDB4703.csv.gz", data.table = F)

allReads = rbind(allReads, data.frame(run = oryza_glumipatula[,"Run"], replicate = oryza_glumipatula[,"BioSample"], layout = "PE", genome = "oryza_glumipatula", ploidy = 2))
```

## Oryza longistaminata
```{r}
oryza_longistaminata = fread("../workflow/data/SRArunInfo/SraRunInfo_oryza_longistaminata_PRJDB4705.csv.gz", data.table = F)

allReads = rbind(allReads, data.frame(run = oryza_longistaminata[,"Run"], replicate = oryza_longistaminata[,"BioSample"], layout = "PE", genome = "oryza_longistaminata", ploidy = 2))
```

## Oryza meridionalis
```{r}
oryza_meridionalis = fread("../workflow/data/SRArunInfo/SraRunInfo_oryza_meridionalis_PRJDB4704.csv.gz", data.table = F)

# add reads to growing list
# ***not sure about ploidy here***
allReads = rbind(allReads, data.frame(run = oryza_meridionalis[,"Run"], replicate = oryza_meridionalis[,"BioSample"], layout = "PE", genome = "oryza_meridionalis", ploidy = 2))

```

## Oryza nivara
```{r eval=FALSE, include=FALSE}
oryza_nivara = fread("../workflow/data/SRArunInfo/SraRunInfo_oryza_nivara_PRJNA606132.csv.gz", data.table = F)
```

## Oryza punctata
```{r}
oryza_punctata = fread("../workflow/data/SRArunInfo/SraRunInfo_oryza_punctata_PRJDB4620.csv.gz", data.table = F)

allReads = rbind(allReads, data.frame(run = oryza_punctata[,"Run"], replicate = oryza_punctata[,"BioSample"], layout = "PE", genome = "oryza_punctata", ploidy = 2))
```

## Oryza rufipogon
```{r}
oryza_rufipogon = fread("../workflow/data/SRArunInfo/SraRunInfo_oryza_rufipogon_PRJEB2829.csv.gz", data.table = F)

# only include this species
oryza_rufipogon = oryza_rufipogon[grepl("Oryza rufipogon", oryza_rufipogon$ScientificName),]

length(unique(oryza_rufipogon$BioSample))

# add reads to growing list
# ***not sure about ploidy here***
allReads = rbind(allReads, data.frame(run = oryza_rufipogon[,"Run"], replicate = oryza_rufipogon[,"BioSample"], layout = "PE", genome = "oryza_rufipogon", ploidy = 2))
```

## Oryza sativa
```{r}
oryza_sativa = fread("../workflow/data/SRArunInfo/SraRunInfo_oryza_sativa_PRJEB6180.csv.gz", data.table = F)

# how many genotypes are there?
length(unique(oryza_sativa$BioSample))

# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(biosample in unique(oryza_sativa$BioSample)){
#   techreps = oryza_sativa[(oryza_sativa$BioSample == biosample),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }

# oryza_sativa = oryza_sativa[(oryza_sativa$Run %in% runs),]

# add layout
oryza_sativa$layout = NA
oryza_sativa$layout = "PE" # even the samples without mates are actually paired-end

# add reads to growing list
allReads = rbind(allReads, data.frame(run = oryza_sativa[,"Run"], replicate = oryza_sativa[,"BioSample"], layout = oryza_sativa[,"layout"], genome = "oryza_sativa", ploidy = 2))
```


## Panicum halli
```{r}
# To get run numbers for panicum halli
# Copy bioprojects from table S1 in this publication: https://doi.org/10.1038/s41467-018-07669-x
# to a list in doc/panicum_halli_search.txt
# then use regular expressions to make a query to be used with ENTREZ
# cat panicum_halli_search.txt | tr '\n' ' ' | sed 's/ / OR /g'
# esearch -db sra -query $QUERY | efetch --format runinfo > panicum_halli.csv
panicum_halli = fread("../workflow/data/SRArunInfo/SraRunInfo_panicum_halli.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(panicum_halli$Experiment))
any(duplicated(panicum_halli$SampleName))
any(duplicated(panicum_halli$BioSample))

# runs = NULL
# for(biosample in unique(panicum_halli$BioSample)){
#   techreps = panicum_halli[(panicum_halli$BioSample == biosample),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }
# 
# panicum_halli = panicum_halli[(panicum_halli$Run %in% runs),]

# remove pacbio reads
panicum_halli = panicum_halli[(panicum_halli$Platform != "PACBIO_SMRT"),]

# add layout
# reads without mates are actually PE
panicum_halli$layout = NA
panicum_halli$layout = "PE"
#panicum_halli$layout[(panicum_halli$spots_with_mates > 0)] = "PE"
#panicum_halli$layout[(panicum_halli$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = panicum_halli[,"Run"], replicate = panicum_halli[,"BioSample"], layout = panicum_halli[,"layout"], genome = "panicum_halli", ploidy = 2))
```

## Panicum virgatum
```{r}
panicum_virgatum = fread("../workflow/data/SRArunInfo/SraRunInfo_panicum_virgatum_PRJNA622568.csv.gz", data.table = F)

# add layout
panicum_virgatum$layout = NA
panicum_virgatum$layout[(panicum_virgatum$spots_with_mates > 0)] = "PE"
panicum_virgatum$layout[(panicum_virgatum$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = panicum_virgatum[,"Run"], replicate = panicum_virgatum[,"BioSample"], layout = panicum_virgatum[,"layout"], genome = "panicum_virgatum", ploidy = 4))
```

## Papaver somniferum
```{r}
papaver_somniferum = fread("../workflow/data/SRArunInfo/SraRunInfo_papaver_somniferum_PRJNA759347.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = papaver_somniferum[,"Run"], replicate = papaver_somniferum[,"BioSample"], layout = "SE", genome = "papaver_somniferum", ploidy = 2))
```

## Phaseolus vulgaris
```{r}
phaseolus_vulgaris = fread("../workflow/data/SRArunInfo/SraRunInfo_phaseolus_vulgaris_PRJNA515107.csv.gz", data.table = F)

# subset relevant reads, need random library selection
phaseolus_vulgaris = phaseolus_vulgaris[(phaseolus_vulgaris$LibrarySelection == "RANDOM"),]

# check for technical replicates
any(duplicated(phaseolus_vulgaris$Experiment))
any(duplicated(phaseolus_vulgaris$SampleName))
any(duplicated(phaseolus_vulgaris$BioSample))

# add layout
phaseolus_vulgaris$layout = NA
phaseolus_vulgaris$layout[(phaseolus_vulgaris$spots_with_mates > 0)] = "PE"
phaseolus_vulgaris$layout[(phaseolus_vulgaris$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = phaseolus_vulgaris[,"Run"], replicate = phaseolus_vulgaris[,"BioSample"], layout = phaseolus_vulgaris[,"layout"], genome = "phaseolus_vulgaris", ploidy = 2))
```

## Phoenix dactylifera
```{r}
phoenix_dactylifera = fread("../workflow/data/SRArunInfo/SraRunInfo_phoenix_dactylifera_PRJNA505141.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = phoenix_dactylifera[,"Run"], replicate = phoenix_dactylifera[,"BioSample"], layout = "PE", genome = "phoenix_dactylifera", ploidy = 2))
```

## Pisum sativum
```{r}
pisum_sativum_1 = fread("../workflow/data/SRArunInfo/SraRunInfo_pisum_sativum_PRJNA285605.csv.gz", data.table = F)
pisum_sativum_2 = fread("../workflow/data/SRArunInfo/SraRunInfo_pisum_sativum_PRJNA431567.csv.gz", data.table = F)

# subset to just pisum sativum
pisum_sativum_1 = pisum_sativum_1[grepl("Pisum sativum", pisum_sativum_1$ScientificName),]
pisum_sativum_2 = pisum_sativum_2[grepl("Pisum sativum", pisum_sativum_2$ScientificName),]

pisum_sativum = rbind(pisum_sativum_1,pisum_sativum_2)

# add layout
pisum_sativum$layout = NA
pisum_sativum$layout[(pisum_sativum$spots_with_mates > 0)] = "PE"
pisum_sativum$layout[(pisum_sativum$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = pisum_sativum[,"Run"], replicate = pisum_sativum[,"BioSample"], layout = pisum_sativum[,"layout"], genome = "pisum_sativum", ploidy = 2))
```

## Populus deltoides
```{r}
populus_deltoides = fread("../workflow/data/SRArunInfo/SraRunInfo_populus_deltoides_PRJNA628142.csv.gz", data.table = F)

# add reads to growing list
# not sure about the ploidy here
allReads = rbind(allReads, data.frame(run = populus_deltoides[,"Run"], replicate = populus_deltoides[,"BioSample"], layout = "PE", genome = "populus_deltoides", ploidy = 4))
```

## Populus trichocarpa
```{r}
populus_trichocarpa = fread("../workflow/data/SRArunInfo/SraRunInfo_populus_trichocarpa_PRJNA276056.csv.gz", data.table = F)

# subset relevant reads
populus_trichocarpa = populus_trichocarpa[(populus_trichocarpa$ScientificName == "Populus trichocarpa"),]

# add layout
populus_trichocarpa$layout = NA
populus_trichocarpa$layout = "PE"
#populus_trichocarpa$layout[(populus_trichocarpa$spots_with_mates > 0)] = "PE"
#populus_trichocarpa$layout[(populus_trichocarpa$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = populus_trichocarpa[,"Run"], replicate = populus_trichocarpa[,"BioSample"], layout = populus_trichocarpa[,"layout"], genome = "populus_trichocarpa", ploidy = 2))
```

## Prunus armeniaca
```{r}
prunus_armeniaca = fread("../workflow/data/SRArunInfo/SraRunInfo_prunus_armeniaca_PRJEB40984.csv.gz", data.table = F)

prunus_armeniaca = prunus_armeniaca[(prunus_armeniaca$ScientificName == "Prunus armeniaca"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = prunus_armeniaca[,"Run"], replicate = prunus_armeniaca[,"BioSample"], layout = "PE", genome = "prunus_armeniaca", ploidy = 2))
```

## Prunus avium
```{r}
prunus_avium = fread("../workflow/data/SRArunInfo/SraRunInfo_prunus_avium_PRJNA813711.csv.gz", data.table = F)

length(unique(prunus_avium$BioSample))
length(unique(prunus_avium$Experiment))
length(unique(prunus_avium$LibraryName))
length(unique(prunus_avium$SampleName))

# add reads to growing list
allReads = rbind(allReads, data.frame(run = prunus_avium[,"Run"], replicate = prunus_avium[,"BioSample"], layout = "PE", genome = "prunus_avium", ploidy = 2))
```

## Prunus persica
```{r}
prunus_persica = fread("../workflow/data/SRArunInfo/SraRunInfo_prunus_persica_PRJNA388029.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(prunus_persica$Experiment))
any(duplicated(prunus_persica$SampleName))
any(duplicated(prunus_persica$BioSample))

# Some of these accessions are not Pruus persica, load table S1 from publication to filter
prunus_persica_supp = fread("../workflow/data/SRArunInfo/doi10.1186s13059-019-1648-9_prunus_persica_TableS1_2022-10-20.csv.gz", data.table = F)

ids = prunus_persica_supp$ID[(prunus_persica_supp$"Botanical varieties" == "P. persica L.")]

prunus_persica = prunus_persica[(prunus_persica$SampleName %in% ids),]

# add layout
prunus_persica$layout = NA
prunus_persica$layout[(prunus_persica$spots_with_mates > 0)] = "PE"
prunus_persica$layout[(prunus_persica$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = prunus_persica[,"Run"], replicate = prunus_persica[,"BioSample"], layout = prunus_persica[,"layout"], genome = "prunus_persica", ploidy = 2))
```

## Quercus lobata
```{r}
quercus_lobata = fread("../workflow/data/SRArunInfo/SraRunInfo_quercus_lobata_PRJNA729978.csv.gz", data.table = F)

# subset only quercus lobata
quercus_lobata = quercus_lobata[(quercus_lobata$ScientificName == "Quercus lobata"),]

# number of genotypes
length(unique(quercus_lobata$BioSample))
length(unique(quercus_lobata$Experiment))
length(unique(quercus_lobata$LibraryName))
length(unique(quercus_lobata$SampleName))

# add layout
quercus_lobata$layout = NA
quercus_lobata$layout[(quercus_lobata$spots_with_mates > 0)] = "PE"
quercus_lobata$layout[(quercus_lobata$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = quercus_lobata[,"Run"], replicate = quercus_lobata[,"BioSample"], layout = quercus_lobata[,"layout"], genome = "quercus_lobata", ploidy = 2))
```

## Quercus robur
```{r}
quercus_robur = fread("../workflow/data/SRArunInfo/SraRunInfo_quercus_robur_PRJEB30573.csv.gz", data.table = F)

quercus_robur = quercus_robur[(quercus_robur$ScientificName == "Quercus robur"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = quercus_robur[,"Run"], replicate = quercus_robur[,"BioSample"], layout = "PE", genome = "quercus_robur", ploidy = 2))
```

## Raphanus sativus
```{r}
raphanus_sativus = fread("../workflow/data/SRArunInfo/SraRunInfo_raphanus_sativus_PRJNA634208.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = raphanus_sativus[,"Run"], replicate = raphanus_sativus[,"BioSample"], layout = "PE", genome = "raphanus_sativus", ploidy = 2))
```

## Rhododendron griersonianum
## not sure about this ploidy
```{r}
rhododendron_griersonianum = fread("../workflow/data/SRArunInfo/SraRunInfo_rhododendron_griersonianum_PRJNA659608.csv.gz", data.table = F)

rhododendron_griersonianum = rhododendron_griersonianum[(rhododendron_griersonianum$LibraryStrategy == "WGS" & rhododendron_griersonianum$ScientificName == "Rhododendron griersonianum" & rhododendron_griersonianum$Platform == "ILLUMINA"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = rhododendron_griersonianum[,"Run"], replicate = rhododendron_griersonianum[,"BioSample"], layout = "PE", genome = "rhododendron_griersonianum", ploidy = 2))
```

## Salix dunnii
```{r}
salix_dunnii = fread("../workflow/data/SRArunInfo/SraRunInfo_salix_dunnii_PRJNA671733.csv.gz", data.table = F)

salix_dunnii = salix_dunnii[(salix_dunnii$Platform == "ILLUMINA" & salix_dunnii$LibrarySelection != "Restriction Digest"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = salix_dunnii[,"Run"], replicate = salix_dunnii[,"BioSample"], layout = "PE", genome = "salix_dunnii", ploidy = 2))
```


## Secale cereale
```{r}
secale_cereale = fread("../workflow/data/SRArunInfo/SraRunInfo_secale_cereale_PRJEB6215.csv.gz", data.table = F)

# remove wild lines
secale_cereale = secale_cereale[(secale_cereale$ScientificName == "Secale cereale"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = secale_cereale[,"Run"], replicate = secale_cereale[,"BioSample"], layout = "PE", genome = "secale_cereale", ploidy = 2))

```

## Sesamum indicum
```{r}
sesamum_indicum = fread("../workflow/data/SRArunInfo/SraRunInfo_sesamum_indicum_PRJEB8078.csv.gz", data.table = F)

# subset only sesamum indicum
sesamum_indicum = sesamum_indicum[(sesamum_indicum$ScientificName == "Sesamum indicum"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = sesamum_indicum[,"Run"], replicate = sesamum_indicum[,"BioSample"], layout = "PE", genome = "sesamum_indicum", ploidy = 2))
```

## Setaria italica
```{r}
setaria_italica = fread("../workflow/data/SRArunInfo/SraRunInfo_setaria_italica_PRJEB1234.csv.gz", data.table = F)

# remove Setaria virdis lines, though maybe I could keep them later?
setaria_italica = setaria_italica[(setaria_italica$ScientificName == "Setaria italica"),]

# check for technical replicates
any(duplicated(setaria_italica$Experiment))
any(duplicated(setaria_italica$SampleName))
any(duplicated(setaria_italica$BioSample))

# add layout
setaria_italica$layout = NA
setaria_italica$layout[(setaria_italica$spots_with_mates > 0)] = "PE"
setaria_italica$layout[(setaria_italica$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = setaria_italica[,"Run"], replicate = setaria_italica[,"BioSample"], layout = setaria_italica[,"layout"], genome = "setaria_italica", ploidy = 2))
```

## Setaria viridis
```{r}
setaria_viridis = fread("../workflow/data/SRArunInfo/SraRunInfo_setaria_viridis_PRJNA633413.csv.gz", data.table = F)

setaria_viridis = setaria_viridis[(setaria_viridis$ScientificName == "Setaria viridis"),]

# add layout
setaria_viridis$layout = NA
setaria_viridis$layout[(setaria_viridis$spots_with_mates > 0)] = "PE"
setaria_viridis$layout[(setaria_viridis$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = setaria_viridis[,"Run"], replicate = setaria_viridis[,"BioSample"], layout = setaria_viridis[,"layout"], genome = "setaria_viridis", ploidy = 2))
```

## Solanum chilense
```{r}
solanum_chilense = fread("../workflow/data/SRArunInfo/SraRunInfo_solanum_chilense_PRJEB47577.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = solanum_chilense[,"Run"], replicate = solanum_chilense[,"BioSample"], layout = "PE", genome = "solanum_chilense", ploidy = 2))
```

## Solanum lycopersicum
```{r}
solanum_lycopersicum = fread("../workflow/data/SRArunInfo/SraRunInfo_solanum_lycopersicum_PRJNA259308.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(solanum_lycopersicum$Experiment))
any(duplicated(solanum_lycopersicum$SampleName))
any(duplicated(solanum_lycopersicum$BioSample))

# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(biosample in unique(solanum_lycopersicum$BioSample)){
#   techreps = solanum_lycopersicum[(solanum_lycopersicum$BioSample == biosample),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }
# 
# solanum_lycopersicum = solanum_lycopersicum[(solanum_lycopersicum$Run %in% runs),]

# add layout
solanum_lycopersicum$layout = NA
solanum_lycopersicum$layout[(solanum_lycopersicum$spots_with_mates > 0)] = "PE"
solanum_lycopersicum$layout[(solanum_lycopersicum$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = solanum_lycopersicum[,"Run"], replicate = solanum_lycopersicum[,"BioSample"], layout = solanum_lycopersicum[,"layout"], genome = "solanum_lycopersicum", ploidy = 2))
```

## Solanum melongena
```{r}
solanum_melongena = fread("../workflow/data/SRArunInfo/SraRunInfo_solanum_melongena_PRJNA649091.csv.gz", data.table = F)

# remove hi-c library
solanum_melongena = solanum_melongena[(solanum_melongena$LibraryStrategy == "WGS" & solanum_melongena$ScientificName == "Solanum melongena"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = solanum_melongena[,"Run"], replicate = solanum_melongena[,"BioSample"], layout = "PE", genome = "solanum_melongena", ploidy = 2))
```

## Solanum pimpinellifolium
```{r}
solanum_pimpinellifolium = fread("../workflow/data/SRArunInfo/SraRunInfo_solanum_pimpinellifolium_PRJNA454805.csv.gz", data.table = F)

# subset just this species
solanum_pimpinellifolium = solanum_pimpinellifolium[(solanum_pimpinellifolium$ScientificName == "Solanum pimpinellifolium"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = solanum_pimpinellifolium[,"Run"], replicate = solanum_pimpinellifolium[,"BioSample"], layout = "PE", genome = "solanum_pimpinellifolium", ploidy = 2))

```

## Solanum stenotomum
```{r}
solanum_stenotomum = fread("../workflow/data/SRArunInfo/SraRunInfo_solanum_stenotomum_PRJNA394943.csv.gz", data.table = F)

# subset to one species
solanum_stenotomum = solanum_stenotomum[grepl("Solanum stenotomum", solanum_stenotomum$ScientificName),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = solanum_stenotomum[,"Run"], replicate = solanum_stenotomum[,"BioSample"], layout = "PE", genome = "solanum_stenotomum", ploidy = 2))
```

## Sorghum bicolor
```{r}
sorghum_bicolor = fread("../workflow/data/SRArunInfo/SraRunInfo_sorghum_bicolor_PRJEB50066.csv.gz", data.table = F)

# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(biosample in unique(sorghum_bicolor$BioSample)){
#   techreps = sorghum_bicolor[(sorghum_bicolor$BioSample == biosample),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }
# 
# sorghum_bicolor = sorghum_bicolor[(sorghum_bicolor$Run %in% runs),]

# add layout
# reads without mates are still PE
sorghum_bicolor$layout = NA
sorghum_bicolor$layout = "PE"
#sorghum_bicolor$layout[(sorghum_bicolor$spots_with_mates > 0)] = "PE"
#sorghum_bicolor$layout[(sorghum_bicolor$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = sorghum_bicolor[,"Run"], replicate = sorghum_bicolor[,"BioSample"], layout = sorghum_bicolor[,"layout"], genome = "sorghum_bicolor", ploidy = 2))
```

## Spinacia oleracea
```{r}
spinacia_oleracea = fread("../workflow/data/SRArunInfo/SraRunInfo_spinacia_oleracea_PRJNA598728.csv.gz", data.table = F)

spinacia_oleracea = spinacia_oleracea[(spinacia_oleracea$ScientificName == "Spinacia oleracea"),]

# check for technical replicates
any(duplicated(spinacia_oleracea$Experiment))
any(duplicated(spinacia_oleracea$SampleName))
any(duplicated(spinacia_oleracea$BioSample))

# add layout
# checked that reads without mates are PE
spinacia_oleracea$layout = NA
spinacia_oleracea$layout = "PE"
#spinacia_oleracea$layout[(spinacia_oleracea$spots_with_mates > 0)] = "PE"
#spinacia_oleracea$layout[(spinacia_oleracea$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = spinacia_oleracea[,"Run"], replicate = spinacia_oleracea[,"BioSample"], layout = spinacia_oleracea[,"layout"], genome = "spinacia_oleracea", ploidy = 2))
```

## Spirodela polyrhiza
```{r}
spirodela_polyrhiza = fread("../workflow/data/SRArunInfo/SraRunInfo_spirodela_polyrhiza_PRJNA552613.csv.gz", data.table = F)

# check for technical replicates
any(duplicated(spirodela_polyrhiza$Experiment))
any(duplicated(spirodela_polyrhiza$SampleName))
any(duplicated(spirodela_polyrhiza$BioSample))

# add layout
# reads without mates are SE
spirodela_polyrhiza$layout = NA
spirodela_polyrhiza$layout[(spirodela_polyrhiza$spots_with_mates > 0)] = "PE"
spirodela_polyrhiza$layout[(spirodela_polyrhiza$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = spirodela_polyrhiza[,"Run"], replicate = spirodela_polyrhiza[,"BioSample"], layout = spirodela_polyrhiza[,"layout"], genome = "spirodela_polyrhiza", ploidy = 2))
```

## Striga hermonthica
```{r}
# load data
striga_hermonthica = fread("../workflow/data/SRArunInfo/SraRunInfo_striga_hermonthica_PRJNA801489.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = striga_hermonthica[,"Run"], replicate = striga_hermonthica[,"BioSample"], layout = "PE", genome = "striga_hermonthica", ploidy = 2))
```

## Tetracentron sinense
## not sure about this ploidy
```{r}
# load data
tetracentron_sinense = fread("../workflow/data/SRArunInfo/SraRunInfo_tetracentron_sinense_PRJNA625382.csv.gz", data.table = F)

tetracentron_sinense = tetracentron_sinense[(tetracentron_sinense$Platform == "ILLUMINA" & tetracentron_sinense$LibraryStrategy == "WGS"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = tetracentron_sinense[,"Run"], replicate = tetracentron_sinense[,"BioSample"], layout = "PE", genome = "tetracentron_sinense", ploidy = 2))

```

## Thinopyrum intermedium
```{r}
# load data
thinopyrum_intermedium = fread("../workflow/data/SRArunInfo/SraRunInfo_thinopyrum_intermedium_PRJNA722274.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = thinopyrum_intermedium[,"Run"], replicate = thinopyrum_intermedium[,"BioSample"], layout = "SE", genome = "thinopyrum_intermedium", ploidy = 6))
```

## Thlaspi arvense
```{r}
# load data
thlaspi_arvense = fread("../workflow/data/SRArunInfo/SraRunInfo_thlaspi_arvense_PRJEB46635.csv.gz", data.table = F)

thlaspi_arvense = thlaspi_arvense[(thlaspi_arvense$LibraryStrategy == "WGS" & thlaspi_arvense$Platform == "ILLUMINA"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = thlaspi_arvense[,"Run"], replicate = thlaspi_arvense[,"BioSample"], layout = "PE", genome = "thlaspi_arvense", ploidy = 2))

```

## Trifolium pratense
```{r}
trifolium_pratense = fread("../workflow/data/SRArunInfo/SraRunInfo_trifolium_pratense_PRJNA765476.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = trifolium_pratense[,"Run"], replicate = trifolium_pratense[,"BioSample"], layout = "SE", genome = "trifolium_pratense", ploidy = 2))
```

## Triticum aestivum
```{r}
# load data
triticum_aestivum_1 = fread("../workflow/data/SRArunInfo/SraRunInfo_triticum_aestivum_PRJEB48738.csv.gz", data.table = F)
triticum_aestivum_2 = fread("../workflow/data/SRArunInfo/SraRunInfo_triticum_aestivum_PRJEB48988.csv.gz", data.table = F)

# concatenate
triticum_aestivum = rbind(triticum_aestivum_1, triticum_aestivum_2)

# check for technical replicates
any(duplicated(triticum_aestivum$Experiment))
any(duplicated(triticum_aestivum$SampleName))
any(duplicated(triticum_aestivum$BioSample))

# add layout
# checked that reads without mates are PE
triticum_aestivum$layout = NA
triticum_aestivum$layout = "PE"
#triticum_aestivum$layout[(triticum_aestivum$spots_with_mates > 0)] = "PE"
#triticum_aestivum$layout[(triticum_aestivum$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = triticum_aestivum[,"Run"], replicate = triticum_aestivum[,"BioSample"], layout = triticum_aestivum[,"layout"], genome = "triticum_aestivum", ploidy = 6))

```

## Triticum dicoccoides
```{r}
triticum_dicoccoides = fread("../workflow/data/SRArunInfo/SraRunInfo_triticum_dicoccoides_PRJEB42105.csv.gz", data.table = F)

# subset just RAD-seq because WGS data is for chloroplast only
triticum_dicoccoides = triticum_dicoccoides[(triticum_dicoccoides$LibraryStrategy == "RAD-Seq"),]

# subset just T. dicoccoides or it's synonym
unique(triticum_dicoccoides$ScientificName)

triticum_dicoccoides = triticum_dicoccoides[(triticum_dicoccoides$ScientificName == "Triticum dicoccoides" | triticum_dicoccoides$ScientificName == "Triticum dicoccum"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = triticum_dicoccoides[,"Run"], replicate = triticum_dicoccoides[,"BioSample"], layout = "SE", genome = "triticum_dicoccoides", ploidy = 4))
```

## Triticum turgidum
```{r}
triticum_turgidum = fread("../workflow/data/SRArunInfo/SraRunInfo_triticum_turgidum_PRJNA663409.csv.gz", data.table = F)

# subset relevant species
unique(triticum_turgidum$ScientificName)

triticum_turgidum = triticum_turgidum[grepl("Triticum turgidum", triticum_turgidum$ScientificName),]

# add layout, check if any reads are potentially single-end
any(triticum_turgidum$spots_with_mates == 0)
triticum_turgidum$layout = NA
triticum_turgidum$layout = "PE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = triticum_turgidum[,"Run"], replicate = triticum_turgidum[,"BioSample"], layout = triticum_turgidum[,"layout"], genome = "triticum_turgidum", ploidy = 4))

```

## Vanilla planifolia
```{r}
vanilla_planifolia = fread("../workflow/data/SRArunInfo/SraRunInfo_vanilla_planifolia_PRJNA733522.csv.gz", data.table = F)

# subset just V. planifolia
unique(vanilla_planifolia$ScientificName)

vanilla_planifolia = vanilla_planifolia[(vanilla_planifolia$ScientificName == "Vanilla planifolia"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = vanilla_planifolia[,"Run"], replicate = vanilla_planifolia[,"BioSample"], layout = "SE", genome = "vanilla_planifolia", ploidy = 2))
```

## Vigna radiata
```{r}
vigna_radiata = fread("../workflow/data/SRArunInfo/SraRunInfo_vigna_radiata_PRJNA885164.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = vigna_radiata[,"Run"], replicate = vigna_radiata[,"BioSample"], layout = "SE", genome = "vigna_radiata", ploidy = 2))

```

## Vigna umbellata
```{r}
vigna_umbellata = fread("../workflow/data/SRArunInfo/SraRunInfo_vigna_umbellata_PRJNA803965.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = vigna_umbellata[,"Run"], replicate = vigna_umbellata[,"BioSample"], layout = "PE", genome = "vigna_umbellata", ploidy = 2))
```

## Vigna unguiculata
```{r}
vigna_unguiculata = fread("../workflow/data/SRArunInfo/SraRunInfo_vigna_unguiculata_SRP077082.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = vigna_unguiculata[,"Run"], replicate = vigna_unguiculata[,"BioSample"], layout = "PE", genome = "vigna_unguiculata", ploidy = 2))

```

## Vitis vinifera
```{r}
vitis_vinifera = fread("../workflow/data/SRArunInfo/SraRunInfo_vitis_vinifera_PRJNA373967.csv.gz", data.table = F)

# subset WGS reads
vitis_vinifera = vitis_vinifera[(vitis_vinifera$LibraryStrategy == "WGS" & vitis_vinifera$ScientificName == "Vitis vinifera"),]

# check for technical replicates
any(duplicated(vitis_vinifera$Experiment))
any(duplicated(vitis_vinifera$SampleName))
any(duplicated(vitis_vinifera$BioSample))

# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(biosample in unique(vitis_vinifera$BioSample)){
#   techreps = vitis_vinifera[(vitis_vinifera$BioSample == biosample),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }
# 
# vitis_vinifera = vitis_vinifera[(vitis_vinifera$Run %in% runs),]

# add layout
# reads without mates are SE
vitis_vinifera$layout = NA
vitis_vinifera$layout[(vitis_vinifera$spots_with_mates > 0)] = "PE"
vitis_vinifera$layout[(vitis_vinifera$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = vitis_vinifera[,"Run"], replicate = vitis_vinifera[,"BioSample"], layout = vitis_vinifera[,"layout"], genome = "vitis_vinifera", ploidy = 2))
```

## Xanthoceras sorbifolium
```{r}
# load data
xanthoceras_sorbifolium = fread("../workflow/data/SRArunInfo/SraRunInfo_xanthoceras_sorbifolium_PRJNA790469.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = xanthoceras_sorbifolium[,"Run"], replicate = xanthoceras_sorbifolium[,"BioSample"], layout = "PE", genome = "xanthoceras_sorbifolium", ploidy = 2))
```

## Zea mays
```{r}
# load data
zea_mays_1 = fread("../workflow/data/SRArunInfo/SraRunInfo_zea_mays_PRJNA260788.csv.gz", data.table = F)
zea_mays_2 = fread("../workflow/data/SRArunInfo/SraRunInfo_zea_mays_PRJNA389800.csv.gz", data.table = F)
zea_mays_3 = fread("../workflow/data/SRArunInfo/SraRunInfo_zea_mays_PRJNA399729.csv.gz", data.table = F)

# concatenate
zea_mays = rbind(zea_mays_1, zea_mays_2, zea_mays_3)

# for each technical replicate, just take the sample with the most bases
# runs = NULL
# for(biosample in unique(zea_mays$BioSample)){
#   techreps = zea_mays[(zea_mays$BioSample == biosample),]
#   techreps = techreps[order(techreps$bases, decreasing = T),]
#   runs = c(runs, techreps[1,"Run"])
# }
# 
# zea_mays = zea_mays[(zea_mays$Run %in% runs),]

# confirm no duplicates
any(duplicated(zea_mays$Experiment))
any(duplicated(zea_mays$SampleName))
any(duplicated(zea_mays$BioSample))

# add layout
# reads without mates are SE
zea_mays$layout = NA
zea_mays$layout[(zea_mays$spots_with_mates > 0)] = "PE"
zea_mays$layout[(zea_mays$spots_with_mates == 0)] = "SE"

# add reads to growing list
allReads = rbind(allReads, data.frame(run = zea_mays[,"Run"], replicate = zea_mays[,"BioSample"], layout = zea_mays[,"layout"], genome = "zea_mays", ploidy = 2))
```

## Zizania palustris
```{r}
# load data
zizania_palustris = fread("../workflow/data/SRArunInfo/SraRunInfo_zizania_palustris_PRJNA774842.csv.gz", data.table = F)

# subset just Z. palustris
unique(zizania_palustris$ScientificName)
zizania_palustris = zizania_palustris[(zizania_palustris$ScientificName == "Zizania palustris"),]

# add reads to growing list
allReads = rbind(allReads, data.frame(run = zizania_palustris[,"Run"], replicate = zizania_palustris[,"BioSample"], layout = "SE", genome = "zizania_palustris", ploidy = 2))
```

## Ziziphus jujuba
```{r}
# load data
ziziphus_jujuba = fread("../workflow/data/SRArunInfo/SraRunInfo_ziziphus_jujuba_PRJNA560664.csv.gz", data.table = F)

# add reads to growing list
allReads = rbind(allReads, data.frame(run = ziziphus_jujuba[,"Run"], replicate = ziziphus_jujuba[,"BioSample"], layout = "PE", genome = "ziziphus_jujuba", ploidy = 2))
```

# Summarize read counts
```{r}
# breakdown of reads per species
# write.table(table(allReads$genome, allReads$layout), "breakdownOfReadSets")
readBreakdown = table(allReads$genome, allReads$layout)
print(readBreakdown)

# number of unique genotypes per species
countUnique = function(x){
  length(unique(x))
}
indvBreakdown = aggregate(replicate ~ genome, data = allReads, FUN = countUnique)
print(indvBreakdown)

# number of species included thus far
length(unique(allReads$genome))

# number of unique genotypes
length(unique(allReads$replicate))

# total number of read sets
nrow(allReads)

# save table of breakdown
# write.csv(cbind(readBreakdown, indvBreakdown[,-1]), "breakdownOfReadSetsByGenome.csv")

# remove intermediate objects
rm(arabidopsis_lyrata_1, arabidopsis_lyrata_2, arabidopsis_lyrata_3, brassica, boechera_stricta_1, boechera_stricta_2, glycine_max_1, glycine_max_2, glycine_max_3, oryza, pisum_sativum_1, pisum_sativum_2, triticum_aestivum_1, triticum_aestivum_2, zea_mays_1, zea_mays_2, zea_mays_3, musa, malus, arachis_1, arachis_2, arachis_3, arachis, prunus_persica_supp, indvBreakdown, readBreakdown, samples, runs2drop, dioscorea_alata_ploidy, sraprojects, countUnique, ids)

# Estimate number of basepairs included in dataset
# bind all metadata together
# paste(ls(), collapse = ", ")
runsAllMetaData = list(actinidia_chinensis, aegilops_tauschii, amaranthus_hypochondriacus, ananas_comosus, arabidopsis_halleri, arabidopsis_lyrata, arabidopsis_suecica, arabidopsis_thaliana, arabis_alpina, arabis_nemorensis, arachis_duranensis, arachis_hypogaea, arachis_ipaensis, benincasa_hispida, beta_vulgaris, boechera_stricta, brachypodium_distachyon, brassica_juncea, brassica_napus, brassica_nigra, brassica_oleracea, brassica_rapa, buddleja_alternifolia, cajanus_cajan, camelina_sativa, camellia_sinensis, cannabis_sativa, capsella_grandiflora, capsella_rubella, capsicum_annuum, castanea_mollissima, chenopodium_quinoa, cicer_arietinum, citrullus_lanatus, coffea_arabica, coffea_canephora, corylus_avellana, cucumis_melo, cucumis_sativus, cucurbita_argyrosperma, cucurbita_pepo, digitaria_exilis, dioscorea_alata, dioscorea_rotundata, durio_zibethinus, elaeis_guineensis, eleusine_coracana, ficus_carica, glycine_max, glycine_soja, gossypium_arboreum, gossypium_barbadense, gossypium_hirsutum, helianthus_annuus, heliosperma_pusillum, hordeum_vulgare, juglans_regia, lactuca_sativa, lens_culinaris, lens_ervoides, linum_usitatissimum, lotus_japonicus, lupinus_angustifolius, macadamia_integrifolia, malus_domestica, malus_sylvestris, mangifera_indica, manihot_esculenta, medicago_truncatula, mimulus_guttatus, miscanthus_sinensis, momordica_charantia, musa_acuminata, nelumbo_nucifera, nicotiana_tabacum, olea_europaea, oryza_barthii, oryza_brachyantha, oryza_glaberrima, oryza_glumipatula, oryza_longistaminata, oryza_meridionalis, oryza_punctata, oryza_rufipogon, oryza_sativa, panicum_halli, panicum_virgatum, papaver_somniferum, phaseolus_vulgaris, phoenix_dactylifera, pisum_sativum, populus_deltoides, populus_trichocarpa, prunus_armeniaca, prunus_avium, prunus_persica, quercus_lobata, quercus_robur, raphanus_sativus, rhododendron_griersonianum, salix_dunnii, secale_cereale, sesamum_indicum, setaria_italica, setaria_viridis, solanum_chilense, solanum_lycopersicum, solanum_melongena, solanum_pimpinellifolium, solanum_stenotomum, sorghum_bicolor, spinacia_oleracea, spirodela_polyrhiza, striga_hermonthica, tetracentron_sinense, thinopyrum_intermedium, thlaspi_arvense, trifolium_pratense, triticum_aestivum, triticum_dicoccoides, triticum_turgidum, vanilla_planifolia, vigna_radiata, vigna_umbellata, vigna_unguiculata, vitis_vinifera, xanthoceras_sorbifolium, zea_mays, zizania_palustris, ziziphus_jujuba)

sumColumnInList = function(x){
  sum(as.numeric(x[,"bases"]))
}

sum(unlist(lapply(runsAllMetaData, sumColumnInList)))
```


# Loop over replicates, if there are both SE and PE reads for a single sample, just use the PE reads
```{r}
# I could either throw out read 2 of the PE reads and treat all data as single end, or I could retain just paired-end data... Not sure which is better
# I'll throw out the single-end reads because the paired-end data will align better
allReadsFinal = NULL

for(replicate in unique(allReads$replicate)){
  dataSubset = allReads[(allReads$replicate == replicate),]
  if(all(c("PE", "SE") %in% dataSubset[,"layout"])){
    dataSubset = dataSubset[(dataSubset$layout == "PE"),]
  }
  allReadsFinal = rbind(allReadsFinal, dataSubset)
}

# breakdown of PE vs SE for each genome
table(allReadsFinal$genome, allReadsFinal$layout)

# number of unique genotypes included
length(unique(allReadsFinal$replicate))

# number of read sets
nrow(allReadsFinal)

# number of genomes
length(unique(allReadsFinal$genome))
```

# output read metadata for snakemake input
```{r}
# test dataset for merging reads into single datasets
#mergeTestSet = allReads[(allReads$replicate %in% c("SAMEA2567517", "SAMEA2567535", "SAMEA2567633", "SAMEA2567634", "SAMN03005107", "SAMN03005132")),]
#write.table(mergeTestSet, file = "samples.tsv", sep = "\t", quote = F, row.names = F)

# analyze just a couple species as an example
testSet = allReads[(allReads$genome %in% c("arabidopsis_halleri", "arabidopsis_lyrata", "arabidopsis_thaliana", "arabidopsis_suecica", "boechera_stricta", "brachypodium_distachyon", "brassica_oleracea", "brassica_rapa", "capsella_grandiflora", "capsella_rubella")),]
write.table(testSet, file = "samples.tsv", sep = "\t", quote = F, row.names = F)

write.table(allReads, file = "../workflow/data/samples.tsv", sep = "\t", quote = F, row.names = F)

# write run numbers and metadata to workflow input
# need to replace write path with "../workflow/data/samples_pe.tsv" when I'm ready
#write.table(seReads, file = "samples_se.tsv", sep = "\t", quote = F, row.names = F)
#write.table(allReads, file = "samples_pe.tsv", sep = "\t", quote = F, row.names = F)

```
















# Manually check top few bioprojects for each species to see if they're WGS data
```{r}

```

# Split SRA run data into single-end and paired end
## Arabidopsis thaliana
```{r}
athalSraRunInfo = fread("../workflow/data/SraRunInfo_arabidopsis_thaliana.csv.gz", data.table = F)

# separate paired end and single-end data
# if spot_with_mates == 0 in sra run info, then data is single-end
athalSe = athalSraRunInfo[(athalSraRunInfo$spots_with_mates == 0),"Run"]
athalPe = athalSraRunInfo[(athalSraRunInfo$spots_with_mates != 0), "Run"]

# add assembly name to file
# can subset data here to get toy dataset
athalSe = data.frame(samples = athalSe, genome = "arabidopsis_thaliana")
athalPe = data.frame(samples = athalPe, genome = "arabidopsis_thaliana")

# write single-end and paired-end samples to separate files
write.table(athalSe, file = "../workflow/data/samples_se.tsv", sep = "\t", quote = F, row.names = F)
write.table(athalPe, file = "../workflow/data/samples_pe.tsv", sep = "\t", quote = F, row.names = F)
```