---
title: "K-mer approach to Lewontin's paradox results"
author: "Miles Roberts"
date: "2024-01-17"
output:
  html_document:
    df_print: paged
  pdf_document: default
---

# load packages
```{r}
rm(list = ls())

# remove.packages(c("dplyr", "ggplot2", "caper", "stringr", "data.table", "cowplot", "rWCVP", "lwgeom", "sf"))

library(dplyr)
library(ggplot2)
library(caper)
library(stringr)
library(data.table)
library(cowplot)
#library(scico)
#library(giscoR)
library(rWCVP)
library(lwgeom)
library(sf)
library(ggpubr)


today = Sys.Date()
```

# Data preparation

## Prepare diversity data 
```{r}
# List input files
pi = list.files("../results/2023-03-26/genome-wide-pi", full.names = T)
bcd = list.files("../results/2023-03-26/bray-curtis-dissimilarities", full.names = T)
jac = list.files("../results/2023-03-26/jaccard-dissimilarities", full.names = T)

# for now, remove species that can't be processed further for whatever reason
# pi: zizania palustris, panicum virgatum
# jac and bcd: panicum hallii, lens culinaris
jac = jac[c(-52, -76)]
bcd = bcd[c(-52, -76)]
pi = pi[c(-75,-103)]

# get species IDs
species = gsub("_pairwise-pi.txt", "", gsub("../results/2023-03-26/genome-wide-pi/", "", pi))

# Load input files
pi = lapply(pi, fread, header = T, data.table = F)
bcd = lapply(bcd, fread, header = F, data.table = F)
jac = lapply(jac, fread, header = F, data.table = F)

# calculate mean pairwise dissimilarity
bcdMean = lapply(bcd, function(x){mean(x[,2])})
jacMean = lapply(jac, function(x){mean(x[,2])})

# calculate average number of k-mers included in union
unionMean = lapply(jac, function(x){mean(x[,4])})

# calculate number of individuals sampled for each dataset
# You get this equation by rewriting the formula for the number of pairwise comparisons (n^2 - n)/2 = N as a polynomial and applying the quadratic formula
backCalcN = function(x){
  sqrt(1/4 + 2*nrow(x)) + 1/2
}

nidv = lapply(bcd, backCalcN)

# cbind all diversity data into one dataframe per species
alldiv = Map(cbind, pi, bcdMean, jacMean, unionMean, nidv)

# Some species have extra data, remove for now
speciesWithExtraData = which(unlist(lapply(alldiv, ncol)) == 12)
for(i in speciesWithExtraData){
  alldiv[[i]] = alldiv[[i]][,c(1, 3:6, 9:12)]
}

# rbind all species data into one dataframe
alldiv = lapply(alldiv, setNames, c("h", "ntotal", "nvariant", "ninvariant", "pi", "bcd", "jac", "umean", "nidv"))

alldiv = do.call(rbind, alldiv)

# add species names
alldiv$species = species
alldiv$species = str_to_sentence(gsub("_", " ", alldiv$species))

# For now, remove species where invariant sites were not properly called
# alldiv = alldiv[(alldiv$ninvariant > 0),]

# For now, remove species where variants sites were not properly called
# alldiv = alldiv[(alldiv$nvariant > 30),]

# remove species where there aren't many sampled individuals
# alldiv = alldiv[(alldiv$nidv >= 10),]

# add coverage covariate
coverage = read.table("../results/2023-03-26/bp_sequenced_2024-01-31.txt", header = T, sep = "\t")
alldiv = merge(alldiv, coverage, by = "species", all.x = TRUE)

# 
# gsize = read.table("../results/2023-03-26/genome_size.txt", header = T, sep = "\t")
# alldiv = merge(alldiv, gsize, by = "species")
# 
# alldiv$totalcov = alldiv$totalbp/alldiv$assemblySize
```

## Load range data
```{r}
# load range data
areas = list.files("../results/2023-03-26/ranges/areas", full.names = T)

# remove files with no data
areas = areas[c(-33, -44, -96, -106)]

# extract species names
areas_species = gsub("_areas.txt", "", gsub("../results/2023-03-26/ranges/areas/", "", areas))
areas_species = str_to_sentence(gsub("_", " ", areas_species))

# load data
areas = lapply(areas, read.table, header = +T, sep = "\t")

# add names to list
names(areas) = areas_species

#str(areas)
```

## remove areas outside of native range 
```{r}
# Refer to plants of the world online for native range maps
# powo.science.kew.org

# Amaranthus hybridus native to only North and South America
areas$`Amaranthus hypochondriacus` = areas$`Amaranthus hypochondriacus`[1:2,]

# Arabidopsis thaliana is only native to Europe, Asia, and Africa
areas$`Arabidopsis thaliana` = areas$`Arabidopsis thaliana`[2:4,]

# Arachis duranensis is native to south america, remove asia
areas$`Arachis duranensis` = areas$`Arachis duranensis`[1,]

# Arachis glabrata is native to Brazil, remove North America
areas$`Arachis hypogaea` = areas$`Arachis hypogaea`[2,]

# Beta vulgaris subsp. vulgaris is native to only Europe and Africa
areas$`Beta vulgaris` = areas$`Beta vulgaris`[2:3,]

# Brachypodium distachyon is native to Europe, Asia, Africa
areas$`Brachypodium distachyon` = areas$`Brachypodium distachyon`[2:4,]

# Brassica juncea is native only to Europe
areas$`Brassica juncea` = areas$`Brassica juncea`[4,]

# Brassica cretica native to europe
areas$`Brassica oleracea` = areas$`Brassica oleracea`[1,]

# Brassica fruticulosa native to europe
areas$`Brassica rapa` = areas$`Brassica rapa`[2,]

# Buddleja alternifolia is native to Asia
areas$`Buddleja alternifolia` = areas$`Buddleja alternifolia`[2,]

# Cajanus scarabaeoides native to Africa, Asia, Australia
areas$`Cajanus cajan` = areas$`Cajanus cajan`[c(1,3,4),]

# Camelina sativa is not native to North America
areas$`Camelina sativa` = areas$`Camelina sativa`[c(2,3),]

# Cannabis sativa is native only to Asia
areas$`Cannabis sativa` = areas$`Cannabis sativa`[5,]

# Capsicum annuum is native to north america and south america
areas$`Capsicum annuum` = areas$`Capsicum annuum`[c(1,2),]

# Castanea mollissima is native only to asia
areas$`Castanea mollissima` = areas$`Castanea mollissima`[2,]

# Cucumis myriocarpus is native only to africa
areas$`Cucumis melo` = areas$`Cucumis melo`[2,]

# Cucurbita palmata is only native to North America
areas$`Cucurbita pepo` = areas$`Cucurbita pepo`[1,]

# Digitaria longiflora native to africa, asia, australia
areas$`Digitaria exilis`= areas$`Digitaria exilis`[2:4,]

# Dioscorea communis is native only to africa
areas$`Dioscorea rotundata` = areas$`Dioscorea rotundata`[2,]

# Gossypium arboreum native to asia
areas$`Gossypium arboreum` = areas$`Gossypium arboreum`[3,]

# Gossypium thurberi is native only to north america
areas$`Gossypium hirsutum` = areas$`Gossypium hirsutum`[1,]

# Helianthus maximiliani is native only to north america
areas$`Helianthus annuus` = areas$`Helianthus annuus`[1,]

# Juglans mandshuica is native only to asia
areas$`Juglans regia` = areas$`Juglans regia`[2,]

# Lactuca sativa is native to Iraq
areas$`Lactuca sativa` = areas$`Lactuca sativa`[3,]

# Linum bienne is native to europe, africa, and asia
areas$`Linum usitatissimum` = areas$`Linum usitatissimum`[c(3,4),]

# Malus sylvestris is only native to Europe
areas$`Malus sylvestris` = areas$`Malus sylvestris`[2,]

# Medicago truncatula only native to europe and africa
areas$`Medicago truncatula` = areas$`Medicago truncatula`[c(4,5),]

# Mimulus guttatus only native to North America
areas$`Mimulus guttatus` = areas$`Mimulus guttatus`[1,]

# Miscanthus sacchariflorus is native only to asia
areas$`Miscanthus sinensis` = areas$`Miscanthus sinensis`[3,]

# Musa relative only native to asia and india
areas$`Musa acuminata` = areas$`Musa acuminata`[2,]

# Nicotiana glauca is native only to south america
areas$`Nicotiana tabacum` = areas$`Nicotiana tabacum`[2,]

# Oryza rufipogon is native to asia and australia
areas$`Oryza rufipogon` = areas$`Oryza rufipogon`[c(2,3),]

# Panicum hallii is native only to north america
areas$`Panicum hallii` = areas$`Panicum hallii`[1,]

# Panicum capillare is native only to north america
areas$`Panicum virgatum` = areas$`Panicum virgatum`[1,]

# Papaver rhoeas is native to Africa, Europe, Asia
areas$`Papaver somniferum` = areas$`Papaver somniferum`[4:6,]

# Populus deltoides only native to north america
areas$`Populus deltoides` = areas$`Populus deltoides`[1,]

# Prunus avium is native to europe, asia, africa
areas$`Prunus avium` = areas$`Prunus avium`[2:3,]

# Raphanus raphanistrum is native to europe and africa
areas$`Raphanus sativus` = areas$`Raphanus sativus`[3:4,]

# Setaria faberi is native only to asia
areas$`Setaria italica` = areas$`Setaria italica`[3,]

# Setaria viridis is native to europe, africa, asia, australia
areas$`Setaria viridis` = areas$`Setaria viridis`[2:5,]

# Thlaspi arvense is native to europe and asia
areas$`Thlaspi arvense` = areas$`Thlaspi arvense`[3:4,]

# Triticum urartu is native only to asia 
areas$`Triticum aestivum` = areas$`Triticum aestivum`[2,]
```

## Merge range data with diversity data
```{r}
# function to get non-water area
get_nonwater_area = function(x){
  sum(x[,"total_area"] - x[,"water_area"])
}

# format area data as frame
areas = data.frame(
  species = areas_species,
  gbif.area = unlist(lapply(areas, get_nonwater_area))
)

# merge with diversity data
alldiv = merge(alldiv, areas, by = "species", all.x = T)

print(alldiv)

# save final table
#write.csv(alldiv, paste("../results/2023-03-26/diversityAndRangeData_", today, ".csv", sep = ""), row.names = F, quote = F)
```

## Prepare phylogenetic tree
```{r}
# write species list to upload to timetree.org, converting some names to those recognized by timetree
timetree_names = alldiv$species
timetree_names[(timetree_names == "Brassica rapa")] = "Brassica rapa subsp. rapa"
timetree_names[(timetree_names == "Brassica oleracea")] = "Brassica oleracea var. capitata"
timetree_names[(timetree_names == "Solanum stenotomum")] = "Solanum tuberosum"
timetree_names[(timetree_names == "Chenopodium quinoa")] = "Chenopodium album"

#write.table(timetree_names, paste("../results/2023-03-26/species_list_", today, ".txt", sep = ""), row.names = F, col.names = F, quote = F)

# load time tree from timetree.org
phy = read.tree("../results/2023-03-26/species_list_2024-02-02.nwk")

# remove underscores
phy$tip.label = gsub("_", " ", phy$tip.label)

# remove variety and subspecies names, convert tip labels to my standard
phy$tip.label = gsub(" var\\..*", "", phy$tip.label)
phy$tip.label = gsub(" subsp\\..*", "", phy$tip.label)

phy$tip.label[(phy$tip.label == "Solanum tuberosum")] = "Solanum stenotomum"
phy$tip.label[(phy$tip.label == "Chenopodium album")] = "Chenopodium quinoa"
phy$tip.label[(phy$tip.label == "Erythranthe guttata")] = "Mimulus guttatus"

# sanity check
plot(phy)

# check that all species I have data for are in phylogeny and vice versa. Output should be TRUE
all(alldiv$species %in% phy$tip.label)
all(phy$tip.label %in% alldiv$species)
#alldiv$species[!(alldiv$species %in% phy$tip.label)]
```

## Load range areas
```{r}
# load local copies of World Checklist of Vascular Plants
wcvp_dist = fread("../workflow/data/range/wcvp_distribution.csv.gz", sep = "|", header = T, quote = "", fill = T, encoding = "UTF-8")

wcvp_names = fread("../workflow/data/range/wcvp_names.csv.gz", sep = "|", header = T, quote = "", fill = T, encoding = "UTF-8")

# update species names to match latest names in WCVP
updated_species_names = data.frame(species = unique(alldiv$species),species_updated = unique(alldiv$species))

updated_species_names$species_updated[updated_species_names$species == "Arabis nemorensis"] = "Arabis planisiliqua"
updated_species_names$species_updated[updated_species_names$species == "Dioscorea rotundata"] = "Dioscorea cayenensis subsp. rotundata"
updated_species_names$species_updated[updated_species_names$species == "Glycine soja"] = "Glycine max subsp. soja"
updated_species_names$species_updated[updated_species_names$species == "Lens ervoides"] = "Vicia lenticula"
updated_species_names$species_updated[updated_species_names$species == "Mimulus guttatus"] = "Erythranthe guttata"
updated_species_names$species_updated[updated_species_names$species == "Populus trichocarpa"] = "Populus tristis"
updated_species_names$species_updated[updated_species_names$species == "Raphanus sativus"] = "Raphanus raphanistrum subsp. sativus"
updated_species_names$species_updated[updated_species_names$species == "Salix dunnii"] = "Salix tetrasperma"
updated_species_names$species_updated[updated_species_names$species == "Solanum stenotomum"] = "Solanum tuberosum"
updated_species_names$species_updated[updated_species_names$species == "Boechera stricta"] = "Boechera drummondii"

# loop over each species, and calculate range area in square kilometers
region_areas = NULL

for(species in updated_species_names$species_updated){

  print(species)
  
  # get native distribution
species_dist = wcvp_distribution(
taxon = species,
taxon_rank = "species",
native = TRUE,
introduced = FALSE,
extinct = TRUE,
location_doubtful = FALSE,
wcvp_distributions = wcvp_dist,
wcvp_names = wcvp_names
)

# calculate area, convert to square kilometers (1000 x 1000 meters)
region_area = sum(st_area(species_dist))/1e6

region_areas = rbind(region_areas, data.frame(species_updated = species, area = region_area))

}

# merge range areas with old species names
region_areas = merge(updated_species_names, region_areas, by = "species_updated")

# remove weird units
region_areas$area = as.numeric(region_areas$area)

# merge with diversity data
alldiv = merge(alldiv, region_areas, by = "species")
```

## Load plant height data, a proxy for body size, which is a proxy for population density
```{r}
height = fread("../workflow/data/eol_plant_height_data_2023-10-03/records.tsv.gz", header = T, fill = T, quote = "", sep = "\t")

# convert feet to meters
height$final_values = height$`Measurement Value`
height$final_values[(height$`Measurement Unit` == "feet")] = height$final_values[(height$`Measurement Unit` == "feet")]*0.3048

# get height measurements for only species in my dataset
height_my_species = NULL
for(i in 1:nrow(alldiv)){
  # print progress
  print(i)
  
  # get species name(s)
  old_name = alldiv$species[i]
  new_name = alldiv$species_updated[i]
  
  # check old name
  height_sub = height[grepl(old_name, height$`Scientific Name`),]
  
  # if new name is different, check new species name
  if(new_name != old_name){
    height_sub = rbind(height_sub, height[grepl(new_name, height$`Scientific Name`),])
  }
  
  # average across values
  height_one_species = data.frame(species = old_name, height = mean(height_sub$final_values, na.rm = T))
  height_my_species = rbind(height_my_species, height_one_species)
}

# add height data to final dataframe
summary(height_my_species$height)
alldiv = merge(alldiv, height_my_species, all.x = T, by = "species")
```

## genome size, ploidy, mating system data
```{r}
# load extra species metadata
metadata = read.table("../results/2023-03-26/species_metadata_2024-01-31.txt", sep = "\t", fill = T, quote = "", header = T)
names(metadata)[1] = "species"

# remove weird space character
metadata$species = gsub("<a0>", " ", metadata$species)
metadata$species = gsub(" $", "", metadata$species)

# merge metadata with diversity data
alldiv = merge(alldiv, metadata[,c("species", "DNA.amount..1.C..pg.", "Mating.system", "Ploidy", "Height..m.", "cultivation.status")], by = "species", all.x = T)

# combine different height columns into one
alldiv$height_final = alldiv$height
alldiv$height_final[is.na(alldiv$height_final)] = alldiv$Height..m.[is.na(alldiv$height_final)]

# calculate quantity that's proportional to population size (range/squared height)
alldiv$WCVP.popsize = (alldiv$area * 1e6)/(alldiv$height_final^2)
summary(alldiv$WCVP.popsize)

alldiv$GBIF.popsize = (alldiv$gbif.area * 1e6)/(alldiv$height_final^2)
summary(alldiv$GBIF.popsize)

# convert genome size in picograms to megabases
alldiv$genome.size = alldiv$DNA.amount..1.C..pg. * 0.978e9

# calculate average coverage per individual
alldiv$meancov = alldiv$totalbp/(alldiv$genome.size*alldiv$nidv)

# re-order mating system variable
alldiv$Mating.system = factor(alldiv$Mating.system, levels = c("selfing", "mixed", "outcrossing"))
```

## save final organized dataframe
```{r eval=FALSE, include=FALSE}
write.table(alldiv, paste("final_covariates_", today, ".txt", sep = ""), row.names = F, quote = F, sep = "\t")
```

# Initial data analysis

## load saved data for plotting
```{r eval=FALSE, include=FALSE}
alldiv = read.table("../results/2023-03-26/final_covariates_2023-11-06.txt", sep = "\t", header = T)
```

## relationship between coverage and diveristy

```{r}
# scico(2, palette = "lapaz")
# "#190C64" "#FEF2F2"

# function for simple scatterplot
my_ggplot_wrap <- function(data, xcol, ycol, xname, yname, filename, filew = 7, fileh = 7) {
  myplot = ggplot(data = data, aes_string(x = xcol, y = ycol)) + 
    geom_point() +
    geom_smooth(method = "loess", color = "pink") +
    theme_classic() +
    stat_cor(method = "spearman", digits = 2, cor.coef.name = "rho") +
    labs(x = xname, y = yname)
  
  myplot
  
  ggsave(filename, width = filew, height = fileh)

  return(myplot)
}

# function to color points by third variable
my_ggplot_wrap_xyz <- function(data, xcol, ycol, xname, yname, filename, zcol, lcol, hcol, lname, mylims, filew = 9, fileh = 7) {
  myplot = ggplot(data = data, aes_string(x = xcol, y = ycol, color = zcol)) + 
    geom_point() +
    geom_smooth(method = "loess", color = "pink") +
    theme_classic() +
    stat_cor(method = "spearman", digits = 2, cor.coef.name = "rho") +
    scale_color_gradient(low = lcol, high = hcol, name = lname, limits = mylims) +
    labs(x = xname, y = yname)
  
  myplot
  
  ggsave(filename, width = filew, height = fileh)

  return(myplot)
}

# function to create boxplots
my_ggplot_wrap_box <- function(data, xcol, ycol, xname, yname, filename, filew = 7, fileh = 7) {
  myplot = ggplot(data = data, aes_string(x = xcol, y = ycol)) + 
    geom_boxplot() +
    theme_classic() +
    stat_kruskal_test() +
    #stat_anova_test(type = 3, method = "one_way") +
    labs(x = xname, y = yname)
  
  myplot
  
  ggsave(filename, width = filew, height = fileh)

  return(myplot)
}

# function to have scatterplot with 1:1 line
my_ggplot_wrap_line <- function(data, xcol, ycol, xname, yname, filename, filew = 7, fileh = 7) {
  myplot = ggplot(data = data, aes_string(x = xcol, y = ycol)) + 
    geom_point() +
    geom_abline(slope = 1, intercept = 0) +
    theme_classic() +
    stat_cor(method = "spearman", digits = 2, cor.coef.name = "rho") +
    labs(x = xname, y = yname)
  
  myplot
  
  ggsave(filename, width = filew, height = fileh)

  return(myplot)
}

# scatterplots
ida_meancov_vs_pi = my_ggplot_wrap(alldiv, "log10(meancov)", "log10(pi)", "log10(mean coverage per biosample)", "log10(Nucleotide diversity)", paste("ida_coverage-vs-pi_", today, ".png", sep = ""))

ida_meancov_vs_jac = my_ggplot_wrap(alldiv, "log10(meancov)", "jac", "log10(mean coverage per biosample)", "Jaccard dissimilarity", paste("ida_coverage-vs-jac_", today, ".png", sep = ""))

ida_meancov_vs_bcd = my_ggplot_wrap(alldiv, "log10(meancov)", "bcd", "log10(mean coverage per biosample)", "Bray-Curtis dissimilarity", paste("ida_coverage-vs-bcd_", today, ".png", sep = ""))

ida_cvbp_vs_pi = my_ggplot_wrap(alldiv, "cvbp", "log10(pi)", "Coefficient of variation in bp sequenced", "log10(Nucleotide diversity)", paste("ida_cvbp-vs-pi_", today, ".png", sep = ""))

ida_cvbp_vs_jac = my_ggplot_wrap(alldiv, "cvbp", "jac", "Coefficient of variation in bp sequenced", "Jaccard dissimilarity", paste("ida_cvbp-vs-jac_", today, ".png", sep = ""))

ida_cvbp_vc_bcd = my_ggplot_wrap(alldiv, "cvbp", "bcd", "Coefficient of variation in bp sequenced", "Bray-Curtis dissimilarity", paste("ida_cvbp-vs-bcd_", today, ".png", sep = ""))

ida_jac_vs_bcd = my_ggplot_wrap(alldiv, "jac", "bcd", "Jaccard Dissimilarity", "Bray-Curtis dissimilarity", paste("ida_jac-vs-bcd_", today, ".png", sep = ""))

ida_pi_vs_jac = my_ggplot_wrap_xyz(alldiv, "log10(pi)", "jac", "log10(Nucleotide diversity)", "Jaccard dissimilarity", paste("ida_pi-vs-jac_", today, ".png", sep = ""), "log10(meancov)", "#FEF2F2", "#190C64", "log10(mean cov)", c(min(log10(alldiv$meancov)), max(log10(alldiv$meancov))))

ida_pi_vs_bcd = my_ggplot_wrap_xyz(alldiv, "log10(pi)", "bcd", "log10(Nucleotide diversity)", "Bray-Curtis dissimilarity", paste("ida_pi-vs-bcd_", today, ".png", sep = ""), "log10(meancov)", "#FEF2F2", "#190C64", "log10(mean cov)", c(min(log10(alldiv$meancov)), max(log10(alldiv$meancov))))

my_ggplot_wrap_xyz(alldiv, "jac", "bcd", "Jaccard dissimilarity", "Bray-Curtis dissimilarity", paste("ida_jac-vs-bcd_", today, ".png", sep = ""), "log10(meancov)", "#FEF2F2", "#190C64", "log10(mean cov/sample)", c(min(log10(alldiv$meancov)), max(log10(alldiv$meancov))))

ida_wcvp_vs_gbif = my_ggplot_wrap_line(alldiv, "log10(area)", "log10(gbif.area)", "log10(WCVP range size in square km)", "log10(GBIF range size in square km)", paste("ida_wcvp-vs-gbif_", today, ".png", sep = ""))

# plot diversity by mating system
ida_mating_vs_pi = my_ggplot_wrap_box(alldiv, "Mating.system", "log10(pi)", "Mating system", "log10(Nucleotide diversity)", paste("mating_vs_pi_", today, ".png", sep = ""))

ida_mating_vs_jac = my_ggplot_wrap_box(alldiv, "Mating.system", "jac", "Mating system", "Jaccard dissimilarity", paste("mating_vs_jac_", today, ".png", sep = ""))

ida_mating_vs_bcd = my_ggplot_wrap_box(alldiv, "Mating.system", "bcd", "Mating system", "Bray-Curtis dissimilarity", paste("mating_vs_bcd_", today, ".png", sep = ""))

# plot diversity by ploidy
ida_ploidy_vs_pi = my_ggplot_wrap_box(alldiv, "Ploidy", "log10(pi)", "Ploidy", "log10(Nucleotide diversity)", paste("ploidy_vs_pi_", today, ".png", sep = ""))

ida_ploidy_vs_jac = my_ggplot_wrap_box(alldiv, "Ploidy", "jac", "Ploidy", "Jaccard dissimilarity", paste("ploidy_vs_jac_", today, ".png", sep = ""))

ida_ploidy_vs_bcd = my_ggplot_wrap_box(alldiv, "Ploidy", "bcd", "Ploidy", "Bray-Curtis dissimilarity", paste("ploidy_vs_bcd_", today, ".png", sep = ""))

# plot diversity by cultivation status
ida_cult_vs_pi = my_ggplot_wrap_box(alldiv, "cultivation.status", "log10(pi)", "Cultivation status", "log10(Nucleotide diversity)", paste("cultivation_vs_pi_", today, ".png", sep = ""))

ida_cult_vs_jac = my_ggplot_wrap_box(alldiv, "cultivation.status", "jac", "Cultivation status", "Jaccard dissimilarity", paste("cultivation_vs_jac_", today, ".png", sep = ""))

ida_cult_vs_bcd = my_ggplot_wrap_box(alldiv, "cultivation.status", "bcd", "Cultivation status", "Bray-Curtis dissimilarity", paste("cultivation_vs_bcd_", today, ".png", sep = ""))

# plot height for each species, from tallest to shortest
species_order = alldiv$species[sort(alldiv$height_final, decreasing = T, index.return = T)$ix]

ggplot(alldiv, aes(x = species, y = height_final)) +
  geom_bar(stat = "identity") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  scale_x_discrete(limits = species_order) + 
  labs(x = "Species", y = "Height (m)")

ggsave(paste("ida_height-by-species_", today, ".png", sep = ""), width = 16, height = 4.5, units = "in")
```


## distributions of variables
```{r eval=FALSE, include=FALSE}
# histograms of diversity data
ggplot(alldiv, aes(x = log10(pi))) +
  geom_histogram() +
  theme_classic() +
  labs(x = "log10(Nucleotide diversity)")

ggplot(alldiv, aes(x = jac)) +
  geom_histogram() +
  theme_classic() +
  labs(x = "Jaccard Dissimilarity")

ggplot(alldiv, aes(x = bcd)) +
  geom_histogram() +
  theme_classic() +
  labs(x = "Bray-Curtis Dissimilarity")
```

## correlations between variables
```{r}
# Does pi correlate with k-mer diversity?
cor(alldiv$pi, alldiv$bcd, method = "spearman", use = "pairwise.complete.obs")
cor(alldiv$pi, alldiv$jac, method = "spearman", use = "pairwise.complete.obs")

# Does k-mer diversity correlate with population size (i.e. range area) more than pi?
cor(alldiv$pi, alldiv$rangeArea, method = "spearman", use = "pairwise.complete.obs")
cor(alldiv$bcd, alldiv$rangeArea, method = "spearman", use = "pairwise.complete.obs")
cor(alldiv$jac, alldiv$rangeArea, method = "spearman", use = "pairwise.complete.obs")

cor(alldiv$pi, alldiv$area, method = "spearman", use = "pairwise.complete.obs")
cor(alldiv$bcd, alldiv$area, method = "spearman", use = "pairwise.complete.obs")
cor(alldiv$jac, alldiv$area, method = "spearman", use = "pairwise.complete.obs")

# How does the number of individuals correlate with the diversity measures
# More individuals = more opportunities to discover variants
cor(alldiv$pi, alldiv$nidv)
cor(alldiv$bcd, alldiv$nidv)
cor(alldiv$jac, alldiv$nidv)

# Correlations between population size and diversity, not-corrected for phylogeny
cor(alldiv$pi, alldiv$popsize, method = "spearman", use = "pairwise.complete.obs")
cor(alldiv$bcd, alldiv$popsize, method = "spearman", use = "pairwise.complete.obs")
```

# Confirmatory data analysis

## remove outliers before analysis
```{r}
# For now, remove species where invariant sites were not properly called
#alldiv = alldiv[(alldiv$ninvariant > 0),]

# For now, remove species where variants sites were not properly called
#alldiv = alldiv[(alldiv$nvariant > 100),]

# remove critically endangered species
# alldiv = alldiv[(alldiv$rangeArea > 10),]

# remove species where there aren't many sampled individuals
#alldiv = alldiv[(alldiv$nidv >= 5),]

# remove any rows that are NA for some reason
#alldiv = alldiv[!is.na(alldiv$species),]

# remove all species with <1x coverage on average across individuals
subdiv = alldiv[(log10(alldiv$meancov) > 0),]
subdiv = subdiv[(subdiv$nvariant > 1000),]
#subdiv = subdiv[(subdiv$species != "Brassica rapa"),] # omit other outlier species
```

## replot trends
```{r}
my_ggplot_wrap(subdiv, "log10(meancov)", "log10(pi)", "log10(Average coverage per individual)", "log10(Nucleotide diversity)", paste("cda_coverage-vs-pi_", today, ".png"))

my_ggplot_wrap(subdiv, "log10(meancov)", "jac", "log10(Average coverage per individual)", "Jaccard dissimilarity", paste("cda_coverage-vs-jac_", today, ".png"))

my_ggplot_wrap(subdiv, "log10(meancov)", "bcd", "log10(Average coverage per individual)", "Bray-Curtis dissimilarity", paste("cda_coverage-vs-bcd_", today, ".png"))

my_ggplot_wrap(subdiv, "cvbp", "log10(pi)", "Coefficient of variation in base pairs sequenced across all samples", "log10(Nucleotide diversity)", paste("cda_cvbp-vs-pi_", today, ".png"))

my_ggplot_wrap(subdiv, "cvbp", "jac", "Coefficient of variation in base pairs sequenced across all samples", "Jaccard dissimilarity", paste("cda_cvbp-vs-jac_", today, ".png"))

my_ggplot_wrap(subdiv, "cvbp", "bcd", "Coefficient of variation in base pairs sequenced across all samples", "Bray-Curtis dissimilarity", paste("cda_cvbp-vs-bcd_", today, ".png"))

cda_pi_vs_jac = my_ggplot_wrap_xyz(subdiv, "log10(pi)", "jac", "log10(Nucleotide diversity)", "Jaccard dissimilarity", paste("cda_pi-vs-jac_", today, ".png"), "log10(meancov)", "#FEF2F2", "#190C64", "log10(mean cov)", c(min(log10(alldiv$meancov)), max(log10(alldiv$meancov))))

cda_pi_vs_bcd = my_ggplot_wrap_xyz(subdiv, "log10(pi)", "bcd", "log10(Nucleotide diversity)", "Bray-Curtis dissimilarity", paste("cda_pi-vs-bcd_", today, ".png"), "log10(meancov)", "#FEF2F2", "#190C64", "log10(mean cov)", c(min(log10(alldiv$meancov)), max(log10(alldiv$meancov))))

myggplotwrap(subdiv, "jac", "bcd", "Jaccard dissimilarity", "Bray-Curtis dissimilarity", paste("cda_jac-vs-bcd_", today, ".png"))

# compare population size proxies with diversity
cda_pi_vs_wcvp = my_ggplot_wrap(subdiv, "log10(pi)", "log10(WCVP.popsize)", "log10(Nucleotide diversity)", "log10(WCVP pop size proxy)", paste("cda_pi-vs-wcvp_", today, ".png"))

cda_pi_vs_gbif = my_ggplot_wrap(subdiv, "log10(pi)", "log10(GBIF.popsize)", "log10(Nucleotide diversity)", "log10(GBIF pop size proxy)", paste("cda_pi-vs-gbif_", today, ".png"))

cda_jac_vs_wcvp = my_ggplot_wrap(subdiv, "jac", "log10(WCVP.popsize)", "Jaccard dissimilarity", "log10(WCVP pop size proxy)", paste("cda_jac-vs-wcvp_", today, ".png"))

cda_jac_vs_gbif = my_ggplot_wrap(subdiv, "jac", "log10(GBIF.popsize)", "Jaccard dissimilarity", "log10(GBIF pop size proxy)", paste("cda_jac-vs-gbif_", today, ".png"))

cda_bcd_vs_wcvp = my_ggplot_wrap(subdiv, "bcd", "log10(WCVP.popsize)", "Bray-Curtis dissimilarity", "log10(WCVP pop size proxy)", paste("cda_bcd-vs-wcvp_", today, ".png"))

cda_bcd_vs_gbif = my_ggplot_wrap(subdiv, "bcd", "log10(GBIF.popsize)", "Bray-Curtis dissimilarity", "log10(GBIF pop size proxy)", paste("cda_bcd-vs-gbif_", today, ".png"))
```

## Combine tree and data to get comparative object
```{r}
# create comparative data object for caper models
# keep GBIF data separate from WCVP be
compdata = comparative.data(phy, subdiv[,c("species","pi", "jac", "bcd", "genome.size", "meancov", "nidv", "cvbp", "cultivation.status", "Mating.system", "WCVP.popsize", "area", "height_final")], names.col = "species", vcv = T)

gbif_compdata = comparative.data(phy, subdiv[,c("species","pi", "jac", "bcd", "genome.size", "meancov", "nidv", "cvbp", "cultivation.status", "Mating.system", "GBIF.popsize", "gbif.area")], names.col = "species", vcv = T)
```

## fit models

```{r}
# check for multicollinarity before fitting models
cor(compdata$data[,c(-8,-9)], method = "spearman", use = "pairwise.complete.obs")

##############################
# WCVP population size vs pi #
##############################

# build models 
mod_pi_1 = pgls(scale(log10(pi)) ~ log10(WCVP.popsize) + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, compdata)
summary(mod_pi_1)
plot(mod_pi_1)

mod_pi_2 = pgls(scale(log10(pi)) ~ log10(WCVP.popsize) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, compdata)
summary(mod_pi_2)

mod_pi_3 = pgls(scale(log10(pi)) ~ log10(WCVP.popsize) + nidv + log10(genome.size) + cvbp + Mating.system, compdata)
summary(mod_pi_3)
plot(mod_pi_3)

mod_mat_wcvp_pop_vs_pi = model.matrix(~ log10(WCVP.popsize) + nidv + log10(genome.size) + cvbp + Mating.system, data = compdata$data) # minimal model, extract matrix

# Jaccard dissimilarity vs WCVP popsize
mod_jac_1 = pgls(scale(jac) ~ log10(WCVP.popsize) + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, compdata)
summary(mod_jac_1)
plot(mod_jac_1)

mod_mat_wcvp_pop_vs_jac = model.matrix(~ log10(WCVP.popsize) + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, compdata$data) # minimal model, extract matrix

# Bray-Curtis dissimilarity vs WCVP popsize
mod_bcd_1 = pgls(scale(bcd) ~ log10(WCVP.popsize) + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, compdata)
summary(mod_bcd_1)
plot(mod_bcd_1)

mod_mat_wcvp_pop_vs_bcd = model.matrix(~ log10(WCVP.popsize) + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, compdata$data) # minimal model, extract matrix

################################
# WCVP range size vs diversity #
################################

# pi
mod_pi_wcvp_area_1 = pgls(scale(log10(pi)) ~ log10(area) + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, compdata)
summary(mod_pi_wcvp_area_1)

mod_pi_wcvp_area_2 = pgls(scale(log10(pi)) ~ log10(area) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, compdata)
summary(mod_pi_wcvp_area_2)

mod_pi_wcvp_area_3 = pgls(scale(log10(pi)) ~ log10(area) + nidv + log10(genome.size) + cvbp + Mating.system, compdata)
summary(mod_pi_wcvp_area_3)
plot(mod_pi_wcvp_area_3)

mod_mat_wcvp_area_vs_pi = model.matrix(~log10(area) + nidv + log10(genome.size) + cvbp + Mating.system, data = compdata$data) # minimal model, extract matrix

# jaccard dissimilarity
mod_jac_wcvp_area_1 = pgls(scale(jac) ~ log10(area) + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, compdata)
summary(mod_jac_wcvp_area_1)
plot(mod_jac_wcvp_area_1)

mod_mat_wcvp_area_vs_jac = model.matrix(~ log10(area) + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, data = compdata$data) # minimal model, extract matrix

# bray-curtis dissimilarity
mod_bcd_wcvp_area_1 = pgls(scale(bcd) ~ log10(area) + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, compdata)
summary(mod_bcd_wcvp_area_1)

mod_bcd_wcvp_area_2 = pgls(scale(bcd) ~ log10(area) + log10(meancov) + nidv + log10(genome.size) + cvbp + cultivation.status, compdata)
summary(mod_bcd_wcvp_area_2)
plot(mod_bcd_wcvp_area_2)

mod_mat_wcvp_area_vs_bcd = model.matrix(~ log10(area) + log10(meancov) + nidv + log10(genome.size) + cvbp + cultivation.status, data = compdata$data) # minimal model, extract matrix

#######################
# height vs diversity #
#######################

# pi
mod_pi_height_1 = pgls(scale(log10(pi)) ~ height_final + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, compdata)
summary(mod_pi_height_1)

mod_pi_height_2 = pgls(scale(log10(pi)) ~ height_final + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, compdata)
summary(mod_pi_height_2)

mod_pi_height_3 = pgls(scale(log10(pi)) ~ height_final + nidv + log10(genome.size) + cvbp + Mating.system, compdata)
summary(mod_pi_height_3)

# jaccard dissimilarity
mod_jac_height_1 = pgls(scale(jac) ~ height_final + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, compdata)
summary(mod_jac_height_1)

# bray-curtis dissimilarity
mod_bcd_height_1 = pgls(scale(bcd) ~ height_final + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, compdata)
summary(mod_bcd_height_1)

mod_bcd_height_2 = pgls(scale(bcd) ~ height_final + log10(meancov) + nidv + log10(genome.size) + cvbp + cultivation.status, compdata)
summary(mod_bcd_height_2)

#####################################
# GBIF population size vs diversity #
#####################################

# pi
mod_pi_gbif_pop_1 = pgls(scale(log10(pi)) ~ log10(GBIF.popsize + 1) + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, gbif_compdata)
summary(mod_pi_gbif_pop_1)

mod_pi_gbif_pop_2 = pgls(scale(log10(pi)) ~ log10(GBIF.popsize + 1) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, gbif_compdata)
summary(mod_pi_gbif_pop_2)

mod_pi_gbif_pop_3 = pgls(scale(log10(pi)) ~ log10(GBIF.popsize + 1) + nidv + log10(genome.size) + cvbp + Mating.system, gbif_compdata)
summary(mod_pi_gbif_pop_3)

# jaccard dissimilarity
mod_jac_gbif_pop_1 = pgls(scale(jac) ~ log10(GBIF.popsize + 1) + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, gbif_compdata)
summary(mod_jac_gbif_pop_1)

mod_jac_gbif_pop_2 = pgls(scale(jac) ~ log10(GBIF.popsize + 1) + log10(meancov) + nidv + log10(genome.size) + Mating.system + cultivation.status, gbif_compdata)
summary(mod_jac_gbif_pop_2)

# bray-curtis dissimilarity
mod_bcd_gbif_pop_1 = pgls(scale(bcd) ~ log10(GBIF.popsize + 1) + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, gbif_compdata)
summary(mod_bcd_gbif_pop_1)

################################
# GBIF range size vs diversity #
################################

# pi
mod_pi_gbif_area_1 = pgls(scale(log10(pi)) ~ log10(gbif.area + 1) + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, gbif_compdata)
summary(mod_pi_gbif_area_1)

mod_pi_gbif_area_2 = pgls(scale(log10(pi)) ~ log10(gbif.area + 1) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, gbif_compdata)
summary(mod_pi_gbif_area_2)

mod_pi_gbif_area_3 = pgls(scale(log10(pi)) ~ log10(gbif.area + 1) + nidv + log10(genome.size) + cvbp + Mating.system, gbif_compdata)
summary(mod_pi_gbif_area_3)

# jaccard dissimilarity
mod_jac_gbif_area_1 = pgls(scale(jac) ~ log10(gbif.area + 1) + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, gbif_compdata)
summary(mod_jac_gbif_area_1)

mod_jac_gbif_area_2 = pgls(scale(jac) ~ log10(gbif.area + 1) + log10(meancov) + nidv + log10(genome.size) + Mating.system + cultivation.status, gbif_compdata)
summary(mod_jac_gbif_area_2)

# bray-curtis dissimilarity
mod_bcd_gbif_area_1 = pgls(scale(bcd) ~ log10(gbif.area + 1) + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, gbif_compdata)
summary(mod_bcd_gbif_area_1)

# function to perform phylogenetic least squares, adjust for covariates, and plot primary variable against responses
correct_and_plot = function(mod_mat, y, V, primary, xlab, ylab){
  # Get cholesky decomposition of variance covariance matrix
  # then invert the matrix
  # then transpose it
  C_inv = t(solve(chol(V)))
  
  # scale data by phylogenetic effect
  # This translates the generalized least squares model into an ordinary least squares model
  y_star = C_inv %*% y
  X_star = C_inv %*% mod_mat
  
  # perform ordinary least squares regression
  # omit intercept because it's included in design matrix
  mod_star = lm(y_star ~ X_star - 1)
  coeff_star = coefficients(mod_star)
  
  names(coeff_star) = gsub("^X_star", "", names(coeff_star)) # remove new names from coefficients
  
  # extract p-value for slope of primary variable
  pvalues = summary(mod_star)$coefficients
  rownames(pvalues) = gsub("^X_star", "", rownames(pvalues)) # remove new names from coefficients
  slope_p_value = signif(pvalues[primary,4], digits = 2)
  
  # correct response for phylogeny and other covariates, then plot against predictor of interest
  technicals = colnames(X_star)[(colnames(X_star) != primary)]
  y_star_corrected = y_star - (X_star[,technicals] %*% coeff_star[technicals])
  
  p.label = paste("p = ", slope_p_value, sep = "")
  
  # plot result
  myplot = ggplot(mapping = aes(x = X_star[,primary], y = y_star_corrected)) +
    geom_point() +
    geom_abline(intercept = 0, slope = coeff_star[primary], color = "pink") +
    theme_classic() +
    stat_regline_equation(aes(label = paste(..eq.label.., ..adj.rr.label.., sep = "~~~~"))) +
    annotate(geom = "text", x = 1.5, y = (max(y_star_corrected) + min(y_star_corrected))*0.5, label = p.label) +
    labs(x = xlab, y = ylab)
  
  #print(X_star[,primary])
    
  return(myplot)
}

# create plots, save to objects

wcvp_pop_vs_pi_corrected = correct_and_plot(mod_mat_wcvp_pop_vs_pi, mod_pi_1$y, mod_pi_1$Vt, "log10(WCVP.popsize)", "log10(WCVP pop size proxy), scaled", "log10(Nucleotide diversity), scaled and adjusted")

wcvp_pop_vs_jac_corrected = correct_and_plot(mod_mat_wcvp_pop_vs_jac, mod_jac_1$y, mod_jac_1$Vt, "log10(WCVP.popsize)", "log10(WCVP pop size proxy), scaled", "Jaccard dissimilarity, scaled and adjusted")

wcvp_pop_vs_bcd_corrected = correct_and_plot(mod_mat_wcvp_pop_vs_bcd, mod_bcd_1$y, mod_bcd_1$Vt, "log10(WCVP.popsize)", "log10(WCVP pop size proxy), scaled", "Bray-Curtis dissimilarity, scaled and adjusted")

wcvp_area_vs_pi_corrected = correct_and_plot(mod_mat_wcvp_area_vs_pi, mod_pi_wcvp_area_3$y, mod_pi_wcvp_area_3$Vt, "log10(area)", "log10(WCVP range size in square km), scaled", "log10(Nucleotide diversity), scaled and adjusted")

wcvp_area_vs_jac_corrected = correct_and_plot(mod_mat_wcvp_area_vs_jac, mod_jac_wcvp_area_1$y, mod_jac_wcvp_area_1$Vt, "log10(area)", "log10(WCVP range size in square km), scaled", "Jaccard dissimilarity, scaled and adjusted")

wcvp_area_vs_bcd_corrected = correct_and_plot(mod_mat_wcvp_area_vs_bcd, mod_bcd_wcvp_area_2$y, mod_bcd_wcvp_area_2$Vt, "log10(area)", "log10(WCVP range size in square km), scaled", "Bray-Curtis dissimilarity, scaled and adjusted")

```

# Final plots

```{r}
# make initial data analysis plots, confirming relationship between WCVP and GBIF, jaccard and bray-curtis
ggarrange(ida_wcvp_vs_gbif, ida_jac_vs_bcd, nrow = 1, ncol = 2, labels = c("A", "B"))

ggsave(paste("wcvp_vs_gbif_and_jac_vs_bcd_", today, ".png", sep = ""))

# Look at how coverage influences metrics
ggarrange(ida_meancov_vs_pi, ida_cvbp_vs_pi, ida_meancov_vs_jac, ida_cvbp_vs_jac, ida_meancov_vs_bcd, ida_cvbp_vc_bcd, ncol = 2, nrow = 3, labels = c("A", "B", "C", "D", "E", "F"))

ggsave(paste("coverage_vs_diversity_", today, ".png", sep = ""), width = 8, height = 10, units = "in")

# Look at how mating system, cultivation status, ploidy influence diversity
ggarrange(ida_mating_vs_pi, ida_cult_vs_pi, ida_ploidy_vs_pi, ida_mating_vs_jac, ida_cult_vs_jac, ida_ploidy_vs_jac, ida_mating_vs_bcd, ida_cult_vs_bcd, ida_ploidy_vs_bcd, ncol = 3, nrow = 3, labels = c("A", "B", "C", "D", "E", "F", "G", "H"))

ggsave(paste("mating_cultivation_ploidy_vs_diversity_", today, ".png", sep = ""), width = 10, height = 10, units = "in")

# Compare how k-mer metrics compare to SNP metrics after removing low coverage data
ggarrange(ida_pi_vs_jac, cda_pi_vs_jac, ida_pi_vs_bcd, cda_pi_vs_bcd, ncol = 2, nrow = 2, common.legend = T, labels = c("A", "B", "C", "D"))

ggsave(paste("pi_vs_jac_and_bcd_high_vs_low_cov_", today, ".png", sep = ""), width = 8, height = 7, units = "in")

# Compare population sizes to diversity metrics, without correcting for phylogeny
ggarrange(cda_pi_vs_gbif, cda_pi_vs_wcvp, cda_jac_vs_gbif, cda_jac_vs_wcvp, cda_bcd_vs_gbif, cda_bcd_vs_wcvp, nrow = 3, ncol = 2, labels = c("A", "B", "C", "D", "E", "F"))

ggsave(paste("diversity_vs_popsize_", today, ".png", sep = ""), width = 8, height = 10, units = "in")

# plot phylogenetic least squares models
ggarrange(wcvp_pop_vs_pi_corrected, wcvp_pop_vs_jac_corrected, wcvp_pop_vs_bcd_corrected, wcvp_area_vs_pi_corrected, wcvp_area_vs_jac_corrected, wcvp_area_vs_bcd_corrected,nrow = 2, ncol = 3, labels = c("A", "B", "C", "D", "E", "F"))

ggsave(paste("wcpv_vs_diversity_corrected_", today, ".png", sep = ""), height = 9, width = 13.5, units = "in")
```

```{r}
# pi vs bcd
ggplot(alldiv, aes(x = bcd, y = log10(pi), label = species)) +
  geom_point() +
  geom_text(position = position_jitter(seed = 1)) +
  theme_classic() +
  xlim(0,0.8) +
  labs(x = "Bray-Curtis dissimilarity", y = "log10(Nucleotide diversity)")
ggsave(paste("../results/2023-03-26/figures/Bray-CurtisDissimilarityVsPi_", today, ".png", sep = ""), width = 7, height = 7, unit = "in", scale = 0.5)

# range vs pi
r1 = ggplot(alldiv, aes(x = log10(rangeArea), y = log10(pi), label = species)) +
  geom_point() +
  geom_text(position = position_jitter(seed = 1)) +
  geom_smooth(method = "auto") +
  theme_classic() +
  theme(text = element_text(size = 20)) +
  xlim(5.5, 8.5) +
  labs(x = "log10(Range area in square km)", y = "log10(Nucleotide diversity)")

#ggsave(paste("../results/2023-03-26/figures/rangeVsPi_", today, ".png", sep = ""), width = 7, height = 7, unit = "in", scale = 0.5)

# range vs bcd
r2 = ggplot(alldiv, aes(x = log10(rangeArea), y = bcd, label = species)) +
  geom_point() +
  geom_text(position = position_jitter(seed = 1)) + 
  geom_smooth(method = "auto") +
  theme_classic() +
  theme(text = element_text(size = 20)) +
  xlim(5.5, 8.5) +
  labs(x = "log10(Range area in square km)", y = "Bray-Curtis dissimilarity")

#ggsave(paste("../results/2023-03-26/figures/rangeVsBray-Curtis_", today, ".png", sep = ""), width = 7, height = 7, unit = "in", scale = 0.5)

# range vs jaccard dissimilarity
r3 = ggplot(alldiv, aes(x = log10(rangeArea), y = jac, label = species)) +
  geom_point() +
  geom_text(position = position_jitter(seed = 1)) + 
  geom_smooth(method = "auto") +
  theme_classic() +
  theme(text = element_text(size = 20)) +
  xlim(5.5, 8.5) +
  labs(x = "log10(Range area in square km)", y = "Jaccard dissimilarity")

gridplot = plot_grid(r1, r2, r3, labels = "AUTO", ncol = 3)

ggsave(paste("../results/2023-03-26/figures/range_vs_diversity_", today, ".png", sep = ""), width = 21, height = 7)
       
#ggsave(paste("../results/2023-03-26/figures/rangeVsJaccard_", today, ".png", sep = ""), width = 7, height = 7, unit = "in", scale = 0.5)
```

## plot phylogeny alongside data

ggtree causes linear models to crash, so need to do this separately

```{r}
#install.packages("ggtree")
#if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")
#BiocManager::install("ggtree")

#install.packages("ggtree")
library(ggtree)

# test ggtree package
#tr = rtree(10)
#ggtree(tr) + geom_tiplab()

# arrange plot data to match order of tip labels
plotdf = alldiv %>% arrange(factor(species, levels = phy$tip.label))

plotdf$Ploidy = as.numeric(plotdf$Ploidy)

plotdf = plotdf[!is.na(plotdf$species),]

# remove species from phylogeny that lack data
plotphy = drop.tip(phy, tip = phy$tip.label[!(phy$tip.label %in% alldiv$species)])

# change tip labels to be mating system
#plotphy$tip.label = plotdf$Mating.system

 # get color pallete
#plotpal = scico(5, palette = "lipari")
plotpal = c("#021326","#515A79","#A36267","#E99973","#FDF4D9")
#plotpal = c("#021326","#294B70","#6B5F76","#A36267","#E57A61","#E5B488","#FDF4D9")
#plotpal = c("#021326","#3B5378","#7F5F70","#CE685E","#E5AA7F","#FDF4D9")
#plotpal = c("#191900","#663329","#D85F4D","#ECAC54","#FFFECB")

# plot phylogeny alongside other data
ggtree(plotphy, size = 1.5) + 
    geom_facet(panel = "log10(GBIF pop size proxy)", data = plotdf, geom = geom_segment, size = 3, mapping = aes(x = 0, y = y, yend = y, xend = log10(GBIF.popsize)), color = plotpal[1]) +
    geom_facet(panel = "log10(WCVP pop size proxy)", data = plotdf, geom = geom_segment, size = 3, mapping = aes(x = 0, y = y, yend = y, xend = log10(WCVP.popsize)), color = plotpal[2]) +
    geom_facet(panel = "log10(SNP diversity)", data = plotdf, geom = geom_segment, size = 3, mapping = aes(x = 0, y = y, yend = y, xend = log10(pi)), color = plotpal[3]) +
  geom_facet(panel = "Bray-Curtis dissimilarity", data = plotdf, geom = geom_segment, size = 3, mapping = aes(x = 0, y = y, yend = y, xend = bcd), color = plotpal[4]) +
  geom_facet(panel = "Jaccard dissimilarity", data = plotdf, geom = geom_segment, size = 3, mapping = aes(x = 0, y = y, yend = y, xend = jac), color = plotpal[5]) +
  theme_tree2() +
  theme(strip.background =element_rect(fill="white")) +
  theme(strip.text = element_text(colour = 'black'))
  
ggsave(paste("tree_with_primary_variables_", today, ".png", sep = ""), scale = 2)

# plot technical variables
ggtree(plotphy, size = 1.5) + 
  #geom_tippoint(mapping = aes(shape = Mating.system), data = plotdf, size = 1.5) + 
  geom_facet(panel = "Ploidy", data = plotdf, geom = geom_segment, size = 3, mapping = aes(x = 0, y = y, yend = y, xend = Ploidy), color = plotpal[1]) +
  geom_facet(panel = "Genome size", data = plotdf, geom = geom_segment, size = 3, mapping = aes(x = 0, y = y, yend = y, xend = genome.size), color = plotpal[2]) +
  geom_facet(panel = "Individuals", data = plotdf, geom = geom_segment, size = 3, mapping = aes(x = 0, y = y, yend = y, xend = nidv), color = plotpal[3]) +
  geom_facet(panel = "Mean coverage", data = plotdf, geom = geom_segment, size = 3, mapping = aes(x = 0, y = y, yend = y, xend = (totalbp/(genome.size*nidv))), color = plotpal[4]) +
  geom_facet(panel = "CV in bp sequenced", data = plotdf, geom = geom_segment, size = 3, mapping = aes(x = 0, y = y, yend = y, xend = cvbp), color = plotpal[5]) +
  theme_tree2() +
  theme(strip.background =element_rect(fill="white")) +
  theme(strip.text = element_text(colour = 'black'))

ggsave(paste("tree_with_technical_variables_", today, ".png", sep = ""), scale = 2)

```

# DEPRECATED
```{r}
# subtract off effects of covariates from response
mod_mat_pi = model.matrix(~ log10(WCVP.popsize) + nidv + log10(genome.size) + cvbp + Mating.system, data = compdata$data)

mod_mat_jac = model.matrix(~ log10(WCVP.popsize) + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, data = compdata$data)

mod_mat_bcd = model.matrix(~ log10(WCVP.popsize) + log10(meancov) + nidv + log10(genome.size) + cvbp + Mating.system + cultivation.status, data = compdata$data)

mod_coeff_pi = coefficients(mod_pi_3)

mod_coeff_jac = coefficients(mod_jac_1)

mod_coeff_bcd = coefficients(mod_bcd_1)

# Plotting response against summed linear model should give a 1:1 line
plot(mod_bcd_1$y, mod_mat %*% mod_coeff + mod_bcd_1$residuals)
abline(a = 0, b = 1)

# correct response for phylogeny and other covariates, then plot against predictor of interest
C_inv_pi = t(solve(chol(mod_pi_3$Vt)))
C_inv_jac = t(solve(chol(mod_jac_1$Vt)))
C_inv_bcd = t(solve(chol(mod_bcd_1$Vt)))

y_star_pi = C_inv_pi %*% mod_pi_3$y
X_star_pi = (C_inv_pi %*% mod_mat_pi)

y_star_jac = C_inv_jac %*% mod_jac_1$y
X_star_jac = (C_inv_jac %*% mod_mat_jac)

y_star_bcd = C_inv_bcd %*% mod_bcd_1$y
X_star_bcd = (C_inv_bcd %*% mod_mat_bcd)

# confirm that results of pgls match results of ordinary least squares on transformed data
# remove intercept column, already included in 
mod_star_bcd = lm(y_star_bcd ~ X_star_bcd - 1)
coeff_star_bcd = coefficients(mod_star_bcd)
(coeff_star_bcd - mod_coeff_bcd)/mod_coeff_bcd * 100 # percent difference in coefficients

# Isolate effect of covariate I'm interested in
# subtract extra covariates and slope from both sides of the equation
# Remember: the intercept is not a simple constant here! It's been scaled by C_inv
y_star_corrected_pi = y_star_pi - (X_star_pi[,c(1,3:ncol(X_star_pi))] %*% mod_coeff_pi[c(1,3:length(mod_coeff_pi))])

y_star_corrected_jac = y_star_jac - (X_star_jac[,c(1,3:ncol(X_star_jac))] %*% mod_coeff_jac[c(1,3:length(mod_coeff_jac))])

y_star_corrected_bcd = y_star_bcd - (X_star_bcd[,c(1,3:ncol(X_star_bcd))] %*% mod_coeff_bcd[c(1,3:length(mod_coeff_bcd))])

#plot(X_star[,2], y_star_corrected_2)
#abline(a = 0, coeff_star[2])

# plot response, scaled by phylogeny and corrected for other covariates, against diversity
my_ggplot_wrap_lm = function(x,y,b,xlab,ylab){
  myplot = ggplot(mapping = aes(x = x, y = y)) +
    geom_point() +
    geom_abline(intercept = 0, slope = b) +
    theme_classic() +
    labs(x = xlab, y = ylab)
    
  return(myplot)
}

my_ggplot_wrap_lm(X_star_pi[,2], y_star_corrected_pi, mod_coeff_pi[2], "log10(WCVP.popsize)")

my_ggplot_wrap_lm(X_star_jac[,2], y_star_corrected_jac, mod_coeff_jac[2])

my_ggplot_wrap_lm(X_star_bcd[,2], y_star_corrected_bcd, mod_coeff_bcd[2])





y_star_corrected = y_star - (X_star[,3:ncol(X_star)] %*% coeff_star[3:length(coeff_star)])

plot(X_star[,1:2] %*% mod_coeff[1:2], y_star_corrected) # data should center around 1:1 reference line
abline(a = 0, b = 1)

plot(X_star[,2], y_star_corrected[,1])
abline(a = mod_coeff[1], b = mod_coeff[2])

ggplot(mapping = aes(x = X_star[,2], y = y_star_corrected[,1]))+
  geom_point() +
  geom_smooth(method = "lm") +
  theme_classic()

test_mod = lm(y_star_corrected ~ X_star[,1:2] - 1)
plot(test_mod)

plot(C_inv %*% mod_bcd_1$y - (C_inv %*% mod_mat[,3:ncol(mod_mat)] %*% mod_coeff[3:length(mod_coeff)]), C_inv %*% mod_mat[,1:2] %*% mod_coeff[1:2] + C_inv %*% mod_bcd_1$residuals)
abline(a = 0, b = 1)

# Now leave out residuals
plot(mod_bcd_1$y - (mod_mat[,3:ncol(mod_mat)] %*% mod_coeff[3:length(mod_coeff)] + mod_bcd_1$phyres), log10(compdata$data$WCVP.popsize))

plot(C_inv %*% mod_bcd_1$y - (C_inv %*% mod_mat[,3:ncol(mod_mat)] %*% mod_coeff[3:length(mod_coeff)]), C_inv %*% mod_mat[,1:2] %*% mod_coeff[1:2])
abline(a = 0, b = 1)

# plot correlation between diversity, corrected for phylogeny and other covariates, with population size
plot(C_inv %*% mod_bcd_1$y - (C_inv %*% mod_mat[,3:ncol(mod_mat)] %*% mod_coeff[3:length(mod_coeff)]), C_inv %*% log10(compdata$data$WCVP.popsize))

y_corrected = C_inv %*% mod_bcd_1$y - (C_inv %*% mod_mat[,3:ncol(mod_mat)] %*% mod_coeff[3:length(mod_coeff)])

lm(y_corrected ~ C_inv %*% log10(compdata$data$WCVP.popsize))










y_corrected = compdata$data$bcd - (mod_mat[,3:ncol(mod_mat)] %*% mod_coeff[3:length(mod_coeff)])

cor.test(log10(compdata$data$WCVP.popsize), y_corrected , method = "spearman")

plot(log10(compdata$data$WCVP.popsize), y_corrected + mod_bcd_1$phyres)

# model diagnostics
plot(mod_pi)
plot(mod_pi$fitted, mod_pi$fitted - mod_pi$residuals)
abline(a = 0, b = 1)

plot(mod_jac)

plot(mod_bcd)






# plot out coefficients
coefplot = data.frame(coef = coefficients(mod_bcd), err = mod_bcd$sterr, label = names(mod_bcd$sterr))
coefplot = coefplot[-1,]
coefplot$coef = abs(coefplot$coef)

ggplot(coefplot, aes(x=coef, y=label)) + 
  geom_point()+
  geom_errorbar(aes(xmin=coef-err, xmax=coef+err), width=.2, position=position_dodge(0.05)) +
  geom_vline(xintercept = 0, lty = "dotted", size = 2) +
  theme_classic() +
  theme(text = element_text(size = 20)) +
  labs(x = "Absolute value of coefficient", y = "PGLS model parameter")

ggsave(paste("pgls_coefficients_", today, ".png", sep = ""), width = 7, height = 7)

mod_jac = pgls(log10(rangeArea) ~ scale(pi) + scale(jac), compdata)
summary(mod_jac)

# control for the number of individuals sequenced
mod_bcd = pgls(log10(rangeArea) ~ scale(pi) + scale(bcd), compdata)
summary(mod_bcd)

mod_jac = pgls(log10(rangeArea) ~ scale(pi) + scale(jac) + scale(nidv), compdata)
summary(mod_jac)

# control for amount of sequencing data
mod_bcd = pgls(log10(rangeArea) ~ scale(bcd), compdata)
summary(mod_bcd)
plot(mod_bcd)

mod_pi = pgls(log10(rangeArea) ~ scale(pi), compdata)
summary(mod_pi)
plot(mod_pi)

cor(alldiv$nidv, alldiv$totalbp, use = "pairwise.complete.obs")
cor(log10(alldiv$pi), (alldiv$bcd), use = "pairwise.complete.obs", method = "spearman")
cor(alldiv$bcd, alldiv$totalbp, use = "pairwise.complete.obs", method = "spearman")
cor(alldiv$nidv, alldiv$totalbp/alldiv$nidv, method = "spearman")
#
mod_bcd = pgls(log10(rangeArea) ~ scale(bcd) + scale(nidv), compdata)
summary(mod_bcd)

mod_pi = pgls(log10(rangeArea) ~ scale(pi) + scale(nidv), compdata)
summary(mod_pi)

plot(mod_bcd)

# try contrasts instead of variance-covariance matrices
alldiv$scale_bcd = scale(alldiv$bcd)
alldiv$scale_pi = scale(alldiv$pi)
alldiv$scale_jac = scale(alldiv$jac)
alldiv$scale_nidv = scale(alldiv$nidv)
alldiv$scale_bp_per_idv = scale(alldiv$totalbp/alldiv$nidv)
alldiv$log_range = log10(alldiv$rangeArea)

compdata = comparative.data(phy, alldiv, names.col = "species", vcv = T)

crunch_mod = crunch(log_range ~ scale_bcd + scale_pi + scale_nidv + scale_bp_per_idv, data = compdata)
summary(crunch_mod)
plot(crunch_mod)
```