---
title: "2023-04-12 results"
output:
  html_document:
    df_print: paged
  pdf_document: default
---

# load packages
```{r}
rm(list = ls())

library(ggplot2)
library(caper)
library(stringr)

today = Sys.Date()
```

# Data preparation

## Prepare diversity data 
```{r}
# List input files
pi = list.files("../results/2023-03-26/genome-wide-pi", full.names = T)
bcd = list.files("../results/2023-03-26/bray-curtis-dissimilarities", full.names = T)
jac = list.files("../results/2023-03-26/jaccard-dissimilarities", full.names = T)

# get species IDs
species = gsub("_pairwise-pi.txt", "", gsub("../results/2023-03-26/genome-wide-pi/", "", pi))

# Load input files
pi = lapply(pi, read.table, header = T)
bcd = lapply(bcd, read.table, header = F)
jac = lapply(jac, read.table, header = F)

# calculate mean pairwise dissimilarity
bcdMean = lapply(bcd, function(x){mean(x[,2])})
jacMean = lapply(jac, function(x){mean(x[,2])})

# calculate average number of k-mers included in union
unionMean = lapply(jac, function(x){mean(x[,4])})

# cbind all diversity data into one dataframe per species
alldiv = Map(cbind, pi, bcdMean, jacMean, unionMean)

# A. thaliana has extra data, remove for now
alldiv[[7]] = alldiv[[7]][,c(1, 3:6, 9:11)]

# rbind all species data into one dataframe
alldiv = lapply(alldiv, setNames, c("h", "ntotal", "nvariant", "ninvariant", "pi", "bcd", "jac", "umean"))

alldiv = do.call(rbind, alldiv)

# add species names
alldiv$species = species
alldiv$species = str_to_sentence(gsub("_", " ", alldiv$species))

# add coverage covariate
# coverage = read.table("../results/2023-03-26/bp_sequenced.txt", header = T, sep = "\t")
# alldiv = merge(alldiv, coverage, by = "species")
# 
# gsize = read.table("../results/2023-03-26/genome_size.txt", header = T, sep = "\t")
# alldiv = merge(alldiv, gsize, by = "species")
# 
# alldiv$totalcov = alldiv$totalbp/alldiv$assemblySize
```

## Load range data
```{r}
# load range data
areas = list.files("../results/2023-03-26/ranges/areas", full.names = T)

areas_species = gsub("_areas.txt", "", gsub("../results/2023-03-26/ranges/areas/", "", areas))
areas_species = str_to_sentence(gsub("_", " ", areas_species))

areas = lapply(areas, read.table, header = T, sep = "\t")
```

## remove areas outside of native range 
```{r}
# Refer to plants of the world online for native range maps

# Amaranthus hybridus native to only North and South America
areas[[2]] = areas[[2]][c(4,6),]

# Arabidopsis thaliana is only native to Europe, Asia, and Africa
areas[[7]] = areas[[7]][2:4,]

# Nicotiana glauca only native to South America
areas[[14]] = areas[[14]][2,]

# Arachis glabrata is native to Brazil, remove North America
areas[[10]] = areas[[10]][1,]
```

## Merge range data with diversity data
```{r}
# function to get non-water area
get_nonwater_area = function(x){
  sum(x[,"total_area"] - x[,"water_area"])
}

# format area data as frame
areas = data.frame(
  species = areas_species,
  rangeArea = unlist(lapply(areas, get_nonwater_area))
)

# merge with diversity data
alldiv = merge(alldiv, areas, by = "species", all.x = T)

print(alldiv)
```

## Prepare phylogenetic tree
```{r}
# load time tree
phy = read.tree("../results/2023-03-26/speciesList.nwk")
phy$tip.label = gsub("_", " ", phy$tip.label)

plot(phy)
```

## Combine tree and data to get comparative object
```{r}
compdata = comparative.data(phy, alldiv, names.col = "species", vcv = T)
```

# Initial data analysis
```{r}
# histograms of diversity data
ggplot(alldiv, aes(x = log10(pi))) +
  geom_histogram() +
  theme_classic() +
  labs(x = "log10(Nucleotide diversity)")

ggplot(alldiv, aes(x = jac)) +
  geom_histogram() +
  theme_classic() +
  labs(x = "Jaccard Dissimilarity")

ggplot(alldiv, aes(x = bcd)) +
  geom_histogram() +
  theme_classic() +
  labs(x = "Bray-Curtis Dissimilarity")
```

# Confirmatory data analysis
```{r}
# Does pi correlate with k-mer diversity?
cor(alldiv$pi, alldiv$bcd, method = "spearman", use = "pairwise.complete.obs")
cor(alldiv$pi, alldiv$jac, method = "spearman", use = "pairwise.complete.obs")

# Does k-mer diversity correlate with population size (i.e. range area) more than pi?
cor(alldiv$pi, alldiv$rangeArea, method = "spearman", use = "pairwise.complete.obs")
cor(alldiv$bcd, alldiv$rangeArea, method = "spearman", use = "pairwise.complete.obs")
cor(alldiv$jac, alldiv$rangeArea, method = "spearman", use = "pairwise.complete.obs")

ggplot(alldiv, aes(x = log10(rangeArea), y = log10(pi), label = species)) +
  geom_point() +
  geom_text(hjust = 0, vjust = 0) +
  theme_classic() +
  labs(x = "log10(Range area in square km)", y = "log10(Nucleotide diversity)")

ggplot(alldiv, aes(x = log10(rangeArea), y = bcd, label = species)) +
  geom_point() +
  geom_text(hjust = 0, vjust = 0) + 
  theme_classic() +
  labs(x = "log10(Range area in square km)", y = "Bray-Curtis Dissimilarity")

ggplot(alldiv, aes(x = log10(rangeArea), y = jac, label = species)) +
  geom_point() +
  geom_text(hjust = 0, vjust = 0) + 
  theme_classic() +
  labs(x = "log10(Range area in square km)", y = "Jaccard Dissimilarity")

# compare effects with phylogeny vs without phylogeny
lm(log10(rangeArea) ~ log10(pi), alldiv)
lm(log10(rangeArea) ~ bcd, alldiv)
lm(log10(rangeArea) ~ jac, alldiv)

pgls(log10(rangeArea) ~ log10(pi), compdata)
pgls(log10(rangeArea) ~ bcd, compdata)
pgls(log10(rangeArea) ~ jac, compdata)

# Test if k-mer diversity has stronger effect on range area than SNP diversity
mod_bcd = pgls(log10(rangeArea) ~ scale(pi) + scale(bcd), compdata)
summary(mod_bcd)

mod_jac = pgls(log10(rangeArea) ~ scale(pi) + scale(jac), compdata)
summary(mod_jac)
  
# try removing outliers
# compdataNoOut = subset.comparative.data(compdata, subset =  !(species %in% c("Actinidia chinensis", "Camelina sativa")))
```