---
title: "K-mer approach to Lewontin's paradox results"
author: "Miles Roberts"
date: "2023-09-12"
output:
  html_document:
    df_print: paged
  pdf_document: default
---

# load packages
```{r}
rm(list = ls())

library(dplyr)
library(ggplot2)
library(caper)
library(stringr)
library(data.table)
library(cowplot)
#library(scico)
#library(giscoR)
library(rWCVP)
library(lwgeom)
library(sf)

today = Sys.Date()
```

# Data preparation

## Prepare diversity data 
```{r}
# List input files
pi = list.files("../results/2023-03-26/genome-wide-pi", full.names = T)
bcd = list.files("../results/2023-03-26/bray-curtis-dissimilarities", full.names = T)
jac = list.files("../results/2023-03-26/jaccard-dissimilarities", full.names = T)

# for now, remove species that can't be processed further for whatever reason
jac = jac[c(-40, -63)]
bcd = bcd[c(-40, -63)]

#pi = pi[-36]

# get species IDs
species = gsub("_pairwise-pi.txt", "", gsub("../results/2023-03-26/genome-wide-pi/", "", pi))

# Load input files
pi = lapply(pi, fread, header = T, data.table = F)
bcd = lapply(bcd, fread, header = F, data.table = F)
jac = lapply(jac, fread, header = F, data.table = F)

# calculate mean pairwise dissimilarity
bcdMean = lapply(bcd, function(x){mean(x[,2])})
jacMean = lapply(jac, function(x){mean(x[,2])})

# calculate average number of k-mers included in union
unionMean = lapply(jac, function(x){mean(x[,4])})

# calculate number of individuals sampled for each dataset
# You get this equation by rewriting the formula for the number of pairwise comparisons (n^2 - n)/2 = N as a polynomial and applying the quadratic formula
backCalcN = function(x){
  sqrt(1/4 + 2*nrow(x)) + 1/2
}

nidv = lapply(bcd, backCalcN)

# cbind all diversity data into one dataframe per species
alldiv = Map(cbind, pi, bcdMean, jacMean, unionMean, nidv)

# Some species have extra data, remove for now
speciesWithExtraData = which(unlist(lapply(alldiv, ncol)) == 12)
for(i in speciesWithExtraData){
  alldiv[[i]] = alldiv[[i]][,c(1, 3:6, 9:12)]
}

# rbind all species data into one dataframe
alldiv = lapply(alldiv, setNames, c("h", "ntotal", "nvariant", "ninvariant", "pi", "bcd", "jac", "umean", "nidv"))

alldiv = do.call(rbind, alldiv)

# add species names
alldiv$species = species
alldiv$species = str_to_sentence(gsub("_", " ", alldiv$species))

# For now, remove species where invariant sites were not properly called
# alldiv = alldiv[(alldiv$ninvariant > 0),]

# For now, remove species where variants sites were not properly called
# alldiv = alldiv[(alldiv$nvariant > 30),]

# remove species where there aren't many sampled individuals
# alldiv = alldiv[(alldiv$nidv >= 10),]

# add coverage covariate
coverage = read.table("../results/2023-03-26/bp_sequenced_2023-09-19.txt", header = T, sep = "\t")
alldiv = merge(alldiv, coverage, by = "species", all.x = TRUE)

# 
# gsize = read.table("../results/2023-03-26/genome_size.txt", header = T, sep = "\t")
# alldiv = merge(alldiv, gsize, by = "species")
# 
# alldiv$totalcov = alldiv$totalbp/alldiv$assemblySize
```

## Load range data
```{r}
# load range data
areas = list.files("../results/2023-03-26/ranges/areas", full.names = T)

# extract species names
areas_species = gsub("_areas.txt", "", gsub("../results/2023-03-26/ranges/areas/", "", areas))
areas_species = str_to_sentence(gsub("_", " ", areas_species))

# load data
areas = lapply(areas, read.table, header = T, sep = "\t")

# add names to list
names(areas) = areas_species

#str(areas)
```

## remove areas outside of native range 
```{r}
# Refer to plants of the world online for native range maps
# powo.science.kew.org

# Amaranthus hybridus native to only North and South America
areas$`Amaranthus hypochondriacus` = areas$`Amaranthus hypochondriacus`[c(4,6),]

# Arabidopsis thaliana is only native to Europe, Asia, and Africa
areas$`Arabidopsis thaliana` = areas$`Arabidopsis thaliana`[2:4,]

# Arachis glabrata is native to Brazil, remove North America
areas$`Arachis hypogaea` = areas$`Arachis hypogaea`[1,]

# Beta vulgaris subsp. vulgaris is native to only Europe and Africa
areas$`Beta vulgaris` = areas$`Beta vulgaris`[1:2,]

# Brachypodium distachyon is native to Europe, Asia, Africa
areas$`Brachypodium distachyon` = areas$`Brachypodium distachyon`[3:5,]

# Brassica cretica native to europe
areas$`Brassica oleracea` = areas$`Brassica oleracea`[1,]

# Brassica fruticulosa native to europe
areas$`Brassica rapa` = areas$`Brassica rapa`[2,]

# Buddleja alternifolia is native to Asia
areas$`Buddleja alternifolia` = areas$`Buddleja alternifolia`[2,]

# Cajanus scarabaeoides native to Africa, Asia, Australia
areas$`Cajanus cajan` = areas$`Cajanus cajan`[c(1,3,4),]

# Cucurbita palmata is only native to North America
areas$`Cucurbita pepo` = areas$`Cucurbita pepo`[1,]

# Digitaria longiflora native to africa, asia, australia
areas$`Digitaria exilis`= areas$`Digitaria exilis`[3:5,]

# Gossypium arboreum native to asia
areas$`Gossypium arboreum` = areas$`Gossypium arboreum`[2,]

# Lactuca sativa is native to Iraq
areas$`Lactuca sativa` = areas$`Lactuca sativa`[3,]

# Malus sylvestris is only native to Europe
areas$`Malus sylvestris` = areas$`Malus sylvestris`[2,]

# Medicago truncatula only native to europe and africa
areas$`Medicago truncatula` = areas$`Medicago truncatula`[c(4,5),]

# Mimulus guttatus only native to North America
areas$`Mimulus guttatus` = areas$`Mimulus guttatus`[1,]

# Musa relative only native to asia and india
areas$`Musa acuminata` = areas$`Musa acuminata`[2,]

# Populus deltoides only native to north america
areas$`Populus deltoides` = areas$`Populus deltoides`[1,]

# Nicotiana glauca only native to South America
areas$`Nicotiana tabacum` = areas$`Nicotiana tabacum`[2,]

# Raphanus raphanistrum is native to europe and africa
areas$`Raphanus sativus` = areas$`Raphanus sativus`[3:4,]

# Setaria viridis 
areas$`Setaria viridis` = areas$`Setaria viridis`[3:6,]
```

## Merge range data with diversity data
```{r}
# function to get non-water area
get_nonwater_area = function(x){
  sum(x[,"total_area"] - x[,"water_area"])
}

# format area data as frame
areas = data.frame(
  species = areas_species,
  rangeArea = unlist(lapply(areas, get_nonwater_area))
)

# merge with diversity data
alldiv = merge(alldiv, areas, by = "species", all.x = T)

print(alldiv)

# save final table
#write.csv(alldiv, paste("../results/2023-03-26/diversityAndRangeData_", today, ".csv", sep = ""), row.names = F, quote = F)
```

## Prepare phylogenetic tree
```{r}
# write species list to upload to timetree.org
#write.table(alldiv$species, paste("../results/2023-03-26/species_list_", today, ".txt", sep = ""), row.names = F, col.names = F, quote = F)

# load time tree from timetree.org
phy = read.tree("../results/2023-03-26/species_list_2023-10-06.nwk")

# remove underscores
phy$tip.label = gsub("_", " ", phy$tip.label)

# remove variety names
phy$tip.label = gsub(" var..*", "", phy$tip.label)

# sanity check
plot(phy)
```

## Load range areas
```{r}
# load local copies of World Checklist of Vascular Plants
wcvp_dist = fread("../workflow/data/range/wcvp_distribution.csv.gz", sep = "|", header = T, quote = "", fill = T, encoding = "UTF-8")

wcvp_names = fread("../workflow/data/range/wcvp_names.csv.gz", sep = "|", header = T, quote = "", fill = T, encoding = "UTF-8")

# update species names to match latest names in WCVP
updated_species_names = data.frame(species = unique(alldiv$species),species_updated = unique(alldiv$species))

updated_species_names$species_updated[updated_species_names$species == "Arabis nemorensis"] = "Arabis planisiliqua"
updated_species_names$species_updated[updated_species_names$species == "Dioscorea rotundata"] = "Dioscorea cayenensis subsp. rotundata"
updated_species_names$species_updated[updated_species_names$species == "Glycine soja"] = "Glycine max subsp. soja"
updated_species_names$species_updated[updated_species_names$species == "Lens ervoides"] = "Vicia lenticula"
updated_species_names$species_updated[updated_species_names$species == "Mimulus guttatus"] = "Erythranthe guttata"
updated_species_names$species_updated[updated_species_names$species == "Populus trichocarpa"] = "Populus tristis"
updated_species_names$species_updated[updated_species_names$species == "Raphanus sativus"] = "Raphanus raphanistrum subsp. sativus"
updated_species_names$species_updated[updated_species_names$species == "Salix dunnii"] = "Salix tetrasperma"
updated_species_names$species_updated[updated_species_names$species == "Solanum stenotomum"] = "Solanum tuberosum"

# loop over each species, and calculate range area in square kilometers
region_areas = NULL

for(species in updated_species_names$species_updated){

  print(species)
  
  # get native distribution
species_dist = wcvp_distribution(
taxon = species,
taxon_rank = "species",
native = TRUE,
introduced = FALSE,
extinct = TRUE,
location_doubtful = FALSE,
wcvp_distributions = wcvp_dist,
wcvp_names = wcvp_names
)

# calculate area
region_area = sum(st_area(species_dist))/1e6

region_areas = rbind(region_areas, data.frame(species_updated = species, area = region_area))

}

# merge range areas with old species names
region_areas = merge(updated_species_names, region_areas, by = "species_updated")

# remove weird units
region_areas$area = as.numeric(region_areas$area)

# merge with diversity data
alldiv = merge(alldiv, region_areas, by = "species")
```

## Load plant height data, a proxy for body size, which is a proxy for population density
```{r}
height = fread("../workflow/data/eol_plant_height_data_2023-10-03/records.tsv.gz", header = T, fill = T, quote = "", sep = "\t")

# convert feet to meters
height$final_values = height$`Measurement Value`
height$final_values[(height$`Measurement Unit` == "feet")] = height$final_values[(height$`Measurement Unit` == "feet")]*0.3048

# get height measurements for only species in my dataset
height_my_species = NULL
for(i in 1:nrow(alldiv)){
  # print progress
  print(i)
  
  # get species name(s)
  old_name = alldiv$species[i]
  new_name = alldiv$species_updated[i]
  
  # check old name
  height_sub = height[grepl(old_name, height$`Scientific Name`),]
  
  # if new name is different, check new species name
  if(new_name != old_name){
    height_sub = rbind(height_sub, height[grepl(new_name, height$`Scientific Name`),])
  }
  
  # average across values
  height_one_species = data.frame(species = old_name, height = mean(height_sub$final_values, na.rm = T))
  height_my_species = rbind(height_my_species, height_one_species)
}


# add height data to final dataframe
summary(height_my_species$height)
alldiv = merge(alldiv, height_my_species, all.x = T, by = "species")

# calculate quantity that's proportional to population size (range/squared height)
alldiv$popsize = (alldiv$area * 1e6)/(alldiv$height^2)
summary(alldiv$popsize)
```

# Initial data analysis

## relationship between coverage and diveristy
```{r eval=FALSE, include=FALSE}
ggplot(alldiv, aes(x = log10(totalbp), y = log10(pi), label = species)) +
  geom_point() +
  geom_text(position = position_jitter(seed = 1)) +
  theme_classic() +
  labs(x = "log10(Total base pairs sequenced across all samples)", y = "log10(Nucleotide diversity)")

#ggsave("bp-sequenced-vs-diversity.png")

ggplot(alldiv, aes(x = log10(nidv), y = log10(pi), label = species)) +
  geom_point() +
  geom_text(position = position_jitter(seed = 1)) +
  theme_classic() +
  labs(x = "log10(Number of individuals sequenced)", y = "log10(Nucleotide diversity)")

ggplot(alldiv, aes(x = cvbp, y = log10(pi), label = species)) +
  geom_point() +
  geom_text(position = position_jitter(seed = 1)) +
  theme_classic() +
  labs(x = "Coefficient of variation in base pairs sequenced across all samples", y = "log10(Nucleotide diversity)")

ggplot(alldiv, aes(x = log10(totalbp), y = bcd, label = species)) +
  geom_point() +
  geom_text(position = position_jitter(seed = 1)) +
  theme_classic() +
  labs(x = "log10(Total base pairs sequenced across all samples)", y = "Bray-Curtis Dissimilarity")
```

## distributions of variables
```{r eval=FALSE, include=FALSE}
# histograms of diversity data
ggplot(alldiv, aes(x = log10(pi))) +
  geom_histogram() +
  theme_classic() +
  labs(x = "log10(Nucleotide diversity)")

ggplot(alldiv, aes(x = jac)) +
  geom_histogram() +
  theme_classic() +
  labs(x = "Jaccard Dissimilarity")

ggplot(alldiv, aes(x = bcd)) +
  geom_histogram() +
  theme_classic() +
  labs(x = "Bray-Curtis Dissimilarity")
```

## correlations between variables
```{r}
# Does pi correlate with k-mer diversity?
cor(alldiv$pi, alldiv$bcd, method = "spearman", use = "pairwise.complete.obs")
cor(alldiv$pi, alldiv$jac, method = "spearman", use = "pairwise.complete.obs")

# Does k-mer diversity correlate with population size (i.e. range area) more than pi?
cor(alldiv$pi, alldiv$rangeArea, method = "spearman", use = "pairwise.complete.obs")
cor(alldiv$bcd, alldiv$rangeArea, method = "spearman", use = "pairwise.complete.obs")
cor(alldiv$jac, alldiv$rangeArea, method = "spearman", use = "pairwise.complete.obs")

# How does the number of individuals correlate with the diversity measures
# More individuals = more opportunities to discover variants
cor(alldiv$pi, alldiv$nidv)
cor(alldiv$bcd, alldiv$nidv)
cor(alldiv$jac, alldiv$nidv)

#
cor(alldiv$pi, alldiv$popsize, method = "spearman", use = "pairwise.complete.obs")
cor(alldiv$bcd, alldiv$popsize, method = "spearman", use = "pairwise.complete.obs")
```

# Confirmatory data analysis

## remove outliers before analysis
```{r}
# For now, remove species where invariant sites were not properly called
alldiv = alldiv[(alldiv$ninvariant > 0),]

# For now, remove species where variants sites were not properly called
alldiv = alldiv[(alldiv$nvariant > 500),]

# remove critically endangered species
alldiv = alldiv[(alldiv$rangeArea > 10),]

# remove species where there aren't many sampled individuals
alldiv = alldiv[(alldiv$nidv >= 10),]
```

## Combine tree and data to get comparative object
```{r}
# create comparative data object for caper models
alldiv = alldiv[!is.na(alldiv$species),]
compdata = comparative.data(phy, alldiv, names.col = "species", vcv = T)
plot(compdata$phy)
```

## fit models
```{r}
# compare effects with phylogeny vs without phylogeny
lm(log10(rangeArea) ~ log10(pi), alldiv)
lm(log10(rangeArea) ~ bcd, alldiv)
lm(log10(rangeArea) ~ jac, alldiv)

lmmod = lm(log10(rangeArea) ~ scale(bcd) + scale(nidv), alldiv)
summary(lmmod)

pi_vs_bcd = pgls(nidv ~ log10(pi), compdata)
summary(pi_vs_bcd)

pi_vs_bcd = pgls(bcd ~ log10(pi), compdata)
summary(pi_vs_bcd)
plot(pi_vs_bcd)

area_vs_pi = pgls(log10(rangeArea) ~ scale(pi), compdata)
summary(area_vs_pi)
plot(area_vs_pi)

area_vs_bcd = pgls(log10(rangeArea) ~ scale(bcd), compdata)
summary(area_vs_bcd)
plot(area_vs_bcd)

area_vs_jac = pgls(log10(rangeArea) ~ jac, compdata)
summary(area_vs_jac)

# Test if k-mer diversity has stronger effect on range area than SNP diversity
mod_bcd = pgls(log10(rangeArea) ~ scale(pi) + scale(bcd), compdata)
summary(mod_bcd)

# plot out coefficients
coefplot = data.frame(coef = coefficients(mod_bcd), err = mod_bcd$sterr, label = names(mod_bcd$sterr))
coefplot = coefplot[-1,]
coefplot$coef = abs(coefplot$coef)

ggplot(coefplot, aes(x=coef, y=label)) + 
  geom_point()+
  geom_errorbar(aes(xmin=coef-err, xmax=coef+err), width=.2, position=position_dodge(0.05)) +
  geom_vline(xintercept = 0, lty = "dotted", size = 2) +
  theme_classic() +
  theme(text = element_text(size = 20)) +
  labs(x = "Absolute value of coefficient", y = "PGLS model parameter")

ggsave(paste("pgls_coefficients_", today, ".png", sep = ""), width = 7, height = 7)

mod_jac = pgls(log10(rangeArea) ~ scale(pi) + scale(jac), compdata)
summary(mod_jac)

# control for the number of individuals sequenced
mod_bcd = pgls(log10(rangeArea) ~ scale(pi) + scale(bcd), compdata)
summary(mod_bcd)

mod_jac = pgls(log10(rangeArea) ~ scale(pi) + scale(jac) + scale(nidv), compdata)
summary(mod_jac)

# control for amount of sequencing data
mod_bcd = pgls(log10(rangeArea) ~ scale(bcd), compdata)
summary(mod_bcd)
plot(mod_bcd)

mod_pi = pgls(log10(rangeArea) ~ scale(pi), compdata)
summary(mod_pi)
plot(mod_pi)

cor(alldiv$nidv, alldiv$totalbp, use = "pairwise.complete.obs")
cor(log10(alldiv$pi), (alldiv$bcd), use = "pairwise.complete.obs", method = "spearman")
cor(alldiv$bcd, alldiv$totalbp, use = "pairwise.complete.obs", method = "spearman")
cor(alldiv$nidv, alldiv$totalbp/alldiv$nidv, method = "spearman")
#
mod_bcd = pgls(log10(rangeArea) ~ scale(bcd) + scale(nidv), compdata)
summary(mod_bcd)

mod_pi = pgls(log10(rangeArea) ~ scale(pi) + scale(nidv), compdata)
summary(mod_pi)

plot(mod_bcd)

# try contrasts instead of variance-covariance matrices
alldiv$scale_bcd = scale(alldiv$bcd)
alldiv$scale_pi = scale(alldiv$pi)
alldiv$scale_jac = scale(alldiv$jac)
alldiv$scale_nidv = scale(alldiv$nidv)
alldiv$scale_bp_per_idv = scale(alldiv$totalbp/alldiv$nidv)
alldiv$log_range = log10(alldiv$rangeArea)

compdata = comparative.data(phy, alldiv, names.col = "species", vcv = T)

crunch_mod = crunch(log_range ~ scale_bcd + scale_pi + scale_nidv + scale_bp_per_idv, data = compdata)
summary(crunch_mod)
plot(crunch_mod)
```

# Final plots
```{r}
# pi vs bcd
ggplot(alldiv, aes(x = bcd, y = log10(pi), label = species)) +
  geom_point() +
  geom_text(position = position_jitter(seed = 1)) +
  theme_classic() +
  xlim(0,0.8) +
  labs(x = "Bray-Curtis dissimilarity", y = "log10(Nucleotide diversity)")
ggsave(paste("../results/2023-03-26/figures/Bray-CurtisDissimilarityVsPi_", today, ".png", sep = ""), width = 7, height = 7, unit = "in", scale = 0.5)

# range vs pi
r1 = ggplot(alldiv, aes(x = log10(rangeArea), y = log10(pi), label = species)) +
  geom_point() +
  geom_text(position = position_jitter(seed = 1)) +
  geom_smooth(method = "auto") +
  theme_classic() +
  theme(text = element_text(size = 20)) +
  xlim(5.5, 8.5) +
  labs(x = "log10(Range area in square km)", y = "log10(Nucleotide diversity)")

#ggsave(paste("../results/2023-03-26/figures/rangeVsPi_", today, ".png", sep = ""), width = 7, height = 7, unit = "in", scale = 0.5)

# range vs bcd
r2 = ggplot(alldiv, aes(x = log10(rangeArea), y = bcd, label = species)) +
  geom_point() +
  geom_text(position = position_jitter(seed = 1)) + 
  geom_smooth(method = "auto") +
  theme_classic() +
  theme(text = element_text(size = 20)) +
  xlim(5.5, 8.5) +
  labs(x = "log10(Range area in square km)", y = "Bray-Curtis dissimilarity")

#ggsave(paste("../results/2023-03-26/figures/rangeVsBray-Curtis_", today, ".png", sep = ""), width = 7, height = 7, unit = "in", scale = 0.5)

# range vs jaccard dissimilarity
r3 = ggplot(alldiv, aes(x = log10(rangeArea), y = jac, label = species)) +
  geom_point() +
  geom_text(position = position_jitter(seed = 1)) + 
  geom_smooth(method = "auto") +
  theme_classic() +
  theme(text = element_text(size = 20)) +
  xlim(5.5, 8.5) +
  labs(x = "log10(Range area in square km)", y = "Jaccard dissimilarity")

gridplot = plot_grid(r1, r2, r3, labels = "AUTO", ncol = 3)

ggsave(paste("../results/2023-03-26/figures/range_vs_diversity_", today, ".png", sep = ""), width = 21, height = 7)
       
#ggsave(paste("../results/2023-03-26/figures/rangeVsJaccard_", today, ".png", sep = ""), width = 7, height = 7, unit = "in", scale = 0.5)
```

## plot phylogeny alongside data

ggtree causes linear models to crash, so need to do this separately

```{r}
#install.packages("ggtree")
#if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")
#BiocManager::install("ggtree")

#install.packages("ggtree")
library(ggtree)

# arrange plot data to match order of tip labels
plotdf = alldiv %>% arrange(factor(species, levels = phy$tip.label))

# remove species from phylogeny that lack data
plotphy = drop.tip(phy, tip = phy$tip.label[!(phy$tip.label %in% alldiv$species)])

# get color pallete
#plotpal = scico(5, palette = "lipari")
plotpal = c("#FFCE66","#C07348","#773339","#4C3D72","#5F88C3","#80E6FF")
#plotpal = c("#021326","#3B5378","#7F5F70","#CE685E","#E5AA7F","#FDF4D9")
#plotpal = c("#191900","#663329","#D85F4D","#ECAC54","#FFFECB")

# plot phylogeny alongside other data
ggtree(plotphy, size = 1.5) + 
    geom_facet(panel = "log10(Range area)", data = plotdf, geom = geom_segment, size = 3, mapping = aes(x = 0, y = y, yend = y, xend = log10(rangeArea)), color = plotpal[1]) +
    geom_facet(panel = "log10(SNP diversity)", data = plotdf, geom = geom_segment, size = 3, mapping = aes(x = 0, y = y, yend = y, xend = log10(pi)), color = plotpal[2]) +
  geom_facet(panel = "Bray-Curtis dissimilarity", data = plotdf, geom = geom_segment, size = 3, mapping = aes(x = 0, y = y, yend = y, xend = bcd), color = plotpal[3]) +
  geom_facet(panel = "Jaccard dissimilarity", data = plotdf, geom = geom_segment, size = 3, mapping = aes(x = 0, y = y, yend = y, xend = jac), color = plotpal[4]) +
  geom_facet(panel = "Individuals sequenced", data = plotdf, geom = geom_segment, size = 3, mapping = aes(x = 0, y = y, yend = y, xend = nidv), color = plotpal[5]) +
  geom_facet(panel = "bp sequenced per individual", data = plotdf, geom = geom_segment, size = 3, mapping = aes(x = 0, y = y, yend = y, xend = totalbp/nidv), color = plotpal[6]) +
  theme_tree2() +
  theme(strip.background =element_rect(fill="white")) +
  theme(strip.text = element_text(colour = 'black'))

ggsave(paste("../results/2023-03-26/tree_with_data_", today, ".png", sep = ""), scale = 2)
```