---
title: "2023-04-12 results"
output:
  html_document:
    df_print: paged
  pdf_document: default
---

# load packages
```{r}
rm(list = ls())

library(ggplot2)
library(caper)
library(stringr)
library(data.table)

today = Sys.Date()
```

# Data preparation

## Prepare diversity data 
```{r}
# List input files
pi = list.files("../results/2023-03-26/genome-wide-pi", full.names = T)
bcd = list.files("../results/2023-03-26/bray-curtis-dissimilarities", full.names = T)
jac = list.files("../results/2023-03-26/jaccard-dissimilarities", full.names = T)

# for now, remove species where invariant sites are not properly called
#pi = pi[c(-14:-15, -17:-19, -21)]
#jac = jac[c(-14:-15, -17:-19, -21, -23)]
#bcd = bcd[c(-14:-15, -17:-19, -21, -23)]

# get species IDs
species = gsub("_pairwise-pi.txt", "", gsub("../results/2023-03-26/genome-wide-pi/", "", pi))

# Load input files
pi = lapply(pi, fread, header = T, data.table = F)
bcd = lapply(bcd, fread, header = F, data.table = F)
jac = lapply(jac, fread, header = F, data.table = F)

# calculate mean pairwise dissimilarity
bcdMean = lapply(bcd, function(x){mean(x[,2])})
jacMean = lapply(jac, function(x){mean(x[,2])})

# calculate average number of k-mers included in union
unionMean = lapply(jac, function(x){mean(x[,4])})

# calculate number of individuals sampled for each dataset
# You get this equation by rewriting the formula for the number of pairwise comparisons (n^2 - n)/2 = N as a polynomial and applying the quadratic formula
backCalcN = function(x){
  sqrt(1/4 + 2*nrow(x)) + 1/2
}

nidv = lapply(bcd, backCalcN)

# cbind all diversity data into one dataframe per species
alldiv = Map(cbind, pi, bcdMean, jacMean, unionMean, nidv)

# Some species have extra data, remove for now
speciesWithExtraData = which(unlist(lapply(alldiv, ncol)) == 12)
for(i in speciesWithExtraData){
  alldiv[[i]] = alldiv[[i]][,c(1, 3:6, 9:12)]
}

# rbind all species data into one dataframe
alldiv = lapply(alldiv, setNames, c("h", "ntotal", "nvariant", "ninvariant", "pi", "bcd", "jac", "umean", "nidv"))

alldiv = do.call(rbind, alldiv)

# add species names
alldiv$species = species
alldiv$species = str_to_sentence(gsub("_", " ", alldiv$species))

# For now, remove species where invariant sites were not properly called
alldiv = alldiv[(alldiv$ninvariant > 0),]

# For now, remove species where variants sites were not properly called
alldiv = alldiv[(alldiv$nvariant > 0),]

# add coverage covariate
# coverage = read.table("../results/2023-03-26/bp_sequenced.txt", header = T, sep = "\t")
# alldiv = merge(alldiv, coverage, by = "species")
# 
# gsize = read.table("../results/2023-03-26/genome_size.txt", header = T, sep = "\t")
# alldiv = merge(alldiv, gsize, by = "species")
# 
# alldiv$totalcov = alldiv$totalbp/alldiv$assemblySize
```

## Load range data
```{r}
# load range data
areas = list.files("../results/2023-03-26/ranges/areas", full.names = T)

# extract species names
areas_species = gsub("_areas.txt", "", gsub("../results/2023-03-26/ranges/areas/", "", areas))
areas_species = str_to_sentence(gsub("_", " ", areas_species))

# load data
areas = lapply(areas, read.table, header = T, sep = "\t")

# add names to list
names(areas) = areas_species
```

## remove areas outside of native range 
```{r}
# Refer to plants of the world online for native range maps
# powo.science.kew.org

# Amaranthus hybridus native to only North and South America
areas$`Amaranthus hypochondriacus` = areas$`Amaranthus hypochondriacus`[c(4,6),]

# Arabidopsis thaliana is only native to Europe, Asia, and Africa
areas$`Arabidopsis thaliana` = areas$`Arabidopsis thaliana`[2:4,]

# Arachis glabrata is native to Brazil, remove North America
areas$`Arachis hypogaea` = areas$`Arachis hypogaea`[1,]

# Beta vulgaris subsp. vulgaris is native to only Europe and Africa
areas$`Beta vulgaris` = areas$`Beta vulgaris`[1:2,]

# Brachypodium distachyon is native to Europe, Asia, Africa
areas$`Brachypodium distachyon` = areas$`Brachypodium distachyon`[3:5,]

# Buddleja alternifolia is native to Asia
areas$`Buddleja alternifolia` = areas$`Buddleja alternifolia`[2,]

# Cucurbita palmata is only native to North America
areas$`Cucurbita pepo` = areas$`Cucurbita pepo`[1,]

# Malus sylvestris is only native to Europe
areas$`Malus sylvestris` = areas$`Malus sylvestris`[2,]

# Medicago truncatula only native to europe and africa
areas$`Medicago truncatula` = areas$`Medicago truncatula`[c(4,5),]

# Mimulus guttatus only native to North America
areas$`Mimulus guttatus` = areas$`Mimulus guttatus`[1,]

# Nicotiana glauca only native to South America
areas$`Nicotiana tabacum` = areas$`Nicotiana tabacum`[2,]
```

## Merge range data with diversity data
```{r}
# function to get non-water area
get_nonwater_area = function(x){
  sum(x[,"total_area"] - x[,"water_area"])
}

# format area data as frame
areas = data.frame(
  species = areas_species,
  rangeArea = unlist(lapply(areas, get_nonwater_area))
)

# merge with diversity data
alldiv = merge(alldiv, areas, by = "species", all.x = T)

print(alldiv)

# save final table
write.csv(alldiv, paste("../results/2023-03-26/diversityAndRangeData_", today, ".csv", sep = ""), row.names = F, quote = F)
```

## Prepare phylogenetic tree
```{r}
# load time tree
phy = read.tree("../results/2023-03-26/speciesList.nwk")
phy$tip.label = gsub("_", " ", phy$tip.label)

plot(phy)
```

## Combine tree and data to get comparative object
```{r}
compdata = comparative.data(phy, alldiv, names.col = "species", vcv = T)
```

# Initial data analysis
```{r}
# histograms of diversity data
ggplot(alldiv, aes(x = log10(pi))) +
  geom_histogram() +
  theme_classic() +
  labs(x = "log10(Nucleotide diversity)")

ggplot(alldiv, aes(x = jac)) +
  geom_histogram() +
  theme_classic() +
  labs(x = "Jaccard Dissimilarity")

ggplot(alldiv, aes(x = bcd)) +
  geom_histogram() +
  theme_classic() +
  labs(x = "Bray-Curtis Dissimilarity")
```

# drop outliers?
```{r eval=FALSE, include=FALSE}
compdata = subset(compdata, subset = pi > 1e-7)

alldiv = alldiv[(alldiv$pi > 1e-7),]
```

# Confirmatory data analysis
```{r}
# Does pi correlate with k-mer diversity?
cor(alldiv$pi, alldiv$bcd, method = "spearman", use = "pairwise.complete.obs")
cor(alldiv$pi, alldiv$jac, method = "spearman", use = "pairwise.complete.obs")

# Does k-mer diversity correlate with population size (i.e. range area) more than pi?
cor(alldiv$pi, alldiv$rangeArea, method = "spearman", use = "pairwise.complete.obs")
cor(alldiv$bcd, alldiv$rangeArea, method = "spearman", use = "pairwise.complete.obs")
cor(alldiv$jac, alldiv$rangeArea, method = "spearman", use = "pairwise.complete.obs")

# compare effects with phylogeny vs without phylogeny
lm(log10(rangeArea) ~ log10(pi), alldiv)
lm(log10(rangeArea) ~ bcd, alldiv)
lm(log10(rangeArea) ~ jac, alldiv)

pgls(log10(rangeArea) ~ log10(pi), compdata)
pgls(log10(rangeArea) ~ bcd, compdata)
pgls(log10(rangeArea) ~ jac, compdata)


# Test if k-mer diversity has stronger effect on range area than SNP diversity
mod_bcd = pgls(log10(rangeArea) ~ scale(pi) + scale(bcd), compdata)
summary(mod_bcd)

mod_jac = pgls(log10(rangeArea) ~ scale(pi) + scale(jac), compdata)
summary(mod_jac)

# control for the number of individuals sequenced
mod_bcd = pgls(log10(rangeArea) ~ scale(pi) + scale(bcd) + scale(nidv), compdata)
summary(mod_bcd)

mod_jac = pgls(log10(rangeArea) ~ scale(pi) + scale(jac) + scale(nidv), compdata)
summary(mod_jac)
  
# try removing outliers
# compdataNoOut = subset.comparative.data(compdata, subset =  !(species %in% c("Actinidia chinensis", "Camelina sativa")))
```

# Final plots
```{r}
# pi vs bcd
ggplot(alldiv, aes(x = bcd, y = log10(pi), label = species)) +
  geom_point() +
  geom_text(position = position_jitter(seed = 1)) +
  theme_classic() +
  xlim(0,0.8) +
  labs(x = "Bray-Curtis dissimilarity", y = "log10(Nucleotide diversity)")
ggsave(paste("../results/2023-03-26/figures/Bray-CurtisDissimilarityVsPi_", today, ".png", sep = ""), width = 7, height = 7, unit = "in", scale = 0.5)

# range vs pi
ggplot(alldiv, aes(x = log10(rangeArea), y = log10(pi), label = species)) +
  geom_point() +
  geom_text(position = position_jitter(seed = 1)) +
  theme_classic() +
  xlim(5.5, 8.5) +
  labs(x = "log10(Range area in square km)", y = "log10(Nucleotide diversity)")

ggsave(paste("../results/2023-03-26/figures/rangeVsPi_", today, ".png", sep = ""), width = 7, height = 7, unit = "in", scale = 0.5)

# range vs bcd
ggplot(alldiv, aes(x = log10(rangeArea), y = bcd, label = species)) +
  geom_point() +
  geom_text(position = position_jitter(seed = 1)) + 
  theme_classic() +
  xlim(5.5, 8.5) +
  labs(x = "log10(Range area in square km)", y = "Bray-Curtis dissimilarity")

ggsave(paste("../results/2023-03-26/figures/rangeVsBray-Curtis_", today, ".png", sep = ""), width = 7, height = 7, unit = "in", scale = 0.5)

# range vs jaccard dissimilarity
ggplot(alldiv, aes(x = log10(rangeArea), y = jac, label = species)) +
  geom_point() +
  geom_text(position = position_jitter(seed = 1)) + 
  theme_classic() +
  xlim(5.5, 8.5) +
  labs(x = "log10(Range area in square km)", y = "Jaccard dissimilarity")
ggsave(paste("../results/2023-03-26/figures/rangeVsJaccard_", today, ".png", sep = ""), width = 7, height = 7, unit = "in", scale = 0.5)
```